---
output: html_document
editor_options: 
  chunk_output_type: console
---

# Calculating genetic distances between protein sequences

<!-- add data pre -->

A common first step in performing a phylogenetic analysis is to calculate the **pairwise genetic distances** between sequences. The genetic distance is an estimate of the divergence between two sequences, and is usually measured in quantity of evolutionary change (an estimate of the number of mutations that have occurred since the two sequences shared a common ancestor).

We can calculate the genetic distances between protein sequences using the `dist.alignment()` function in the `seqinr` package. The `dist.alignment()` function takes a multiple sequence alignment as input. Based on the multiple alignment that you give it, `dist.alignment()` calculates the **genetic distance** between each pair of proteins in the multiple alignment. For example, to calculate genetic distances between the virus phosphoproteins based on the multiple sequence alignment stored in virusaln, we type:

```{r, eval = F}
virusdist <- dist.alignment(x)                 # Calculate the genetic distances
```

Print out the genetic distance
```{r, eval = F}
virusdist
```


The genetic distance matrix above shows the genetic distance between each pair of proteins.

The sequences are referred to by their UniProt accessions.  These are phosphoproteins from the following organisms or samples:

1. P06747 = rabies virus 
1. P0C569 = Mokola virus 
1. O56773 = Lagos bat virus  
1. Q5VKP1 = Western Caucasian bat virus 

Based on the genetic distance matrix above, we can see that the genetic distance between Lagos bat virus phosphoprotein (O56773) and Mokola virus phosphoprotein (P0C569) is smallest (about 0.414).

<!-- (these values are different - check!  originals are from Clustal, these are from muscle...) -->

Similarly, the genetic distance between Western Caucasian bat virus phosphoprotein (Q5VKP1) and Lagos bat virus phosphoprotein (O56773) is the biggest (about 0.507).

The larger the genetic distance between two sequences, the more amino acid changes or indels that have occurred since they shared a common ancestor, and the longer ago this common ancestor probably lived.

## Calculating genetic distances between DNA/mRNA sequences

Just like for protein sequences, you can calculate genetic distances between DNA (or mRNA) sequences based on an alignment of the sequences.

For example, the NCBI accession AF049118 contains mRNA sequence for Mokola virus phosphoprotein, RefSeq AF049114 contains mRNA sequence for Mokola virus phosphoprotein, and AF049119 contains the mRNA sequence for Lagos bat virus phosphoprotein, while AF049115 contains the mRNA sequence for Duvenhage virus phosphoprotein.

To retrieve these sequences from the NCBI database, we can search the ACNUC “genbank” sub-database (since these are nucleotide sequences), by typing:

Make a vector containing the names of the sequences
```{r, eval = F}
seqnames <- c("AF049118", "AF049114", "AF049119", "AF049115")  # 
```


 Retrieve the sequences and store them in list variable "seqs"
```{r, eval = F}
seqs <- retrieveseqs(seqnames,"genbank")                       #
```

We can then write out the sequences to a FASTA-format file by typing:

```{r, eval = F}
write.fasta(seqs, seqnames, file="virusmRNA.fasta")
```

We can then use CLUSTAL to create a PHYLIP-format alignment of the sequences, and store it in the alignment file “virusmRNA.phy”. This picture shows part of the alignment:


We can then read the alignment into R:

```{r, eval = F}
virusmRNAaln  <- read.alignment(file = "virusmRNA.phy", format = "phylip")
```

We saw above that the function `dist.alignment()` can be used to calculate a genetic distance matrix based on a protein sequence alignment.

You can calculate a genetic distance for DNA or mRNA sequences using the `dist.dna()` function in the `ape` R package. `dist.dna()` takes a multiple alignment of DNA or mRNA sequences as its input, and calculates the genetic distance between each pair of DNA sequences in the multiple alignment.

The `dist.dna()` function requires the input alignment to be in a special format known as **DNAbin** , so we must use the `as.DNAbin()` function to convert our DNA alignment into this format before using `dist.dna()`.

For example, to calculate the genetic distance between each pair of mRNA sequences for the virus phosphoproteins, we type:

Convert the alignment to "DNAbin" format
```{r, eval = F}
virusmRNAalnbin <- as.DNAbin(virusmRNAaln) 
```

Calculate the genetic distance matrix
```{r, eval = F}
virusmRNAdist <- dist.dna(virusmRNAalnbin) 
```
 
Print out the genetic distance matrix
```{r, eval = F}
virusmRNAdist                              
```


