---
output: html_document
editor_options: 
  chunk_output_type: console
---
# Histograms

## Preliminaries



### Load packages
```{r}
library(compbio4all)
library(ggplot2)
library(cowplot)
library(ggpubr)
```


```{r}
Cvalue <- c(3.63, 3.25, 2.26, 3.43, 3.48,
  3.98, 4.90, 4.02, 3.61, 3.52,
  3.36, 3.36, 3.4,  4.08, 3.93,
  3.57, 3.23, 3.5,  3.08, 3.37,
  3.54, 3.28, 3.25, 3.76, 3.41,
  3.56, 3.29, 3.56, 3.33, 3.5,
  3.5,  3.12, 3.67, 4.32, 3.58,
  3.61, 3.63, 3.53, 3.68, 3.66,
  3.3, 5.36)
i <- 1:length(Cvalue)
df <- data.frame(i,Cvalue)

gghistogram(data = df, x = "Cvalue")

Cvalue <- sort(Cvalue)
Cvalue.rounded <- round(Cvalue,digits = 1)
length(seq(2.3,5.4, by = 0.1))

df2 <- data.frame(i,Cvalue.rounded)

gghistogram(data = df2, x = "Cvalue.rounded")

x <- Cvalue.rounded

x.unique <- as.numeric(names(table(x)))

y <- rep(0, length(x))
for(i in 1:length(x.unique)){
  x.i <- which(Cvalue.rounded == x.unique[i])
  y[x.i] <- 1:length(x.i)
}

df3 <- data.frame(name = 1:length(x),
                    C.value = x,
                  y.value = c(y-0.5))
ggscatter(data = df3,
          x = "C.value",
          y = "y.value",
         # shape = 0,
         shape = 15,
          size = 8#,
          #color = "white",
          #label.rectangle = T,
          #label = "name" 
          ) +
geom_vline(xintercept = c(seq(2.25,5.45,0.1)))+
  geom_hline(yintercept = c(seq(0,9,1))) +
  scale_x_continuous(breaks=seq(2.2,5.5,0.1))
```




### Load data

"hpc" stands for "human protein-coding" genes.

```{r}
data("genes_hpc")
data("frogarms")
```


##


## Examples of histogram



### Normal-ish data
A normal curve looking one

```{r}
gghistogram(data = frogarms,
            x = "forearm")
```


## Skewed data

```{r}
gghistogram(data = genes_hpc,
            x = "GCpercent")
```

## Heavily skewed data

```{r}
gghistogram(data = genes_hpc,
            x = "length")
```

Usually a warning
Warning message:
Using `bins = 30` by default. Pick better value with the argument `bins`



Distribution has a few outliers




```{r}
max(genes_hpc$length)
```

```{r}
ges <- which(genes_hpc$length > 15000)
```



```{r}
i.MT <- which(genes_hpc$chromosome == "MT")
i.X <- which(genes_hpc$chromosome == "X")
i.Y <- which(genes_hpc$chromosome == "Y")
i.1 <- which(genes_hpc$chromosome == "1")
i.15 <- which(genes_hpc$chromosome == "15")
i.22 <- which(genes_hpc$chromosome == "22")
```



## 

X chromosome looks nice
Y chromosome goofy
1 looks ok, large peak
15 ok, large peak
22 better than 1
```{r}
gghistogram(data = genes_hpc[i.22,],
            x = "length",
            bins = 30)
```


```{r}
gghistogram(data = genes_hpc,
            x = "length",
            facet.by = "chromosome",
            bins = 30)
```



## Fill


Add colored fill; note that it is "fill" not "color".  Color changes the color of the lines
```{r}
gghistogram(data = genes_hpc,
            x = "length",
            fill = "red",
            xlab = "length (grams)")
```



## Central tendency

```{r}
gghistogram(data = genes_hpc,
            x = "length",
            fill = "red",
            xlab = "length (grams)",
            add = "mean")
```

Warning messages:
1: Using `bins = 30` by default. Pick better value with the argument `bins`. 
2: geom_vline(): Ignoring `mapping` because `xintercept` was provided. 
3: geom_vline(): Ignoring `data` because `xintercept` was provided.


## Median
```{r}
gghistogram(data = genes_hpc,
            x = "length",
            fill = "red",
            xlab = "length (grams)",
            add = "median")
```



Add title
not usually done for publication but useful for keeping track of things and for presentations
```{r}

gghistogram(data = genes_hpc,
            x = "length",
            fill = "red",
            xlab = "length (grams)",
          main = "length of Australian frogs ") #Main title

```


## Density plot

```{r}

ggdensity(data = genes_hpc,
            x = "length",
            fill = "red",
            xlab = "length (grams)",
          main = "length of Australian frogs ") #Main title

```

### Plotting multple plots with cowplot::plot_grid

make into boxplot version exercise since shown first here as hist

```{r, eval = T, echo = T}

my.frogs <- make_my_data2L(dat = frogarms, 
                           my.code = "nlb24", # #<=  change this!
                           cat.var = "sex",
                           n.sample = 20, 
                           with.rep = FALSE)
dim(my.frogs)
```


We can save a plot to an R object
```{r, warning = F}
gg.hist.my.frogs <- gghistogram(data = my.frogs,
            x = "sv.length",
            fill = "red",
            xlab = "length (grams)") #Main title


gg.hist.frog <- gghistogram(data = frogarms,
            x = "sv.length",
            fill = "red",
            xlab = "length (grams)") #Main title

```



Now plot both

```{r}
plot_grid(gg.hist.my.frogs,
          gg.hist.frog)
```



Add labels.  Note that alignment is off sometimes.

```{r}
plot_grid(gg.hist.my.frogs, 
          gg.hist.frog,
          labels = c("a)My fogs","b)All the frogs"))
```


Add lines for means


We can save a plot to an R object
```{r, warnings = F}
gg.hist.my.frogs <- gghistogram(data = my.frogs,
            x = "sv.length",
            fill = "red",
            add = "mean",
            xlab = "length (grams)") #Main title


gg.hist.frog <- gghistogram(data = frogarms,
            x = "sv.length",
            fill = "red",
            add = "mean",
            xlab = "length (grams)") #Main title

```



Now plot both

```{r}
plot_grid(gg.hist.my.frogs,
          gg.hist.frog)
```

## Align vert

```{r}
plot_grid(gg.hist.my.frogs, 
          gg.hist.frog,
          nrow = 2)
```


## Density plot overlay

## Density plot

```{r}

ggdensity(data = frogarms,
            x = "sv.length",
            fill = "sex",
            xlab = "length (grams)",
          main = "length of Australian frogs ") #Main title

```

