<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 20 Assignment: Building dataframes by hand | Computational Biology for All!</title>
  <meta name="description" content="Computational Biology textbook" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 20 Assignment: Building dataframes by hand | Computational Biology for All!" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Computational Biology textbook" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 20 Assignment: Building dataframes by hand | Computational Biology for All!" />
  
  <meta name="twitter:description" content="Computational Biology textbook" />
  

<meta name="author" content="Nathan L. BRouwer" />


<meta name="date" content="2021-08-13" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="load-data-from-internet.html"/>
<link rel="next" href="dataframe-by-hand-problem-set-key.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Computational biology and bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a></li>
<li class="chapter" data-level="2" data-path="how-to-review-this-book.html"><a href="how-to-review-this-book.html"><i class="fa fa-check"></i><b>2</b> How to review this book</a>
<ul>
<li class="chapter" data-level="2.1" data-path="how-to-review-this-book.html"><a href="how-to-review-this-book.html#file-format"><i class="fa fa-check"></i><b>2.1</b> File format</a></li>
<li class="chapter" data-level="2.2" data-path="how-to-review-this-book.html"><a href="how-to-review-this-book.html#what-can-you-do"><i class="fa fa-check"></i><b>2.2</b> What can you do?</a></li>
<li class="chapter" data-level="2.3" data-path="how-to-review-this-book.html"><a href="how-to-review-this-book.html#adding-comments"><i class="fa fa-check"></i><b>2.3</b> Adding comments</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="how-to-review-this-book.html"><a href="how-to-review-this-book.html#html-tagged-comments"><i class="fa fa-check"></i><b>2.3.1</b> HTML-tagged comments</a></li>
<li class="chapter" data-level="2.3.2" data-path="how-to-review-this-book.html"><a href="how-to-review-this-book.html#code-chunk-comments"><i class="fa fa-check"></i><b>2.3.2</b> Code chunk comments</a></li>
<li class="chapter" data-level="2.3.3" data-path="how-to-review-this-book.html"><a href="how-to-review-this-book.html#keys-to-exercise"><i class="fa fa-check"></i><b>2.3.3</b> Keys to exercise</a></li>
<li class="chapter" data-level="2.3.4" data-path="how-to-review-this-book.html"><a href="how-to-review-this-book.html#problem---fix-this-code"><i class="fa fa-check"></i><b>2.3.4</b> Problem - fix this code</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="working-outline.html"><a href="working-outline.html"><i class="fa fa-check"></i>Working outline</a>
<ul>
<li class="chapter" data-level="2.4" data-path="working-outline.html"><a href="working-outline.html#todo"><i class="fa fa-check"></i><b>2.4</b> TODO</a></li>
<li class="chapter" data-level="2.5" data-path="working-outline.html"><a href="working-outline.html#general-book-conventions"><i class="fa fa-check"></i><b>2.5</b> General book conventions</a></li>
<li class="chapter" data-level="2.6" data-path="working-outline.html"><a href="working-outline.html#book-names-to-use-riff-off-of-for-book-names"><i class="fa fa-check"></i><b>2.6</b> Book names to use / riff off of for BOOK names</a></li>
<li class="chapter" data-level="2.7" data-path="working-outline.html"><a href="working-outline.html#outline"><i class="fa fa-check"></i><b>2.7</b> Outline</a>
<ul>
<li class="chapter" data-level="2.7.1" data-path="working-outline.html"><a href="working-outline.html#book-1-introduction-programming-with-data-chambers-1998"><i class="fa fa-check"></i><b>2.7.1</b> BOOK 1: Introduction: Programming with data (Chambers 1998)</a></li>
<li class="chapter" data-level="2.7.2" data-path="working-outline.html"><a href="working-outline.html#book-2-biology-by-the-numbers-burton"><i class="fa fa-check"></i><b>2.7.2</b> BOOK 2: “Biology by the numbers” (Burton)</a></li>
<li class="chapter" data-level="2.7.3" data-path="working-outline.html"><a href="working-outline.html#book-3-dimension-reduction-or-boiling-down-the-numbers"><i class="fa fa-check"></i><b>2.7.3</b> BOOK 3: (Dimension reduction, or boiling down the numbers)</a></li>
<li class="chapter" data-level="2.7.4" data-path="working-outline.html"><a href="working-outline.html#book-4-sequences"><i class="fa fa-check"></i><b>2.7.4</b> BOOK 4: Sequences</a></li>
<li class="chapter" data-level="2.7.5" data-path="working-outline.html"><a href="working-outline.html#book-5-tree-thinking"><i class="fa fa-check"></i><b>2.7.5</b> BOOK 5: “Tree thinking”</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="working-outline.html"><a href="working-outline.html#ideal-features-of-book"><i class="fa fa-check"></i><b>2.8</b> Ideal features of book</a></li>
<li class="chapter" data-level="2.9" data-path="working-outline.html"><a href="working-outline.html#test-code"><i class="fa fa-check"></i><b>2.9</b> Test code</a>
<ul>
<li class="chapter" data-level="2.9.1" data-path="working-outline.html"><a href="working-outline.html#this-is-a-default-sized-plot"><i class="fa fa-check"></i><b>2.9.1</b> This is a default-sized plot</a></li>
<li class="chapter" data-level="2.9.2" data-path="working-outline.html"><a href="working-outline.html#fig.height-5-fig.width-7"><i class="fa fa-check"></i><b>2.9.2</b> fig.height = 5, fig.width = 7</a></li>
<li class="chapter" data-level="2.9.3" data-path="working-outline.html"><a href="working-outline.html#fig.height-4-fig.width-6"><i class="fa fa-check"></i><b>2.9.3</b> fig.height = 4, fig.width = 6</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path=""><a href="#part-introduction"><i class="fa fa-check"></i>(PART) Introduction</a>
<ul>
<li class="chapter" data-level="" data-path=""><a href="#predface"><i class="fa fa-check"></i>Predface</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>3</b> Welcome to computational biology for all!</a>
<ul>
<li class="chapter" data-level="3.1" data-path="intro.html"><a href="intro.html#r-and-rstudio---your-new-best-friends"><i class="fa fa-check"></i><b>3.1</b> R and RStudio - your new best friends!</a></li>
<li class="chapter" data-level="3.2" data-path="intro.html"><a href="intro.html#how-to-use-this-book---be-an-active-learner"><i class="fa fa-check"></i><b>3.2</b> How to use this book - be an active learner!</a></li>
<li class="chapter" data-level="3.3" data-path="intro.html"><a href="intro.html#biological-scope-of-this-book"><i class="fa fa-check"></i><b>3.3</b> Biological scope of this book</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="what-is-computational-biology.html"><a href="what-is-computational-biology.html"><i class="fa fa-check"></i><b>4</b> What is Computational Biology?</a>
<ul>
<li class="chapter" data-level="4.1" data-path="what-is-computational-biology.html"><a href="what-is-computational-biology.html#notes"><i class="fa fa-check"></i><b>4.1</b> Notes</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="what-is-computational-biology.html"><a href="what-is-computational-biology.html#nih-definition"><i class="fa fa-check"></i><b>4.1.1</b> NIH definition</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="what-is-computational-biology.html"><a href="what-is-computational-biology.html#luscombe-et-al-2001"><i class="fa fa-check"></i><b>4.2</b> Luscombe et al 2001</a></li>
<li class="chapter" data-level="4.3" data-path="what-is-computational-biology.html"><a href="what-is-computational-biology.html#taprich-et-al-20121"><i class="fa fa-check"></i><b>4.3</b> Taprich et al 20121</a></li>
<li class="chapter" data-level="4.4" data-path="what-is-computational-biology.html"><a href="what-is-computational-biology.html#smith-2015-frontiers-in-genetics"><i class="fa fa-check"></i><b>4.4</b> Smith 2015 Frontiers in Genetics</a></li>
<li class="chapter" data-level="4.5" data-path="what-is-computational-biology.html"><a href="what-is-computational-biology.html#good-readings-references"><i class="fa fa-check"></i><b>4.5</b> Good readings / references</a></li>
<li class="chapter" data-level="4.6" data-path="what-is-computational-biology.html"><a href="what-is-computational-biology.html#potential-references"><i class="fa fa-check"></i><b>4.6</b> Potential references</a></li>
<li class="chapter" data-level="4.7" data-path="what-is-computational-biology.html"><a href="what-is-computational-biology.html#original-info-from-eco-data-science-project"><i class="fa fa-check"></i><b>4.7</b> Original info, from Eco Data Science project</a></li>
<li class="chapter" data-level="4.8" data-path="what-is-computational-biology.html"><a href="what-is-computational-biology.html#what-is-data-science"><i class="fa fa-check"></i><b>4.8</b> What is data science?</a></li>
<li class="chapter" data-level="4.9" data-path="what-is-computational-biology.html"><a href="what-is-computational-biology.html#refereces"><i class="fa fa-check"></i><b>4.9</b> Refereces</a></li>
<li class="chapter" data-level="4.10" data-path="what-is-computational-biology.html"><a href="what-is-computational-biology.html#bibliography"><i class="fa fa-check"></i><b>4.10</b> Bibliography</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="how-will-this-book-teach-computational-biology.html"><a href="how-will-this-book-teach-computational-biology.html"><i class="fa fa-check"></i><b>5</b> How will this book teach computational biology?</a>
<ul>
<li class="chapter" data-level="5.1" data-path="how-will-this-book-teach-computational-biology.html"><a href="how-will-this-book-teach-computational-biology.html#what-youll-learn"><i class="fa fa-check"></i><b>5.1</b> What you’ll learn</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="how-will-this-book-teach-computational-biology.html"><a href="how-will-this-book-teach-computational-biology.html#general-skills-that-youll-learn"><i class="fa fa-check"></i><b>5.1.1</b> General skills that you’ll learn</a></li>
<li class="chapter" data-level="5.1.2" data-path="how-will-this-book-teach-computational-biology.html"><a href="how-will-this-book-teach-computational-biology.html#computational-biology-skills-youll-learn"><i class="fa fa-check"></i><b>5.1.2</b> Computational biology skills you’ll learn</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="how-will-this-book-teach-computational-biology.html"><a href="how-will-this-book-teach-computational-biology.html#teaching-approach"><i class="fa fa-check"></i><b>5.2</b> Teaching approach</a></li>
<li class="chapter" data-level="5.3" data-path="how-will-this-book-teach-computational-biology.html"><a href="how-will-this-book-teach-computational-biology.html#requirements"><i class="fa fa-check"></i><b>5.3</b> Requirements</a></li>
<li class="chapter" data-level="5.4" data-path="how-will-this-book-teach-computational-biology.html"><a href="how-will-this-book-teach-computational-biology.html#whats-not-in-this-book-course"><i class="fa fa-check"></i><b>5.4</b> What’s not in this book / course</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="what-is-r-and-why-use-it.html"><a href="what-is-r-and-why-use-it.html"><i class="fa fa-check"></i><b>6</b> What is R and why use it?</a>
<ul>
<li class="chapter" data-level="6.1" data-path="what-is-r-and-why-use-it.html"><a href="what-is-r-and-why-use-it.html#how-do-we-typically-use-software-in-science"><i class="fa fa-check"></i><b>6.1</b> How do we typically use software in science?</a></li>
<li class="chapter" data-level="6.2" data-path="what-is-r-and-why-use-it.html"><a href="what-is-r-and-why-use-it.html#what-does-r-do"><i class="fa fa-check"></i><b>6.2</b> What does R do?</a></li>
<li class="chapter" data-level="6.3" data-path="what-is-r-and-why-use-it.html"><a href="what-is-r-and-why-use-it.html#why-use-r"><i class="fa fa-check"></i><b>6.3</b> Why use R</a></li>
<li class="chapter" data-level="6.4" data-path="what-is-r-and-why-use-it.html"><a href="what-is-r-and-why-use-it.html#who-uses-it"><i class="fa fa-check"></i><b>6.4</b> Who uses it?</a></li>
<li class="chapter" data-level="6.5" data-path="what-is-r-and-why-use-it.html"><a href="what-is-r-and-why-use-it.html#r-and-computational-reproducibility"><i class="fa fa-check"></i><b>6.5</b> R and computational reproducibility</a></li>
<li class="chapter" data-level="6.6" data-path="what-is-r-and-why-use-it.html"><a href="what-is-r-and-why-use-it.html#alternatives-to-r"><i class="fa fa-check"></i><b>6.6</b> Alternatives to R</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="the-layout-of-rstudio.html"><a href="the-layout-of-rstudio.html"><i class="fa fa-check"></i><b>7</b> The layout of RStudio</a>
<ul>
<li class="chapter" data-level="7.1" data-path="the-layout-of-rstudio.html"><a href="the-layout-of-rstudio.html#rstudio-desktop-or-cloud"><i class="fa fa-check"></i><b>7.1</b> RStudio (Desktop or Cloud)</a></li>
<li class="chapter" data-level="7.2" data-path="the-layout-of-rstudio.html"><a href="the-layout-of-rstudio.html#r-emulators"><i class="fa fa-check"></i><b>7.2</b> R emulators</a></li>
<li class="chapter" data-level="7.3" data-path="the-layout-of-rstudio.html"><a href="the-layout-of-rstudio.html#other-ways"><i class="fa fa-check"></i><b>7.3</b> Other ways</a></li>
<li class="chapter" data-level="7.4" data-path="the-layout-of-rstudio.html"><a href="the-layout-of-rstudio.html#rstudio-at-a-glance"><i class="fa fa-check"></i><b>7.4</b> RStudio at a glance</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="the-layout-of-rstudio.html"><a href="the-layout-of-rstudio.html#the-console-versus-the-script-editor"><i class="fa fa-check"></i><b>7.4.1</b> The console versus the script editor</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="book-getting-started-with-r.html"><a href="book-getting-started-with-r.html"><i class="fa fa-check"></i>(BOOK) Getting started with R</a>
<ul>
<li><a href="book-getting-started-with-r.html#section"></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="hello-r-a-first-encounter-with-data-visualization.html"><a href="hello-r-a-first-encounter-with-data-visualization.html"><i class="fa fa-check"></i><b>8</b> Hello R! A first encounter with data visualization</a>
<ul>
<li class="chapter" data-level="8.1" data-path="hello-r-a-first-encounter-with-data-visualization.html"><a href="hello-r-a-first-encounter-with-data-visualization.html#key-ideas"><i class="fa fa-check"></i><b>8.1</b> Key Ideas</a></li>
<li class="chapter" data-level="8.2" data-path="hello-r-a-first-encounter-with-data-visualization.html"><a href="hello-r-a-first-encounter-with-data-visualization.html#data-in-r"><i class="fa fa-check"></i><b>8.2</b> Data in R</a></li>
<li class="chapter" data-level="8.3" data-path="hello-r-a-first-encounter-with-data-visualization.html"><a href="hello-r-a-first-encounter-with-data-visualization.html#loading-data-that-comes-with-r"><i class="fa fa-check"></i><b>8.3</b> Loading data that comes with R</a></li>
<li class="chapter" data-level="8.4" data-path="hello-r-a-first-encounter-with-data-visualization.html"><a href="hello-r-a-first-encounter-with-data-visualization.html#plotting-simple-datasets-with-plot"><i class="fa fa-check"></i><b>8.4</b> Plotting simple datasets with <code>plot()</code></a></li>
<li class="chapter" data-level="8.5" data-path="hello-r-a-first-encounter-with-data-visualization.html"><a href="hello-r-a-first-encounter-with-data-visualization.html#commands-in-r"><i class="fa fa-check"></i><b>8.5</b> Commands in R</a></li>
<li class="chapter" data-level="8.6" data-path="hello-r-a-first-encounter-with-data-visualization.html"><a href="hello-r-a-first-encounter-with-data-visualization.html#the-structure-of-commands-in-r"><i class="fa fa-check"></i><b>8.6</b> The structure of commands in R</a></li>
<li class="chapter" data-level="8.7" data-path="hello-r-a-first-encounter-with-data-visualization.html"><a href="hello-r-a-first-encounter-with-data-visualization.html#arguements-in-r"><i class="fa fa-check"></i><b>8.7</b> Arguements in R</a></li>
<li class="chapter" data-level="8.8" data-path="hello-r-a-first-encounter-with-data-visualization.html"><a href="hello-r-a-first-encounter-with-data-visualization.html#arguements-and-more-arguments"><i class="fa fa-check"></i><b>8.8</b> Arguements and more arguments</a></li>
<li class="chapter" data-level="8.9" data-path="hello-r-a-first-encounter-with-data-visualization.html"><a href="hello-r-a-first-encounter-with-data-visualization.html#multiple-arguments-at-the-same-time"><i class="fa fa-check"></i><b>8.9</b> Multiple arguments at the same time</a></li>
<li class="chapter" data-level="8.10" data-path="hello-r-a-first-encounter-with-data-visualization.html"><a href="hello-r-a-first-encounter-with-data-visualization.html#r-commands-and-line-breaks"><i class="fa fa-check"></i><b>8.10</b> R commands and line breaks</a></li>
<li class="chapter" data-level="8.11" data-path="hello-r-a-first-encounter-with-data-visualization.html"><a href="hello-r-a-first-encounter-with-data-visualization.html#code-comments"><i class="fa fa-check"></i><b>8.11</b> Code comments</a></li>
<li class="chapter" data-level="8.12" data-path="hello-r-a-first-encounter-with-data-visualization.html"><a href="hello-r-a-first-encounter-with-data-visualization.html#a-note-on-plotting"><i class="fa fa-check"></i><b>8.12</b> A note on plotting</a></li>
<li class="chapter" data-level="8.13" data-path="hello-r-a-first-encounter-with-data-visualization.html"><a href="hello-r-a-first-encounter-with-data-visualization.html#now-you-try-it"><i class="fa fa-check"></i><b>8.13</b> Now you try it</a>
<ul>
<li class="chapter" data-level="8.13.1" data-path="hello-r-a-first-encounter-with-data-visualization.html"><a href="hello-r-a-first-encounter-with-data-visualization.html#easy-tasks"><i class="fa fa-check"></i><b>8.13.1</b> Easy tasks</a></li>
<li class="chapter" data-level="8.13.2" data-path="hello-r-a-first-encounter-with-data-visualization.html"><a href="hello-r-a-first-encounter-with-data-visualization.html#intermediate-tasks"><i class="fa fa-check"></i><b>8.13.2</b> Intermediate tasks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="downloading-r-packages-and-their-data.html"><a href="downloading-r-packages-and-their-data.html"><i class="fa fa-check"></i><b>9</b> Downloading R packages (and their data)</a>
<ul>
<li class="chapter" data-level="9.1" data-path="downloading-r-packages-and-their-data.html"><a href="downloading-r-packages-and-their-data.html#loading-data-from-r-packages"><i class="fa fa-check"></i><b>9.1</b> Loading data from R packages</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="downloading-r-packages-and-their-data.html"><a href="downloading-r-packages-and-their-data.html#functions-arguements"><i class="fa fa-check"></i><b>9.1.1</b> Functions &amp; Arguements</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="downloading-r-packages-and-their-data.html"><a href="downloading-r-packages-and-their-data.html#optional-what-functions-come-with-base-r"><i class="fa fa-check"></i><b>9.2</b> OPTIONAL: What functions come with base R?</a></li>
<li class="chapter" data-level="9.3" data-path="downloading-r-packages-and-their-data.html"><a href="downloading-r-packages-and-their-data.html#load-data-from-an-external-r-package"><i class="fa fa-check"></i><b>9.3</b> Load data from an external R package</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="downloading-r-packages-and-their-data.html"><a href="downloading-r-packages-and-their-data.html#step-1-downloading-packages-install.packages..."><i class="fa fa-check"></i><b>9.3.1</b> Step 1: Downloading packages <code>install.packages(...)</code></a></li>
<li class="chapter" data-level="9.3.2" data-path="downloading-r-packages-and-their-data.html"><a href="downloading-r-packages-and-their-data.html#step-2-explicitly-loading-a-package"><i class="fa fa-check"></i><b>9.3.2</b> Step 2: Explicitly loading a package</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="downloading-r-packages-and-their-data.html"><a href="downloading-r-packages-and-their-data.html#optional-seeing-all-of-your-installed-packages"><i class="fa fa-check"></i><b>9.4</b> OPTIONAL: Seeing all of your installed packages</a></li>
<li class="chapter" data-level="9.5" data-path="downloading-r-packages-and-their-data.html"><a href="downloading-r-packages-and-their-data.html#downloading-packages-using-rstudio"><i class="fa fa-check"></i><b>9.5</b> Downloading packages using RStudio</a></li>
<li class="chapter" data-level="9.6" data-path="downloading-r-packages-and-their-data.html"><a href="downloading-r-packages-and-their-data.html#your-turn"><i class="fa fa-check"></i><b>9.6</b> Your turn</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="data-in-dataframes.html"><a href="data-in-dataframes.html"><i class="fa fa-check"></i><b>10</b> Data in dataframes</a>
<ul>
<li class="chapter" data-level="10.0.1" data-path="data-in-dataframes.html"><a href="data-in-dataframes.html#looking-at-dataframes-with-view"><i class="fa fa-check"></i><b>10.0.1</b> Looking at dataframes with <code>View()</code></a></li>
<li class="chapter" data-level="10.0.2" data-path="data-in-dataframes.html"><a href="data-in-dataframes.html#looking-at-dataframes-in-the-console"><i class="fa fa-check"></i><b>10.0.2</b> Looking at dataframes in the console</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="plotting-data-in-dataframes.html"><a href="plotting-data-in-dataframes.html"><i class="fa fa-check"></i><b>11</b> Plotting data in dataframes</a>
<ul>
<li class="chapter" data-level="11.1" data-path="plotting-data-in-dataframes.html"><a href="plotting-data-in-dataframes.html#base-r-graphics"><i class="fa fa-check"></i><b>11.1</b> Base-R graphics</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="plotting-data-in-dataframes.html"><a href="plotting-data-in-dataframes.html#type-of-points"><i class="fa fa-check"></i><b>11.1.1</b> Type of points</a></li>
<li class="chapter" data-level="11.1.2" data-path="plotting-data-in-dataframes.html"><a href="plotting-data-in-dataframes.html#plotting-two-columns-of-data"><i class="fa fa-check"></i><b>11.1.2</b> Plotting two columns of data</a></li>
<li class="chapter" data-level="11.1.3" data-path="plotting-data-in-dataframes.html"><a href="plotting-data-in-dataframes.html#changing-the-range-of-a-plot-axis"><i class="fa fa-check"></i><b>11.1.3</b> Changing the range of a plot axis</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="plotting-data-in-dataframes.html"><a href="plotting-data-in-dataframes.html#you-try-it"><i class="fa fa-check"></i><b>11.2</b> You try it</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="plotting-data-in-dataframes.html"><a href="plotting-data-in-dataframes.html#fixer-uppers"><i class="fa fa-check"></i><b>11.2.1</b> Fixer-uppers</a></li>
<li class="chapter" data-level="11.2.2" data-path="plotting-data-in-dataframes.html"><a href="plotting-data-in-dataframes.html#intermediate"><i class="fa fa-check"></i><b>11.2.2</b> Intermediate</a></li>
<li class="chapter" data-level="11.2.3" data-path="plotting-data-in-dataframes.html"><a href="plotting-data-in-dataframes.html#advanced"><i class="fa fa-check"></i><b>11.2.3</b> Advanced</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="build-your-own-dataframe-part-i-vectors.html"><a href="build-your-own-dataframe-part-i-vectors.html"><i class="fa fa-check"></i><b>12</b> Build your own dataframe part I: Vectors</a>
<ul>
<li class="chapter" data-level="12.1" data-path="build-your-own-dataframe-part-i-vectors.html"><a href="build-your-own-dataframe-part-i-vectors.html#dataframes-are-a-type-of-data-structure"><i class="fa fa-check"></i><b>12.1</b> Dataframes are a type of data structure</a></li>
<li class="chapter" data-level="12.2" data-path="build-your-own-dataframe-part-i-vectors.html"><a href="build-your-own-dataframe-part-i-vectors.html#dataframes-are-made-from-vectors"><i class="fa fa-check"></i><b>12.2</b> Dataframes are made from vectors</a></li>
<li class="chapter" data-level="12.3" data-path="build-your-own-dataframe-part-i-vectors.html"><a href="build-your-own-dataframe-part-i-vectors.html#make-your-own-vectors"><i class="fa fa-check"></i><b>12.3</b> Make your own vectors</a></li>
<li class="chapter" data-level="12.4" data-path="build-your-own-dataframe-part-i-vectors.html"><a href="build-your-own-dataframe-part-i-vectors.html#checking-the-length-of-vectors"><i class="fa fa-check"></i><b>12.4</b> Checking the length of vectors</a></li>
<li class="chapter" data-level="12.5" data-path="build-your-own-dataframe-part-i-vectors.html"><a href="build-your-own-dataframe-part-i-vectors.html#another-example"><i class="fa fa-check"></i><b>12.5</b> Another example</a></li>
<li class="chapter" data-level="12.6" data-path="build-your-own-dataframe-part-i-vectors.html"><a href="build-your-own-dataframe-part-i-vectors.html#try-it-yourself"><i class="fa fa-check"></i><b>12.6</b> Try it yourself</a></li>
<li class="chapter" data-level="12.7" data-path="build-your-own-dataframe-part-i-vectors.html"><a href="build-your-own-dataframe-part-i-vectors.html#a-complicate-dataframe-to-make"><i class="fa fa-check"></i><b>12.7</b> A complicate dataframe to make</a></li>
<li class="chapter" data-level="12.8" data-path="build-your-own-dataframe-part-i-vectors.html"><a href="build-your-own-dataframe-part-i-vectors.html#build-the-dataframe"><i class="fa fa-check"></i><b>12.8</b> Build the dataframe</a></li>
<li class="chapter" data-level="12.9" data-path="build-your-own-dataframe-part-i-vectors.html"><a href="build-your-own-dataframe-part-i-vectors.html#references"><i class="fa fa-check"></i><b>12.9</b> References</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="build-your-own-dataframe.html"><a href="build-your-own-dataframe.html"><i class="fa fa-check"></i><b>13</b> Build your own dataframe</a>
<ul>
<li class="chapter" data-level="13.1" data-path="build-your-own-dataframe.html"><a href="build-your-own-dataframe.html#your-turn-1"><i class="fa fa-check"></i><b>13.1</b> Your turn</a></li>
<li class="chapter" data-level="13.2" data-path="build-your-own-dataframe.html"><a href="build-your-own-dataframe.html#references-1"><i class="fa fa-check"></i><b>13.2</b> References</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="build-your-own-dataframe-1.html"><a href="build-your-own-dataframe-1.html"><i class="fa fa-check"></i><b>14</b> Build your own dataframe</a>
<ul>
<li class="chapter" data-level="14.1" data-path="build-your-own-dataframe-1.html"><a href="build-your-own-dataframe-1.html#your-turn-2"><i class="fa fa-check"></i><b>14.1</b> Your turn</a></li>
<li class="chapter" data-level="14.2" data-path="build-your-own-dataframe-1.html"><a href="build-your-own-dataframe-1.html#references-2"><i class="fa fa-check"></i><b>14.2</b> References</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="introduction-to-ggplot-and-ggpubr.html"><a href="introduction-to-ggplot-and-ggpubr.html"><i class="fa fa-check"></i><b>15</b> Introduction to ggplot and ggpubr</a>
<ul>
<li class="chapter" data-level="15.1" data-path="introduction-to-ggplot-and-ggpubr.html"><a href="introduction-to-ggplot-and-ggpubr.html#regression-line"><i class="fa fa-check"></i><b>15.1</b> Regression line</a></li>
<li class="chapter" data-level="15.2" data-path="introduction-to-ggplot-and-ggpubr.html"><a href="introduction-to-ggplot-and-ggpubr.html#smoother"><i class="fa fa-check"></i><b>15.2</b> Smoother</a></li>
<li class="chapter" data-level="15.3" data-path="introduction-to-ggplot-and-ggpubr.html"><a href="introduction-to-ggplot-and-ggpubr.html#correlation"><i class="fa fa-check"></i><b>15.3</b> correlation</a></li>
<li class="chapter" data-level="15.4" data-path="introduction-to-ggplot-and-ggpubr.html"><a href="introduction-to-ggplot-and-ggpubr.html#long-time-series"><i class="fa fa-check"></i><b>15.4</b> Long time series</a></li>
<li class="chapter" data-level="15.5" data-path="introduction-to-ggplot-and-ggpubr.html"><a href="introduction-to-ggplot-and-ggpubr.html#regression-lines"><i class="fa fa-check"></i><b>15.5</b> Regression lines</a></li>
<li class="chapter" data-level="15.6" data-path="introduction-to-ggplot-and-ggpubr.html"><a href="introduction-to-ggplot-and-ggpubr.html#correlations"><i class="fa fa-check"></i><b>15.6</b> Correlations</a></li>
<li class="chapter" data-level="15.7" data-path="introduction-to-ggplot-and-ggpubr.html"><a href="introduction-to-ggplot-and-ggpubr.html#original-data"><i class="fa fa-check"></i><b>15.7</b> Original data</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="loading-packages-data-from-github.html"><a href="loading-packages-data-from-github.html"><i class="fa fa-check"></i><b>16</b> Loading packages &amp; data from GitHub</a>
<ul>
<li class="chapter" data-level="16.1" data-path="loading-packages-data-from-github.html"><a href="loading-packages-data-from-github.html#introduction"><i class="fa fa-check"></i><b>16.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="16.1.1" data-path="loading-packages-data-from-github.html"><a href="loading-packages-data-from-github.html#learning-objectives"><i class="fa fa-check"></i><b>16.1.1</b> Learning objectives</a></li>
<li class="chapter" data-level="16.1.2" data-path="loading-packages-data-from-github.html"><a href="loading-packages-data-from-github.html#learning-goals"><i class="fa fa-check"></i><b>16.1.2</b> Learning goals</a></li>
<li class="chapter" data-level="16.1.3" data-path="loading-packages-data-from-github.html"><a href="loading-packages-data-from-github.html#functions-arguements-1"><i class="fa fa-check"></i><b>16.1.3</b> Functions &amp; Arguements</a></li>
<li class="chapter" data-level="16.1.4" data-path="loading-packages-data-from-github.html"><a href="loading-packages-data-from-github.html#packages"><i class="fa fa-check"></i><b>16.1.4</b> Packages</a></li>
<li class="chapter" data-level="16.1.5" data-path="loading-packages-data-from-github.html"><a href="loading-packages-data-from-github.html#potential-hangups"><i class="fa fa-check"></i><b>16.1.5</b> Potential hangups</a></li>
</ul></li>
<li class="chapter" data-level="16.2" data-path="loading-packages-data-from-github.html"><a href="loading-packages-data-from-github.html#accessing-github-using-devtools"><i class="fa fa-check"></i><b>16.2</b> [ ] Accessing GitHub using devtools</a></li>
<li class="chapter" data-level="16.3" data-path="loading-packages-data-from-github.html"><a href="loading-packages-data-from-github.html#downloading-the-compbio4all-package-with-install_github"><i class="fa fa-check"></i><b>16.3</b> [ ] Downloading the compbio4all package with install_github()</a></li>
<li class="chapter" data-level="16.4" data-path="loading-packages-data-from-github.html"><a href="loading-packages-data-from-github.html#the-wildlifer-packge-webiste"><i class="fa fa-check"></i><b>16.4</b> [ ] The wildlifeR packge webiste</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="installing-from-github.html"><a href="installing-from-github.html"><i class="fa fa-check"></i><b>17</b> Installing from github</a></li>
<li class="chapter" data-level="18" data-path="packages-their-dependencies.html"><a href="packages-their-dependencies.html"><i class="fa fa-check"></i><b>18</b> Packages &amp; their dependencies</a>
<ul>
<li class="chapter" data-level="18.1" data-path="packages-their-dependencies.html"><a href="packages-their-dependencies.html#optional-make-a-plot-with-ggpubr"><i class="fa fa-check"></i><b>18.1</b> Optional: Make a plot with ggpubr</a></li>
<li class="chapter" data-level="18.2" data-path="packages-their-dependencies.html"><a href="packages-their-dependencies.html#challenge"><i class="fa fa-check"></i><b>18.2</b> Challenge</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="load-data-from-internet.html"><a href="load-data-from-internet.html"><i class="fa fa-check"></i><b>19</b> Load data from internet</a>
<ul>
<li class="chapter" data-level="19.1" data-path="load-data-from-internet.html"><a href="load-data-from-internet.html#base-r-plot"><i class="fa fa-check"></i><b>19.1</b> Base R plot</a></li>
<li class="chapter" data-level="19.2" data-path="load-data-from-internet.html"><a href="load-data-from-internet.html#ggpubr---have-to-stack-it"><i class="fa fa-check"></i><b>19.2</b> ggpubr - have to stack it</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="assignment-building-dataframes-by-hand.html"><a href="assignment-building-dataframes-by-hand.html"><i class="fa fa-check"></i><b>20</b> Assignment: Building dataframes by hand</a>
<ul>
<li class="chapter" data-level="20.1" data-path="assignment-building-dataframes-by-hand.html"><a href="assignment-building-dataframes-by-hand.html#introduction-1"><i class="fa fa-check"></i><b>20.1</b> Introduction</a></li>
<li class="chapter" data-level="20.2" data-path="assignment-building-dataframes-by-hand.html"><a href="assignment-building-dataframes-by-hand.html#to-do"><i class="fa fa-check"></i><b>20.2</b> To do</a></li>
<li class="chapter" data-level="20.3" data-path="assignment-building-dataframes-by-hand.html"><a href="assignment-building-dataframes-by-hand.html#preliminaries"><i class="fa fa-check"></i><b>20.3</b> Preliminaries</a>
<ul>
<li class="chapter" data-level="20.3.1" data-path="assignment-building-dataframes-by-hand.html"><a href="assignment-building-dataframes-by-hand.html#non-r-materials"><i class="fa fa-check"></i><b>20.3.1</b> Non-R materials</a></li>
<li class="chapter" data-level="20.3.2" data-path="assignment-building-dataframes-by-hand.html"><a href="assignment-building-dataframes-by-hand.html#packages-1"><i class="fa fa-check"></i><b>20.3.2</b> Packages</a></li>
<li class="chapter" data-level="20.3.3" data-path="assignment-building-dataframes-by-hand.html"><a href="assignment-building-dataframes-by-hand.html#ignore-this"><i class="fa fa-check"></i><b>20.3.3</b> Ignore this</a></li>
</ul></li>
<li class="chapter" data-level="20.4" data-path="assignment-building-dataframes-by-hand.html"><a href="assignment-building-dataframes-by-hand.html#the-data"><i class="fa fa-check"></i><b>20.4</b> The data</a></li>
<li class="chapter" data-level="20.5" data-path="assignment-building-dataframes-by-hand.html"><a href="assignment-building-dataframes-by-hand.html#creating-vectors-in-r"><i class="fa fa-check"></i><b>20.5</b> Creating vectors in R</a></li>
<li class="chapter" data-level="20.6" data-path="assignment-building-dataframes-by-hand.html"><a href="assignment-building-dataframes-by-hand.html#checking-the-length-of-vectors-1"><i class="fa fa-check"></i><b>20.6</b> Checking the length of vectors</a></li>
<li class="chapter" data-level="20.7" data-path="assignment-building-dataframes-by-hand.html"><a href="assignment-building-dataframes-by-hand.html#exponents-and-scientific-notation"><i class="fa fa-check"></i><b>20.7</b> Exponents and scientific notation</a></li>
<li class="chapter" data-level="20.8" data-path="assignment-building-dataframes-by-hand.html"><a href="assignment-building-dataframes-by-hand.html#some-more-vectors"><i class="fa fa-check"></i><b>20.8</b> Some more vectors</a></li>
<li class="chapter" data-level="20.9" data-path="assignment-building-dataframes-by-hand.html"><a href="assignment-building-dataframes-by-hand.html#make-a-dataframe"><i class="fa fa-check"></i><b>20.9</b> Make a dataframe</a></li>
<li class="chapter" data-level="20.10" data-path="assignment-building-dataframes-by-hand.html"><a href="assignment-building-dataframes-by-hand.html#make-figure-1-with-base-r"><i class="fa fa-check"></i><b>20.10</b> Make Figure 1 with base R</a></li>
<li class="chapter" data-level="20.11" data-path="assignment-building-dataframes-by-hand.html"><a href="assignment-building-dataframes-by-hand.html#make-figure-1-with-ggpubr"><i class="fa fa-check"></i><b>20.11</b> Make Figure 1 with ggpubr</a></li>
<li class="chapter" data-level="20.12" data-path="assignment-building-dataframes-by-hand.html"><a href="assignment-building-dataframes-by-hand.html#add-statistics"><i class="fa fa-check"></i><b>20.12</b> Add statistics</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="dataframe-by-hand-problem-set-key.html"><a href="dataframe-by-hand-problem-set-key.html"><i class="fa fa-check"></i><b>21</b> Dataframe by hand - problem set - KEY</a>
<ul>
<li class="chapter" data-level="21.1" data-path="dataframe-by-hand-problem-set-key.html"><a href="dataframe-by-hand-problem-set-key.html#preliminaries-1"><i class="fa fa-check"></i><b>21.1</b> Preliminaries</a></li>
<li class="chapter" data-level="21.2" data-path="dataframe-by-hand-problem-set-key.html"><a href="dataframe-by-hand-problem-set-key.html#step-one-make-each-column"><i class="fa fa-check"></i><b>21.2</b> Step one: Make each column</a>
<ul>
<li class="chapter" data-level="21.2.1" data-path="dataframe-by-hand-problem-set-key.html"><a href="dataframe-by-hand-problem-set-key.html#quality-control"><i class="fa fa-check"></i><b>21.2.1</b> Quality control</a></li>
</ul></li>
<li class="chapter" data-level="21.3" data-path="dataframe-by-hand-problem-set-key.html"><a href="dataframe-by-hand-problem-set-key.html#basic-data-exploration"><i class="fa fa-check"></i><b>21.3</b> Basic Data exploration</a></li>
<li class="chapter" data-level="21.4" data-path="dataframe-by-hand-problem-set-key.html"><a href="dataframe-by-hand-problem-set-key.html#part-2-build-the-dataframe"><i class="fa fa-check"></i><b>21.4</b> Part 2: Build the dataframe</a></li>
<li class="chapter" data-level="21.5" data-path="dataframe-by-hand-problem-set-key.html"><a href="dataframe-by-hand-problem-set-key.html#part-3-make-figure"><i class="fa fa-check"></i><b>21.5</b> Part 3: Make figure</a></li>
<li class="chapter" data-level="21.6" data-path="dataframe-by-hand-problem-set-key.html"><a href="dataframe-by-hand-problem-set-key.html#graph-of-length.bp"><i class="fa fa-check"></i><b>21.6</b> Graph of length.bp</a></li>
<li class="chapter" data-level="21.7" data-path="dataframe-by-hand-problem-set-key.html"><a href="dataframe-by-hand-problem-set-key.html#create-new-variable-t.mlength.bp"><i class="fa fa-check"></i><b>21.7</b> Create new variable: T.m/length.bp</a>
<ul>
<li class="chapter" data-level="21.7.1" data-path="dataframe-by-hand-problem-set-key.html"><a href="dataframe-by-hand-problem-set-key.html#short-answer-question-what-am-i-having-you-do-this-2-4-sentences"><i class="fa fa-check"></i><b>21.7.1</b> Short answer question: What am I having you do this? (2-4 sentences)</a></li>
<li class="chapter" data-level="21.7.2" data-path="dataframe-by-hand-problem-set-key.html"><a href="dataframe-by-hand-problem-set-key.html#short-answer-question-what-am-i-having-you-do-this-2-4-sentences-1"><i class="fa fa-check"></i><b>21.7.2</b> Short answer question: What am I having you do this? (2-4 sentences)</a></li>
</ul></li>
<li class="chapter" data-level="21.8" data-path="dataframe-by-hand-problem-set-key.html"><a href="dataframe-by-hand-problem-set-key.html#new-plot-t.m801-versus-p.gc"><i class="fa fa-check"></i><b>21.8</b> New plot: T.m801 versus P.GC</a>
<ul>
<li class="chapter" data-level="21.8.1" data-path="dataframe-by-hand-problem-set-key.html"><a href="dataframe-by-hand-problem-set-key.html#correct-axes"><i class="fa fa-check"></i><b>21.8.1</b> Correct axes</a></li>
<li class="chapter" data-level="21.8.2" data-path="dataframe-by-hand-problem-set-key.html"><a href="dataframe-by-hand-problem-set-key.html#flipped-axes"><i class="fa fa-check"></i><b>21.8.2</b> Flipped axes</a></li>
<li class="chapter" data-level="21.8.3" data-path="dataframe-by-hand-problem-set-key.html"><a href="dataframe-by-hand-problem-set-key.html#short-answer-question-what-has-changed-with-this-new-plot-why-3-5-sentences"><i class="fa fa-check"></i><b>21.8.3</b> Short answer question: What has changed with this new plot? Why? (3-5 sentences)</a></li>
</ul></li>
<li class="chapter" data-level="21.9" data-path="dataframe-by-hand-problem-set-key.html"><a href="dataframe-by-hand-problem-set-key.html#grading-function"><i class="fa fa-check"></i><b>21.9</b> Grading function</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="accession-number-leave-this-code-as-is.html"><a href="accession-number-leave-this-code-as-is.html"><i class="fa fa-check"></i><b>22</b> Accession number; leave this code as is</a></li>
<li class="chapter" data-level="23" data-path="length-of-16srna-gene-in-bp.html"><a href="length-of-16srna-gene-in-bp.html"><i class="fa fa-check"></i><b>23</b> Length of 16sRNA gene in BP</a></li>
<li class="chapter" data-level="24" data-path="percent-gc.html"><a href="percent-gc.html"><i class="fa fa-check"></i><b>24</b> Percent GC</a></li>
<li class="chapter" data-level="25" data-path="melting-temperature.html"><a href="melting-temperature.html"><i class="fa fa-check"></i><b>25</b> Melting temperature</a></li>
<li class="chapter" data-level="26" data-path="references-leave-this-as-is.html"><a href="references-leave-this-as-is.html"><i class="fa fa-check"></i><b>26</b> References - leave this as is</a></li>
<li class="chapter" data-level="27" data-path="dataframe-by-hand-problem-set.html"><a href="dataframe-by-hand-problem-set.html"><i class="fa fa-check"></i><b>27</b> Dataframe by hand - problem set</a>
<ul>
<li class="chapter" data-level="27.1" data-path="dataframe-by-hand-problem-set.html"><a href="dataframe-by-hand-problem-set.html#preliminaries-2"><i class="fa fa-check"></i><b>27.1</b> Preliminaries</a></li>
<li class="chapter" data-level="27.2" data-path="dataframe-by-hand-problem-set.html"><a href="dataframe-by-hand-problem-set.html#step-one-make-each-column-1"><i class="fa fa-check"></i><b>27.2</b> Step one: Make each column</a>
<ul>
<li class="chapter" data-level="27.2.1" data-path="dataframe-by-hand-problem-set.html"><a href="dataframe-by-hand-problem-set.html#quality-control-1"><i class="fa fa-check"></i><b>27.2.1</b> Quality control</a></li>
</ul></li>
<li class="chapter" data-level="27.3" data-path="dataframe-by-hand-problem-set.html"><a href="dataframe-by-hand-problem-set.html#basic-data-exploration-1"><i class="fa fa-check"></i><b>27.3</b> Basic Data exploration</a></li>
<li class="chapter" data-level="27.4" data-path="dataframe-by-hand-problem-set.html"><a href="dataframe-by-hand-problem-set.html#part-2-build-the-dataframe-1"><i class="fa fa-check"></i><b>27.4</b> Part 2: Build the dataframe</a></li>
<li class="chapter" data-level="27.5" data-path="dataframe-by-hand-problem-set.html"><a href="dataframe-by-hand-problem-set.html#part-3-make-figure-1"><i class="fa fa-check"></i><b>27.5</b> Part 3: Make figure</a></li>
<li class="chapter" data-level="27.6" data-path="dataframe-by-hand-problem-set.html"><a href="dataframe-by-hand-problem-set.html#graph-of-length.bp-vs-t.m"><i class="fa fa-check"></i><b>27.6</b> Graph of length.bp vs T.m</a></li>
<li class="chapter" data-level="27.7" data-path="dataframe-by-hand-problem-set.html"><a href="dataframe-by-hand-problem-set.html#create-new-variable-t.mlength.bp-1"><i class="fa fa-check"></i><b>27.7</b> Create new variable: T.m/length.bp</a>
<ul>
<li class="chapter" data-level="27.7.1" data-path="dataframe-by-hand-problem-set.html"><a href="dataframe-by-hand-problem-set.html#short-answer-question-why-am-i-having-you-do-this-2-4-sentences"><i class="fa fa-check"></i><b>27.7.1</b> Short answer question: Why am I having you do this? (2-4 sentences)</a></li>
<li class="chapter" data-level="27.7.2" data-path="dataframe-by-hand-problem-set.html"><a href="dataframe-by-hand-problem-set.html#short-answer-question-why-am-i-having-you-do-this-2-4-sentences-1"><i class="fa fa-check"></i><b>27.7.2</b> Short answer question: Why am I having you do this? (2-4 sentences)</a></li>
</ul></li>
<li class="chapter" data-level="27.8" data-path="dataframe-by-hand-problem-set.html"><a href="dataframe-by-hand-problem-set.html#new-plot-t.m801-versus-p.gc-1"><i class="fa fa-check"></i><b>27.8</b> New plot: T.m801 versus P.GC</a>
<ul>
<li class="chapter" data-level="27.8.1" data-path="dataframe-by-hand-problem-set.html"><a href="dataframe-by-hand-problem-set.html#short-answer-question-what-has-changed-with-this-new-plot-why-3-5-sentences-1"><i class="fa fa-check"></i><b>27.8.1</b> Short answer question: What has changed with this new plot? Why? (3-5 sentences)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="28" data-path="making-a-simple-dataframe.html"><a href="making-a-simple-dataframe.html"><i class="fa fa-check"></i><b>28</b> Making a simple dataframe</a>
<ul>
<li class="chapter" data-level="28.1" data-path="making-a-simple-dataframe.html"><a href="making-a-simple-dataframe.html#working-with-data-in-2-seperate-vectors"><i class="fa fa-check"></i><b>28.1</b> Working with data in 2 seperate vectors</a>
<ul>
<li class="chapter" data-level="28.1.1" data-path="making-a-simple-dataframe.html"><a href="making-a-simple-dataframe.html#summary-statistics"><i class="fa fa-check"></i><b>28.1.1</b> Summary Statistics</a></li>
<li class="chapter" data-level="28.1.2" data-path="making-a-simple-dataframe.html"><a href="making-a-simple-dataframe.html#making-a-dataframe"><i class="fa fa-check"></i><b>28.1.2</b> Making a dataframe</a></li>
<li class="chapter" data-level="28.1.3" data-path="making-a-simple-dataframe.html"><a href="making-a-simple-dataframe.html#making-a-boxplot"><i class="fa fa-check"></i><b>28.1.3</b> Making a boxplot</a></li>
<li class="chapter" data-level="28.1.4" data-path="making-a-simple-dataframe.html"><a href="making-a-simple-dataframe.html#making-a-histogram"><i class="fa fa-check"></i><b>28.1.4</b> Making a histogram</a></li>
<li class="chapter" data-level="28.1.5" data-path="making-a-simple-dataframe.html"><a href="making-a-simple-dataframe.html#making-a-density-plot"><i class="fa fa-check"></i><b>28.1.5</b> Making a density plot</a></li>
<li class="chapter" data-level="28.1.6" data-path="making-a-simple-dataframe.html"><a href="making-a-simple-dataframe.html#advanced-1"><i class="fa fa-check"></i><b>28.1.6</b> Advanced</a></li>
<li class="chapter" data-level="28.1.7" data-path="making-a-simple-dataframe.html"><a href="making-a-simple-dataframe.html#historical-reference"><i class="fa fa-check"></i><b>28.1.7</b> Historical reference</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="29" data-path="load-data-from-internet-1.html"><a href="load-data-from-internet-1.html"><i class="fa fa-check"></i><b>29</b> Load data from internet</a>
<ul>
<li class="chapter" data-level="29.1" data-path="load-data-from-internet-1.html"><a href="load-data-from-internet-1.html#introduction-2"><i class="fa fa-check"></i><b>29.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="29.1.1" data-path="load-data-from-internet-1.html"><a href="load-data-from-internet-1.html#learning-goals-outcomes"><i class="fa fa-check"></i><b>29.1.1</b> Learning Goals &amp; Outcomes</a></li>
<li class="chapter" data-level="29.1.2" data-path="load-data-from-internet-1.html"><a href="load-data-from-internet-1.html#functions-arguements-2"><i class="fa fa-check"></i><b>29.1.2</b> Functions &amp; Arguements</a></li>
<li class="chapter" data-level="29.1.3" data-path="load-data-from-internet-1.html"><a href="load-data-from-internet-1.html#packages-2"><i class="fa fa-check"></i><b>29.1.3</b> Packages</a></li>
<li class="chapter" data-level="29.1.4" data-path="load-data-from-internet-1.html"><a href="load-data-from-internet-1.html#potential-hangups-1"><i class="fa fa-check"></i><b>29.1.4</b> Potential hangups</a></li>
</ul></li>
<li class="chapter" data-level="29.2" data-path="load-data-from-internet-1.html"><a href="load-data-from-internet-1.html#downloading-a-.csv-file-using-geturl"><i class="fa fa-check"></i><b>29.2</b> [ ] Downloading a .csv file using getURL()</a></li>
<li class="chapter" data-level="29.3" data-path="load-data-from-internet-1.html"><a href="load-data-from-internet-1.html#optional-plotting-west-virginia-eagle-data"><i class="fa fa-check"></i><b>29.3</b> OPTIONAL: Plotting West Virginia Eagle Data</a></li>
</ul></li>
<li class="chapter" data-level="30" data-path="worked-example-data-visualization-in-r.html"><a href="worked-example-data-visualization-in-r.html"><i class="fa fa-check"></i><b>30</b> Worked example: Data visualization in R</a>
<ul>
<li class="chapter" data-level="30.1" data-path="worked-example-data-visualization-in-r.html"><a href="worked-example-data-visualization-in-r.html#preliminaries-3"><i class="fa fa-check"></i><b>30.1</b> Preliminaries</a>
<ul>
<li class="chapter" data-level="30.1.1" data-path="worked-example-data-visualization-in-r.html"><a href="worked-example-data-visualization-in-r.html#packages-3"><i class="fa fa-check"></i><b>30.1.1</b> Packages</a></li>
<li class="chapter" data-level="30.1.2" data-path="worked-example-data-visualization-in-r.html"><a href="worked-example-data-visualization-in-r.html#data"><i class="fa fa-check"></i><b>30.1.2</b> Data</a></li>
</ul></li>
<li class="chapter" data-level="30.2" data-path="worked-example-data-visualization-in-r.html"><a href="worked-example-data-visualization-in-r.html#data-exploration-getting-to-know-data"><i class="fa fa-check"></i><b>30.2</b> Data exploration: Getting to know data</a></li>
<li class="chapter" data-level="30.3" data-path="worked-example-data-visualization-in-r.html"><a href="worked-example-data-visualization-in-r.html#numeric-summaries-data"><i class="fa fa-check"></i><b>30.3</b> Numeric summaries data</a></li>
<li class="chapter" data-level="30.4" data-path="worked-example-data-visualization-in-r.html"><a href="worked-example-data-visualization-in-r.html#plotting-data"><i class="fa fa-check"></i><b>30.4</b> Plotting data</a></li>
<li class="chapter" data-level="30.5" data-path="worked-example-data-visualization-in-r.html"><a href="worked-example-data-visualization-in-r.html#adding-color-to-a-plot"><i class="fa fa-check"></i><b>30.5</b> Adding color to a plot</a></li>
<li class="chapter" data-level="30.6" data-path="worked-example-data-visualization-in-r.html"><a href="worked-example-data-visualization-in-r.html#changing-symbols"><i class="fa fa-check"></i><b>30.6</b> Changing symbols</a></li>
<li class="chapter" data-level="30.7" data-path="worked-example-data-visualization-in-r.html"><a href="worked-example-data-visualization-in-r.html#adding-more-stuff-to-a-plot"><i class="fa fa-check"></i><b>30.7</b> Adding more stuff to a plot</a></li>
<li class="chapter" data-level="30.8" data-path="worked-example-data-visualization-in-r.html"><a href="worked-example-data-visualization-in-r.html#boxplots"><i class="fa fa-check"></i><b>30.8</b> Boxplots</a></li>
<li class="chapter" data-level="30.9" data-path="worked-example-data-visualization-in-r.html"><a href="worked-example-data-visualization-in-r.html#multivariate-groups"><i class="fa fa-check"></i><b>30.9</b> Multivariate groups</a></li>
<li class="chapter" data-level="30.10" data-path="worked-example-data-visualization-in-r.html"><a href="worked-example-data-visualization-in-r.html#notes-for-further-study"><i class="fa fa-check"></i><b>30.10</b> Notes for Further study</a>
<ul>
<li class="chapter" data-level="30.10.1" data-path="worked-example-data-visualization-in-r.html"><a href="worked-example-data-visualization-in-r.html#upgma"><i class="fa fa-check"></i><b>30.10.1</b> UPGMA</a></li>
<li class="chapter" data-level="30.10.2" data-path="worked-example-data-visualization-in-r.html"><a href="worked-example-data-visualization-in-r.html#neighbor-joining"><i class="fa fa-check"></i><b>30.10.2</b> Neighbor joining</a></li>
<li class="chapter" data-level="30.10.3" data-path="worked-example-data-visualization-in-r.html"><a href="worked-example-data-visualization-in-r.html#wards-method"><i class="fa fa-check"></i><b>30.10.3</b> Ward’s method</a></li>
<li class="chapter" data-level="30.10.4" data-path="worked-example-data-visualization-in-r.html"><a href="worked-example-data-visualization-in-r.html#hclust-algorithms"><i class="fa fa-check"></i><b>30.10.4</b> hclust algorithms</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="31" data-path="ggpubr.html"><a href="ggpubr.html"><i class="fa fa-check"></i><b>31</b> ggpubr</a>
<ul>
<li class="chapter" data-level="31.1" data-path="ggpubr.html"><a href="ggpubr.html#vocab"><i class="fa fa-check"></i><b>31.1</b> Vocab</a></li>
<li class="chapter" data-level="31.2" data-path="ggpubr.html"><a href="ggpubr.html#learning-objectives-1"><i class="fa fa-check"></i><b>31.2</b> Learning objectives</a></li>
<li class="chapter" data-level="31.3" data-path="ggpubr.html"><a href="ggpubr.html#introduction-3"><i class="fa fa-check"></i><b>31.3</b> Introduction</a></li>
<li class="chapter" data-level="31.4" data-path="ggpubr.html"><a href="ggpubr.html#preliminaries-4"><i class="fa fa-check"></i><b>31.4</b> Preliminaries</a>
<ul>
<li class="chapter" data-level="31.4.1" data-path="ggpubr.html"><a href="ggpubr.html#download-packaged"><i class="fa fa-check"></i><b>31.4.1</b> Download packaged</a></li>
<li class="chapter" data-level="31.4.2" data-path="ggpubr.html"><a href="ggpubr.html#load-the-libraries"><i class="fa fa-check"></i><b>31.4.2</b> Load the libraries</a></li>
<li class="chapter" data-level="31.4.3" data-path="ggpubr.html"><a href="ggpubr.html#load-the-msleep-package"><i class="fa fa-check"></i><b>31.4.3</b> Load the msleep package</a></li>
</ul></li>
<li class="chapter" data-level="31.5" data-path="ggpubr.html"><a href="ggpubr.html#make-a-basic-ggpubr-plot"><i class="fa fa-check"></i><b>31.5</b> Make a basic ggpubr plot</a>
<ul>
<li class="chapter" data-level="31.5.1" data-path="ggpubr.html"><a href="ggpubr.html#ggpubr-syntax"><i class="fa fa-check"></i><b>31.5.1</b> ggpubr syntax</a></li>
<li class="chapter" data-level="31.5.2" data-path="ggpubr.html"><a href="ggpubr.html#color-coding-data"><i class="fa fa-check"></i><b>31.5.2</b> Color-coding data</a></li>
</ul></li>
<li class="chapter" data-level="31.6" data-path="ggpubr.html"><a href="ggpubr.html#scatter-plots-in-ggpubr"><i class="fa fa-check"></i><b>31.6</b> Scatter plots in ggpubr</a></li>
<li class="chapter" data-level="31.7" data-path="ggpubr.html"><a href="ggpubr.html#coloring-by-a-categorical-variable"><i class="fa fa-check"></i><b>31.7</b> Coloring by a categorical variable</a></li>
<li class="chapter" data-level="31.8" data-path="ggpubr.html"><a href="ggpubr.html#coloring-by-a-continuous-numeric-variable"><i class="fa fa-check"></i><b>31.8</b> Coloring by a continuous numeric variable</a></li>
<li class="chapter" data-level="31.9" data-path="ggpubr.html"><a href="ggpubr.html#adding-a-line-of-best-fit"><i class="fa fa-check"></i><b>31.9</b> Adding a line of best fit</a></li>
<li class="chapter" data-level="31.10" data-path="ggpubr.html"><a href="ggpubr.html#adding-a-data-ellipse"><i class="fa fa-check"></i><b>31.10</b> Adding a data ellipse</a></li>
<li class="chapter" data-level="31.11" data-path="ggpubr.html"><a href="ggpubr.html#add-a-correlation-coefficient"><i class="fa fa-check"></i><b>31.11</b> Add a correlation coefficient</a></li>
<li class="chapter" data-level="31.12" data-path="ggpubr.html"><a href="ggpubr.html#task"><i class="fa fa-check"></i><b>31.12</b> TASK</a></li>
</ul></li>
<li class="chapter" data-level="32" data-path="gpubr-allometric-data.html"><a href="gpubr-allometric-data.html"><i class="fa fa-check"></i><b>32</b> gpubr - allometric data</a>
<ul>
<li class="chapter" data-level="32.1" data-path="gpubr-allometric-data.html"><a href="gpubr-allometric-data.html#vocab-1"><i class="fa fa-check"></i><b>32.1</b> Vocab</a></li>
<li class="chapter" data-level="32.2" data-path="gpubr-allometric-data.html"><a href="gpubr-allometric-data.html#learning-objectives-2"><i class="fa fa-check"></i><b>32.2</b> Learning objectives</a></li>
<li class="chapter" data-level="32.3" data-path="gpubr-allometric-data.html"><a href="gpubr-allometric-data.html#introduction-4"><i class="fa fa-check"></i><b>32.3</b> Introduction</a></li>
<li class="chapter" data-level="32.4" data-path="gpubr-allometric-data.html"><a href="gpubr-allometric-data.html#preliminaries-5"><i class="fa fa-check"></i><b>32.4</b> Preliminaries</a>
<ul>
<li class="chapter" data-level="32.4.1" data-path="gpubr-allometric-data.html"><a href="gpubr-allometric-data.html#download-packaged-1"><i class="fa fa-check"></i><b>32.4.1</b> Download packaged</a></li>
<li class="chapter" data-level="32.4.2" data-path="gpubr-allometric-data.html"><a href="gpubr-allometric-data.html#load-the-libraries-1"><i class="fa fa-check"></i><b>32.4.2</b> Load the libraries</a></li>
<li class="chapter" data-level="32.4.3" data-path="gpubr-allometric-data.html"><a href="gpubr-allometric-data.html#load-the-msleep-package-1"><i class="fa fa-check"></i><b>32.4.3</b> Load the msleep package</a></li>
</ul></li>
<li class="chapter" data-level="32.5" data-path="gpubr-allometric-data.html"><a href="gpubr-allometric-data.html#make-a-basic-ggpubr-scatterplot"><i class="fa fa-check"></i><b>32.5</b> Make a basic ggpubr scatterplot</a>
<ul>
<li class="chapter" data-level="32.5.1" data-path="gpubr-allometric-data.html"><a href="gpubr-allometric-data.html#ggpubr-syntax-1"><i class="fa fa-check"></i><b>32.5.1</b> ggpubr syntax</a></li>
</ul></li>
<li class="chapter" data-level="32.6" data-path="gpubr-allometric-data.html"><a href="gpubr-allometric-data.html#scatter-plots-in-ggpubr-1"><i class="fa fa-check"></i><b>32.6</b> Scatter plots in ggpubr</a></li>
<li class="chapter" data-level="32.7" data-path="gpubr-allometric-data.html"><a href="gpubr-allometric-data.html#adding-a-line-of-best-fit-1"><i class="fa fa-check"></i><b>32.7</b> Adding a line of best fit</a></li>
<li class="chapter" data-level="32.8" data-path="gpubr-allometric-data.html"><a href="gpubr-allometric-data.html#adding-a-data-ellipse-1"><i class="fa fa-check"></i><b>32.8</b> Adding a data ellipse</a></li>
<li class="chapter" data-level="32.9" data-path="gpubr-allometric-data.html"><a href="gpubr-allometric-data.html#add-a-correlation-coefficient-1"><i class="fa fa-check"></i><b>32.9</b> Add a correlation coefficient</a>
<ul>
<li class="chapter" data-level="32.9.1" data-path="gpubr-allometric-data.html"><a href="gpubr-allometric-data.html#allometry"><i class="fa fa-check"></i><b>32.9.1</b> Allometry</a></li>
<li class="chapter" data-level="32.9.2" data-path="gpubr-allometric-data.html"><a href="gpubr-allometric-data.html#allometry-and-log-scales"><i class="fa fa-check"></i><b>32.9.2</b> Allometry and log scales</a></li>
</ul></li>
<li class="chapter" data-level="32.10" data-path="gpubr-allometric-data.html"><a href="gpubr-allometric-data.html#make-an-allometric-plot"><i class="fa fa-check"></i><b>32.10</b> Make an allometric plot</a></li>
<li class="chapter" data-level="32.11" data-path="gpubr-allometric-data.html"><a href="gpubr-allometric-data.html#change-color-by-a-categorical-variable"><i class="fa fa-check"></i><b>32.11</b> Change color by a categorical variable</a></li>
<li class="chapter" data-level="32.12" data-path="gpubr-allometric-data.html"><a href="gpubr-allometric-data.html#change-color-by-a-continuous-variable"><i class="fa fa-check"></i><b>32.12</b> Change color by a continuous variable</a></li>
<li class="chapter" data-level="32.13" data-path="gpubr-allometric-data.html"><a href="gpubr-allometric-data.html#add-a-smoother"><i class="fa fa-check"></i><b>32.13</b> Add a smoother</a></li>
<li class="chapter" data-level="32.14" data-path="gpubr-allometric-data.html"><a href="gpubr-allometric-data.html#task-1"><i class="fa fa-check"></i><b>32.14</b> Task</a></li>
</ul></li>
<li class="chapter" data-level="33" data-path="improving-plots.html"><a href="improving-plots.html"><i class="fa fa-check"></i><b>33</b> Improving plots</a>
<ul>
<li class="chapter" data-level="33.1" data-path="improving-plots.html"><a href="improving-plots.html#vocab-2"><i class="fa fa-check"></i><b>33.1</b> Vocab</a></li>
<li class="chapter" data-level="33.2" data-path="improving-plots.html"><a href="improving-plots.html#learning-objectives-3"><i class="fa fa-check"></i><b>33.2</b> Learning objectives</a></li>
<li class="chapter" data-level="33.3" data-path="improving-plots.html"><a href="improving-plots.html#introduction-5"><i class="fa fa-check"></i><b>33.3</b> Introduction</a></li>
<li class="chapter" data-level="33.4" data-path="improving-plots.html"><a href="improving-plots.html#preliminaries-6"><i class="fa fa-check"></i><b>33.4</b> Preliminaries</a>
<ul>
<li class="chapter" data-level="33.4.1" data-path="improving-plots.html"><a href="improving-plots.html#download-packaged-2"><i class="fa fa-check"></i><b>33.4.1</b> Download packaged</a></li>
<li class="chapter" data-level="33.4.2" data-path="improving-plots.html"><a href="improving-plots.html#load-the-libraries-2"><i class="fa fa-check"></i><b>33.4.2</b> Load the libraries</a></li>
<li class="chapter" data-level="33.4.3" data-path="improving-plots.html"><a href="improving-plots.html#load-the-msleep-package-2"><i class="fa fa-check"></i><b>33.4.3</b> Load the msleep package</a></li>
</ul></li>
<li class="chapter" data-level="33.5" data-path="improving-plots.html"><a href="improving-plots.html#make-a-basic-ggpubr-plot-1"><i class="fa fa-check"></i><b>33.5</b> Make a basic ggpubr plot</a>
<ul>
<li class="chapter" data-level="33.5.1" data-path="improving-plots.html"><a href="improving-plots.html#ggpubr-syntax-2"><i class="fa fa-check"></i><b>33.5.1</b> ggpubr syntax</a></li>
<li class="chapter" data-level="33.5.2" data-path="improving-plots.html"><a href="improving-plots.html#color-coding-data-1"><i class="fa fa-check"></i><b>33.5.2</b> Color-coding data</a></li>
</ul></li>
<li class="chapter" data-level="33.6" data-path="improving-plots.html"><a href="improving-plots.html#scatter-plots-in-ggpubr-2"><i class="fa fa-check"></i><b>33.6</b> Scatter plots in ggpubr</a></li>
<li class="chapter" data-level="33.7" data-path="improving-plots.html"><a href="improving-plots.html#coloring-by-a-categorical-variable-1"><i class="fa fa-check"></i><b>33.7</b> Coloring by a categorical variable</a></li>
<li class="chapter" data-level="33.8" data-path="improving-plots.html"><a href="improving-plots.html#coloring-by-a-continuous-numeric-variable-1"><i class="fa fa-check"></i><b>33.8</b> Coloring by a continuous numeric variable</a></li>
<li class="chapter" data-level="33.9" data-path="improving-plots.html"><a href="improving-plots.html#adding-a-line-of-best-fit-2"><i class="fa fa-check"></i><b>33.9</b> Adding a line of best fit</a></li>
<li class="chapter" data-level="33.10" data-path="improving-plots.html"><a href="improving-plots.html#adding-a-data-ellipse-2"><i class="fa fa-check"></i><b>33.10</b> Adding a data ellipse</a></li>
<li class="chapter" data-level="33.11" data-path="improving-plots.html"><a href="improving-plots.html#add-a-correlation-coefficient-2"><i class="fa fa-check"></i><b>33.11</b> Add a correlation coefficient</a></li>
<li class="chapter" data-level="33.12" data-path="improving-plots.html"><a href="improving-plots.html#task-2"><i class="fa fa-check"></i><b>33.12</b> TASK</a>
<ul>
<li class="chapter" data-level="33.12.1" data-path="improving-plots.html"><a href="improving-plots.html#key"><i class="fa fa-check"></i><b>33.12.1</b> Key</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="34" data-path="scatter-plots.html"><a href="scatter-plots.html"><i class="fa fa-check"></i><b>34</b> Scatter plots</a>
<ul>
<li class="chapter" data-level="34.1" data-path="scatter-plots.html"><a href="scatter-plots.html#checkpoint-plot-the-data"><i class="fa fa-check"></i><b>34.1</b> Checkpoint: Plot the data</a></li>
</ul></li>
<li class="chapter" data-level="35" data-path="d-scatterplots.html"><a href="d-scatterplots.html"><i class="fa fa-check"></i><b>35</b> 3D scatterplots</a>
<ul>
<li class="chapter" data-level="35.1" data-path="d-scatterplots.html"><a href="d-scatterplots.html#key-conepts-vocab"><i class="fa fa-check"></i><b>35.1</b> Key conepts / vocab</a></li>
<li class="chapter" data-level="35.2" data-path="d-scatterplots.html"><a href="d-scatterplots.html#preliminaries-7"><i class="fa fa-check"></i><b>35.2</b> Preliminaries</a>
<ul>
<li class="chapter" data-level="35.2.1" data-path="d-scatterplots.html"><a href="d-scatterplots.html#download-the-package"><i class="fa fa-check"></i><b>35.2.1</b> Download the package</a></li>
<li class="chapter" data-level="35.2.2" data-path="d-scatterplots.html"><a href="d-scatterplots.html#load-packages"><i class="fa fa-check"></i><b>35.2.2</b> Load packages</a></li>
<li class="chapter" data-level="35.2.3" data-path="d-scatterplots.html"><a href="d-scatterplots.html#load-data"><i class="fa fa-check"></i><b>35.2.3</b> Load data</a></li>
</ul></li>
<li class="chapter" data-level="35.3" data-path="d-scatterplots.html"><a href="d-scatterplots.html#d-scatterplots-1"><i class="fa fa-check"></i><b>35.3</b> 3D scatterplots</a>
<ul>
<li class="chapter" data-level="35.3.1" data-path="d-scatterplots.html"><a href="d-scatterplots.html#allometry-and-log-scales-1"><i class="fa fa-check"></i><b>35.3.1</b> Allometry and log scales</a></li>
<li class="chapter" data-level="35.3.2" data-path="d-scatterplots.html"><a href="d-scatterplots.html#basic-3d-scatterplot"><i class="fa fa-check"></i><b>35.3.2</b> Basic 3D scatterplot</a></li>
</ul></li>
<li class="chapter" data-level="35.4" data-path="d-scatterplots.html"><a href="d-scatterplots.html#plane-of-best-fit"><i class="fa fa-check"></i><b>35.4</b> Plane of best fit</a></li>
<li class="chapter" data-level="35.5" data-path="d-scatterplots.html"><a href="d-scatterplots.html#from-3d-plots-to-pca"><i class="fa fa-check"></i><b>35.5</b> From 3D plots to PCA</a></li>
<li class="chapter" data-level="35.6" data-path="d-scatterplots.html"><a href="d-scatterplots.html#task-3"><i class="fa fa-check"></i><b>35.6</b> Task</a></li>
</ul></li>
<li class="chapter" data-level="36" data-path="scatterplots-and-logs.html"><a href="scatterplots-and-logs.html"><i class="fa fa-check"></i><b>36</b> Scatterplots and logs</a>
<ul>
<li class="chapter" data-level="36.1" data-path="scatterplots-and-logs.html"><a href="scatterplots-and-logs.html#regression-correlation-logs"><i class="fa fa-check"></i><b>36.1</b> Regression / correlation; logs</a></li>
</ul></li>
<li class="chapter" data-level="37" data-path="making-fancy-plots.html"><a href="making-fancy-plots.html"><i class="fa fa-check"></i><b>37</b> Making fancy plots</a></li>
<li class="chapter" data-level="38" data-path="using-ggplots-stat-summary.html"><a href="using-ggplots-stat-summary.html"><i class="fa fa-check"></i><b>38</b> Using ggplot’s stat_summary</a>
<ul>
<li class="chapter" data-level="38.1" data-path="using-ggplots-stat-summary.html"><a href="using-ggplots-stat-summary.html#standard-plot-of-means-w-error-bars"><i class="fa fa-check"></i><b>38.1</b> Standard plot of means w/ error bars</a></li>
<li class="chapter" data-level="38.2" data-path="using-ggplots-stat-summary.html"><a href="using-ggplots-stat-summary.html#w-stat_summarygeom-pointrange"><i class="fa fa-check"></i><b>38.2</b> w/ stat_summary(…geom = “pointrange”)</a></li>
<li class="chapter" data-level="38.3" data-path="using-ggplots-stat-summary.html"><a href="using-ggplots-stat-summary.html#plot-just-means"><i class="fa fa-check"></i><b>38.3</b> Plot just means</a></li>
<li class="chapter" data-level="38.4" data-path="using-ggplots-stat-summary.html"><a href="using-ggplots-stat-summary.html#w-geom-point"><i class="fa fa-check"></i><b>38.4</b> w/ geom = “point”</a></li>
</ul></li>
<li class="chapter" data-level="39" data-path="making-a-simple-data-frame-for-boxplots.html"><a href="making-a-simple-data-frame-for-boxplots.html"><i class="fa fa-check"></i><b>39</b> Making a simple data frame for boxplots</a></li>
<li class="chapter" data-level="40" data-path="working-with-data-in-2-seperate-vectors-1.html"><a href="working-with-data-in-2-seperate-vectors-1.html"><i class="fa fa-check"></i><b>40</b> Working with data in 2 seperate vectors</a>
<ul>
<li class="chapter" data-level="40.1" data-path="working-with-data-in-2-seperate-vectors-1.html"><a href="working-with-data-in-2-seperate-vectors-1.html#summary-statistics-1"><i class="fa fa-check"></i><b>40.1</b> Summary Statistics</a></li>
<li class="chapter" data-level="40.2" data-path="working-with-data-in-2-seperate-vectors-1.html"><a href="working-with-data-in-2-seperate-vectors-1.html#making-a-dataframe-1"><i class="fa fa-check"></i><b>40.2</b> Making a dataframe</a></li>
<li class="chapter" data-level="40.3" data-path="working-with-data-in-2-seperate-vectors-1.html"><a href="working-with-data-in-2-seperate-vectors-1.html#making-a-boxplot-1"><i class="fa fa-check"></i><b>40.3</b> Making a boxplot</a></li>
<li class="chapter" data-level="40.4" data-path="working-with-data-in-2-seperate-vectors-1.html"><a href="working-with-data-in-2-seperate-vectors-1.html#making-a-histogram-1"><i class="fa fa-check"></i><b>40.4</b> Making a histogram</a></li>
<li class="chapter" data-level="40.5" data-path="working-with-data-in-2-seperate-vectors-1.html"><a href="working-with-data-in-2-seperate-vectors-1.html#making-a-density-plot-1"><i class="fa fa-check"></i><b>40.5</b> Making a density plot</a></li>
<li class="chapter" data-level="40.6" data-path="working-with-data-in-2-seperate-vectors-1.html"><a href="working-with-data-in-2-seperate-vectors-1.html#advanced-2"><i class="fa fa-check"></i><b>40.6</b> Advanced</a></li>
<li class="chapter" data-level="40.7" data-path="working-with-data-in-2-seperate-vectors-1.html"><a href="working-with-data-in-2-seperate-vectors-1.html#historical-reference-1"><i class="fa fa-check"></i><b>40.7</b> Historical reference</a>
<ul>
<li class="chapter" data-level="40.7.1" data-path="working-with-data-in-2-seperate-vectors-1.html"><a href="working-with-data-in-2-seperate-vectors-1.html#base-r-1"><i class="fa fa-check"></i><b>40.7.1</b> Base R</a></li>
<li class="chapter" data-level="40.7.2" data-path="working-with-data-in-2-seperate-vectors-1.html"><a href="working-with-data-in-2-seperate-vectors-1.html#ggplots-qplot-1"><i class="fa fa-check"></i><b>40.7.2</b> ggplot’s qplot</a></li>
<li class="chapter" data-level="40.7.3" data-path="working-with-data-in-2-seperate-vectors-1.html"><a href="working-with-data-in-2-seperate-vectors-1.html#standard-ggplot-1"><i class="fa fa-check"></i><b>40.7.3</b> Standard ggplot</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="41" data-path="barplots.html"><a href="barplots.html"><i class="fa fa-check"></i><b>41</b> Barplots</a></li>
<li class="chapter" data-level="42" data-path="boxplots-how-they-are-made.html"><a href="boxplots-how-they-are-made.html"><i class="fa fa-check"></i><b>42</b> Boxplots - how they are made</a></li>
<li class="chapter" data-level="43" data-path="boxplots-1.html"><a href="boxplots-1.html"><i class="fa fa-check"></i><b>43</b> Boxplots</a>
<ul>
<li class="chapter" data-level="43.1" data-path="boxplots-1.html"><a href="boxplots-1.html#preliminaries-8"><i class="fa fa-check"></i><b>43.1</b> Preliminaries</a>
<ul>
<li class="chapter" data-level="43.1.1" data-path="boxplots-1.html"><a href="boxplots-1.html#load-packagtes"><i class="fa fa-check"></i><b>43.1.1</b> Load packagtes</a></li>
<li class="chapter" data-level="43.1.2" data-path="boxplots-1.html"><a href="boxplots-1.html#load-data-1"><i class="fa fa-check"></i><b>43.1.2</b> Load data</a></li>
</ul></li>
<li class="chapter" data-level="43.2" data-path="boxplots-1.html"><a href="boxplots-1.html#boxplots-2"><i class="fa fa-check"></i><b>43.2</b> Boxplots</a></li>
<li class="chapter" data-level="43.3" data-path="boxplots-1.html"><a href="boxplots-1.html#aside-notched-boxplots"><i class="fa fa-check"></i><b>43.3</b> Aside: Notched boxplots</a></li>
<li class="chapter" data-level="43.4" data-path="boxplots-1.html"><a href="boxplots-1.html#fill"><i class="fa fa-check"></i><b>43.4</b> Fill</a></li>
<li class="chapter" data-level="43.5" data-path="boxplots-1.html"><a href="boxplots-1.html#adding-raw-data"><i class="fa fa-check"></i><b>43.5</b> Adding raw data</a></li>
<li class="chapter" data-level="43.6" data-path="boxplots-1.html"><a href="boxplots-1.html#jitter-the-raw-data"><i class="fa fa-check"></i><b>43.6</b> Jitter the raw data</a>
<ul>
<li class="chapter" data-level="43.6.1" data-path="boxplots-1.html"><a href="boxplots-1.html#plotting-multple-plots-with-cowplotplot_grid"><i class="fa fa-check"></i><b>43.6.1</b> Plotting multple plots with cowplot::plot_grid</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="44" data-path="central-tendency.html"><a href="central-tendency.html"><i class="fa fa-check"></i><b>44</b> Central tendency</a>
<ul>
<li class="chapter" data-level="44.1" data-path="central-tendency.html"><a href="central-tendency.html#preliminaries-9"><i class="fa fa-check"></i><b>44.1</b> Preliminaries</a>
<ul>
<li class="chapter" data-level="44.1.1" data-path="central-tendency.html"><a href="central-tendency.html#load-packagtes-1"><i class="fa fa-check"></i><b>44.1.1</b> Load packagtes</a></li>
<li class="chapter" data-level="44.1.2" data-path="central-tendency.html"><a href="central-tendency.html#load-data-2"><i class="fa fa-check"></i><b>44.1.2</b> Load data</a></li>
</ul></li>
<li class="chapter" data-level="44.2" data-path="central-tendency.html"><a href="central-tendency.html#getting-to-know-the-frogs"><i class="fa fa-check"></i><b>44.2</b> Getting to know the frogs</a></li>
<li class="chapter" data-level="44.3" data-path="central-tendency.html"><a href="central-tendency.html#summary-statistics-2"><i class="fa fa-check"></i><b>44.3</b> Summary statistics</a>
<ul>
<li class="chapter" data-level="44.3.1" data-path="central-tendency.html"><a href="central-tendency.html#overall-summary"><i class="fa fa-check"></i><b>44.3.1</b> Overall summary</a></li>
<li class="chapter" data-level="44.3.2" data-path="central-tendency.html"><a href="central-tendency.html#individual-summary-stats"><i class="fa fa-check"></i><b>44.3.2</b> Individual summary stats</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="45" data-path="error-plots.html"><a href="error-plots.html"><i class="fa fa-check"></i><b>45</b> Error plots</a>
<ul>
<li class="chapter" data-level="45.1" data-path="error-plots.html"><a href="error-plots.html#plot-means-with-error-bars"><i class="fa fa-check"></i><b>45.1</b> Plot means with error bars</a></li>
<li class="chapter" data-level="45.2" data-path="error-plots.html"><a href="error-plots.html#arms"><i class="fa fa-check"></i><b>45.2</b> arms</a></li>
</ul></li>
<li class="chapter" data-level="46" data-path="how-histograms-are-made.html"><a href="how-histograms-are-made.html"><i class="fa fa-check"></i><b>46</b> How histograms are made</a></li>
<li class="chapter" data-level="47" data-path="histograms.html"><a href="histograms.html"><i class="fa fa-check"></i><b>47</b> Histograms</a>
<ul>
<li class="chapter" data-level="47.1" data-path="histograms.html"><a href="histograms.html#preliminaries-10"><i class="fa fa-check"></i><b>47.1</b> Preliminaries</a>
<ul>
<li class="chapter" data-level="47.1.1" data-path="histograms.html"><a href="histograms.html#load-packages-1"><i class="fa fa-check"></i><b>47.1.1</b> Load packages</a></li>
<li class="chapter" data-level="47.1.2" data-path="histograms.html"><a href="histograms.html#load-data-3"><i class="fa fa-check"></i><b>47.1.2</b> Load data</a></li>
</ul></li>
<li class="chapter" data-level="47.2" data-path="histograms.html"><a href="histograms.html#section-1"><i class="fa fa-check"></i><b>47.2</b> </a></li>
<li class="chapter" data-level="47.3" data-path="histograms.html"><a href="histograms.html#examples-of-histogram"><i class="fa fa-check"></i><b>47.3</b> Examples of histogram</a>
<ul>
<li class="chapter" data-level="47.3.1" data-path="histograms.html"><a href="histograms.html#normal-ish-data"><i class="fa fa-check"></i><b>47.3.1</b> Normal-ish data</a></li>
</ul></li>
<li class="chapter" data-level="47.4" data-path="histograms.html"><a href="histograms.html#skewed-data"><i class="fa fa-check"></i><b>47.4</b> Skewed data</a></li>
<li class="chapter" data-level="47.5" data-path="histograms.html"><a href="histograms.html#heavily-skewed-data"><i class="fa fa-check"></i><b>47.5</b> Heavily skewed data</a></li>
<li class="chapter" data-level="47.6" data-path="histograms.html"><a href="histograms.html#section-2"><i class="fa fa-check"></i><b>47.6</b> </a></li>
<li class="chapter" data-level="47.7" data-path="histograms.html"><a href="histograms.html#fill-1"><i class="fa fa-check"></i><b>47.7</b> Fill</a></li>
<li class="chapter" data-level="47.8" data-path="histograms.html"><a href="histograms.html#central-tendency-1"><i class="fa fa-check"></i><b>47.8</b> Central tendency</a></li>
<li class="chapter" data-level="47.9" data-path="histograms.html"><a href="histograms.html#median"><i class="fa fa-check"></i><b>47.9</b> Median</a></li>
<li class="chapter" data-level="47.10" data-path="histograms.html"><a href="histograms.html#density-plot"><i class="fa fa-check"></i><b>47.10</b> Density plot</a>
<ul>
<li class="chapter" data-level="47.10.1" data-path="histograms.html"><a href="histograms.html#plotting-multple-plots-with-cowplotplot_grid-1"><i class="fa fa-check"></i><b>47.10.1</b> Plotting multple plots with cowplot::plot_grid</a></li>
</ul></li>
<li class="chapter" data-level="47.11" data-path="histograms.html"><a href="histograms.html#align-vert"><i class="fa fa-check"></i><b>47.11</b> Align vert</a></li>
<li class="chapter" data-level="47.12" data-path="histograms.html"><a href="histograms.html#density-plot-overlay"><i class="fa fa-check"></i><b>47.12</b> Density plot overlay</a></li>
<li class="chapter" data-level="47.13" data-path="histograms.html"><a href="histograms.html#density-plot-1"><i class="fa fa-check"></i><b>47.13</b> Density plot</a></li>
</ul></li>
<li class="chapter" data-level="48" data-path="boxplots-3.html"><a href="boxplots-3.html"><i class="fa fa-check"></i><b>48</b> Boxplots</a>
<ul>
<li class="chapter" data-level="48.1" data-path="boxplots-3.html"><a href="boxplots-3.html#preliminaries-11"><i class="fa fa-check"></i><b>48.1</b> Preliminaries</a>
<ul>
<li class="chapter" data-level="48.1.1" data-path="boxplots-3.html"><a href="boxplots-3.html#load-packagtes-2"><i class="fa fa-check"></i><b>48.1.1</b> Load packagtes</a></li>
<li class="chapter" data-level="48.1.2" data-path="boxplots-3.html"><a href="boxplots-3.html#load-data-4"><i class="fa fa-check"></i><b>48.1.2</b> Load data</a></li>
</ul></li>
<li class="chapter" data-level="48.2" data-path="boxplots-3.html"><a href="boxplots-3.html#boxplots-4"><i class="fa fa-check"></i><b>48.2</b> Boxplots</a></li>
<li class="chapter" data-level="48.3" data-path="boxplots-3.html"><a href="boxplots-3.html#fill-2"><i class="fa fa-check"></i><b>48.3</b> Fill</a></li>
<li class="chapter" data-level="48.4" data-path="boxplots-3.html"><a href="boxplots-3.html#central-tendency-2"><i class="fa fa-check"></i><b>48.4</b> Central tendency</a></li>
<li class="chapter" data-level="48.5" data-path="boxplots-3.html"><a href="boxplots-3.html#median-1"><i class="fa fa-check"></i><b>48.5</b> Median</a></li>
<li class="chapter" data-level="48.6" data-path="boxplots-3.html"><a href="boxplots-3.html#density-plot-2"><i class="fa fa-check"></i><b>48.6</b> Density plot</a>
<ul>
<li class="chapter" data-level="48.6.1" data-path="boxplots-3.html"><a href="boxplots-3.html#plotting-multple-plots-with-cowplotplot_grid-2"><i class="fa fa-check"></i><b>48.6.1</b> Plotting multple plots with cowplot::plot_grid</a></li>
</ul></li>
<li class="chapter" data-level="48.7" data-path="boxplots-3.html"><a href="boxplots-3.html#align-vert-1"><i class="fa fa-check"></i><b>48.7</b> Align vert</a></li>
<li class="chapter" data-level="48.8" data-path="boxplots-3.html"><a href="boxplots-3.html#density-plot-overlay-1"><i class="fa fa-check"></i><b>48.8</b> Density plot overlay</a></li>
<li class="chapter" data-level="48.9" data-path="boxplots-3.html"><a href="boxplots-3.html#density-plot-3"><i class="fa fa-check"></i><b>48.9</b> Density plot</a></li>
</ul></li>
<li class="chapter" data-level="49" data-path="frog-arms.html"><a href="frog-arms.html"><i class="fa fa-check"></i><b>49</b> Frog arms</a>
<ul>
<li class="chapter" data-level="49.1" data-path="frog-arms.html"><a href="frog-arms.html#preliminaries-12"><i class="fa fa-check"></i><b>49.1</b> Preliminaries</a>
<ul>
<li class="chapter" data-level="49.1.1" data-path="frog-arms.html"><a href="frog-arms.html#load-packages-2"><i class="fa fa-check"></i><b>49.1.1</b> Load packages</a></li>
<li class="chapter" data-level="49.1.2" data-path="frog-arms.html"><a href="frog-arms.html#load-data-5"><i class="fa fa-check"></i><b>49.1.2</b> Load data</a></li>
</ul></li>
<li class="chapter" data-level="49.2" data-path="frog-arms.html"><a href="frog-arms.html#a-1st-encounter-with-dplyr"><i class="fa fa-check"></i><b>49.2</b> A 1st encounter with dplyr</a></li>
<li class="chapter" data-level="49.3" data-path="frog-arms.html"><a href="frog-arms.html#group_by"><i class="fa fa-check"></i><b>49.3</b> group_by</a></li>
<li class="chapter" data-level="49.4" data-path="frog-arms.html"><a href="frog-arms.html#alternatives"><i class="fa fa-check"></i><b>49.4</b> Alternatives</a>
<ul>
<li class="chapter" data-level="49.4.1" data-path="frog-arms.html"><a href="frog-arms.html#dobysummaryby"><i class="fa fa-check"></i><b>49.4.1</b> doBy::summaryBy</a></li>
<li class="chapter" data-level="49.4.2" data-path="frog-arms.html"><a href="frog-arms.html#tapply"><i class="fa fa-check"></i><b>49.4.2</b> tapply()</a></li>
<li class="chapter" data-level="49.4.3" data-path="frog-arms.html"><a href="frog-arms.html#reshape2dcast"><i class="fa fa-check"></i><b>49.4.3</b> reshape2::dcast</a></li>
</ul></li>
<li class="chapter" data-level="49.5" data-path="frog-arms.html"><a href="frog-arms.html#your-turn-3"><i class="fa fa-check"></i><b>49.5</b> Your turn</a></li>
<li class="chapter" data-level="49.6" data-path="frog-arms.html"><a href="frog-arms.html#doing-math-for-stats-by-hand"><i class="fa fa-check"></i><b>49.6</b> Doing math for stats by hand</a>
<ul>
<li class="chapter" data-level="49.6.1" data-path="frog-arms.html"><a href="frog-arms.html#calculating-the-mean-gene-length-by-hand"><i class="fa fa-check"></i><b>49.6.1</b> Calculating the mean gene length by hand</a></li>
<li class="chapter" data-level="49.6.2" data-path="frog-arms.html"><a href="frog-arms.html#the-denomintor"><i class="fa fa-check"></i><b>49.6.2</b> The denomintor</a></li>
<li class="chapter" data-level="49.6.3" data-path="frog-arms.html"><a href="frog-arms.html#the-mean"><i class="fa fa-check"></i><b>49.6.3</b> The mean</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="50" data-path="doing-math-for-stats-by-hand-standard-deviation.html"><a href="doing-math-for-stats-by-hand-standard-deviation.html"><i class="fa fa-check"></i><b>50</b> Doing math for stats by hand: standard deviation</a>
<ul>
<li class="chapter" data-level="50.1" data-path="doing-math-for-stats-by-hand-standard-deviation.html"><a href="doing-math-for-stats-by-hand-standard-deviation.html#calcualting-variance-standard-deviation-by-hand-step-by-step"><i class="fa fa-check"></i><b>50.1</b> Calcualting variance &amp; standard deviation by hand, step by step</a></li>
<li class="chapter" data-level="50.2" data-path="doing-math-for-stats-by-hand-standard-deviation.html"><a href="doing-math-for-stats-by-hand-standard-deviation.html#deviations-between-the-mean-and-each-observation"><i class="fa fa-check"></i><b>50.2</b> “Deviations” between the mean and each observation</a></li>
<li class="chapter" data-level="50.3" data-path="doing-math-for-stats-by-hand-standard-deviation.html"><a href="doing-math-for-stats-by-hand-standard-deviation.html#start-making-a-dataframe-df"><i class="fa fa-check"></i><b>50.3</b> Start making a dataframe (df)</a></li>
</ul></li>
<li class="chapter" data-level="51" data-path="calculate-the-squared-deviations-between-the-mean-and-each-observation.html"><a href="calculate-the-squared-deviations-between-the-mean-and-each-observation.html"><i class="fa fa-check"></i><b>51</b> Calculate the “Squared deviations” between the mean and each observation</a></li>
<li class="chapter" data-level="52" data-path="sum-of-squares-between-the-mean-and-each-deviation.html"><a href="sum-of-squares-between-the-mean-and-each-deviation.html"><i class="fa fa-check"></i><b>52</b> Sum of squares between the mean and each deviation</a></li>
<li class="chapter" data-level="53" data-path="and-nowthe-variance.html"><a href="and-nowthe-variance.html"><i class="fa fa-check"></i><b>53</b> And now…the variance</a></li>
<li class="chapter" data-level="54" data-path="and-nowthe-standard-deviation.html"><a href="and-nowthe-standard-deviation.html"><i class="fa fa-check"></i><b>54</b> And now…the standard deviation</a></li>
<li class="chapter" data-level="55" data-path="the-standard-error-se.html"><a href="the-standard-error-se.html"><i class="fa fa-check"></i><b>55</b> The standard error (SE)</a></li>
<li class="chapter" data-level="56" data-path="doing-math-in-r.html"><a href="doing-math-in-r.html"><i class="fa fa-check"></i><b>56</b> Doing Math in R</a></li>
<li class="chapter" data-level="57" data-path="introduction-6.html"><a href="introduction-6.html"><i class="fa fa-check"></i><b>57</b> Introduction</a>
<ul>
<li class="chapter" data-level="57.1" data-path="introduction-6.html"><a href="introduction-6.html#the-basic-math-stuff-in-r"><i class="fa fa-check"></i><b>57.1</b> The basic math stuff in R</a></li>
</ul></li>
<li class="chapter" data-level="58" data-path="r-quirks-putting-things-in-vectors.html"><a href="r-quirks-putting-things-in-vectors.html"><i class="fa fa-check"></i><b>58</b> R quirks: putting things in vectors</a>
<ul>
<li class="chapter" data-level="58.1" data-path="r-quirks-putting-things-in-vectors.html"><a href="r-quirks-putting-things-in-vectors.html#how-r-works-with-strings-of-numbers"><i class="fa fa-check"></i><b>58.1</b> How R works with strings of numbers</a></li>
<li class="chapter" data-level="58.2" data-path="r-quirks-putting-things-in-vectors.html"><a href="r-quirks-putting-things-in-vectors.html#how-r-works-with-strings-of-numbers-1"><i class="fa fa-check"></i><b>58.2</b> How R works with strings of numbers</a></li>
<li class="chapter" data-level="58.3" data-path="r-quirks-putting-things-in-vectors.html"><a href="r-quirks-putting-things-in-vectors.html#fixing-the-problem"><i class="fa fa-check"></i><b>58.3</b> Fixing the problem</a></li>
</ul></li>
<li class="chapter" data-level="59" data-path="summary-stats-for-main-pa-eagle-dataframe.html"><a href="summary-stats-for-main-pa-eagle-dataframe.html"><i class="fa fa-check"></i><b>59</b> Summary stats for main PA eagle dataframe</a>
<ul>
<li class="chapter" data-level="59.0.1" data-path="summary-stats-for-main-pa-eagle-dataframe.html"><a href="summary-stats-for-main-pa-eagle-dataframe.html#na-is-a-big-deal-in-r"><i class="fa fa-check"></i><b>59.0.1</b> “NA” is a big deal in R</a></li>
<li class="chapter" data-level="59.0.2" data-path="summary-stats-for-main-pa-eagle-dataframe.html"><a href="summary-stats-for-main-pa-eagle-dataframe.html#r-also-is-very-picky-about-uppper-vs.-lower-case"><i class="fa fa-check"></i><b>59.0.2</b> R also is VERY picky about uppper vs. lower case</a></li>
<li class="chapter" data-level="59.0.3" data-path="summary-stats-for-main-pa-eagle-dataframe.html"><a href="summary-stats-for-main-pa-eagle-dataframe.html#most-basic-r-commands-are-lower-case"><i class="fa fa-check"></i><b>59.0.3</b> Most basic R commands are lower case</a></li>
<li class="chapter" data-level="59.0.4" data-path="summary-stats-for-main-pa-eagle-dataframe.html"><a href="summary-stats-for-main-pa-eagle-dataframe.html#standard-error-in-r"><i class="fa fa-check"></i><b>59.0.4</b> Standard error in R</a></li>
</ul></li>
<li class="chapter" data-level="60" data-path="variability.html"><a href="variability.html"><i class="fa fa-check"></i><b>60</b> Variability</a>
<ul>
<li class="chapter" data-level="60.1" data-path="variability.html"><a href="variability.html#preliminaries-13"><i class="fa fa-check"></i><b>60.1</b> Preliminaries</a>
<ul>
<li class="chapter" data-level="60.1.1" data-path="variability.html"><a href="variability.html#load-packagtes-3"><i class="fa fa-check"></i><b>60.1.1</b> Load packagtes</a></li>
<li class="chapter" data-level="60.1.2" data-path="variability.html"><a href="variability.html#load-data-6"><i class="fa fa-check"></i><b>60.1.2</b> Load data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="61" data-path="p-values.html"><a href="p-values.html"><i class="fa fa-check"></i><b>61</b> P-values</a></li>
<li class="chapter" data-level="62" data-path="introduction-7.html"><a href="introduction-7.html"><i class="fa fa-check"></i><b>62</b> Introduction</a></li>
<li class="chapter" data-level="63" data-path="p-value-definition-to-memorize.html"><a href="p-value-definition-to-memorize.html"><i class="fa fa-check"></i><b>63</b> P-value definition to memorize</a></li>
<li class="chapter" data-level="64" data-path="p-values-are-tricky.html"><a href="p-values-are-tricky.html"><i class="fa fa-check"></i><b>64</b> P-values are tricky</a></li>
<li class="chapter" data-level="65" data-path="p-value-example-coin-flipping.html"><a href="p-value-example-coin-flipping.html"><i class="fa fa-check"></i><b>65</b> P-value example: coin flipping</a></li>
<li class="chapter" data-level="66" data-path="p-value-example-1.html"><a href="p-value-example-1.html"><i class="fa fa-check"></i><b>66</b> P-value example 1</a></li>
<li class="chapter" data-level="67" data-path="p-value-example-2-heights-of-elementary-school-students.html"><a href="p-value-example-2-heights-of-elementary-school-students.html"><i class="fa fa-check"></i><b>67</b> P-value example 2: heights of elementary school students</a></li>
<li class="chapter" data-level="68" data-path="p-values-and-biological-sequences.html"><a href="p-values-and-biological-sequences.html"><i class="fa fa-check"></i><b>68</b> P-values and biological sequences</a></li>
<li class="chapter" data-level="69" data-path="inference-by-eye-using-ra-fishers-cat-data.html"><a href="inference-by-eye-using-ra-fishers-cat-data.html"><i class="fa fa-check"></i><b>69</b> Inference-by-eye using RA Fisher’s Cat Data</a>
<ul>
<li class="chapter" data-level="69.1" data-path="inference-by-eye-using-ra-fishers-cat-data.html"><a href="inference-by-eye-using-ra-fishers-cat-data.html#load-the-data"><i class="fa fa-check"></i><b>69.1</b> Load the data</a></li>
<li class="chapter" data-level="69.2" data-path="inference-by-eye-using-ra-fishers-cat-data.html"><a href="inference-by-eye-using-ra-fishers-cat-data.html#look-at-the-data"><i class="fa fa-check"></i><b>69.2</b> Look at the data</a></li>
<li class="chapter" data-level="69.3" data-path="inference-by-eye-using-ra-fishers-cat-data.html"><a href="inference-by-eye-using-ra-fishers-cat-data.html#summarize-the-body-weight-bwt-data-old-school-using-summaryby"><i class="fa fa-check"></i><b>69.3</b> Summarize the body weight (Bwt) data old-school using summaryBy()</a></li>
<li class="chapter" data-level="69.4" data-path="inference-by-eye-using-ra-fishers-cat-data.html"><a href="inference-by-eye-using-ra-fishers-cat-data.html#plot-the-data"><i class="fa fa-check"></i><b>69.4</b> Plot the data</a>
<ul>
<li class="chapter" data-level="69.4.1" data-path="inference-by-eye-using-ra-fishers-cat-data.html"><a href="inference-by-eye-using-ra-fishers-cat-data.html#visualize-the-raw-data"><i class="fa fa-check"></i><b>69.4.1</b> Visualize the raw data</a></li>
<li class="chapter" data-level="69.4.2" data-path="inference-by-eye-using-ra-fishers-cat-data.html"><a href="inference-by-eye-using-ra-fishers-cat-data.html#plot-the-means-with-error-bars"><i class="fa fa-check"></i><b>69.4.2</b> Plot the means with error bars</a></li>
</ul></li>
<li class="chapter" data-level="69.5" data-path="inference-by-eye-using-ra-fishers-cat-data.html"><a href="inference-by-eye-using-ra-fishers-cat-data.html#t-test"><i class="fa fa-check"></i><b>69.5</b> T-test</a></li>
</ul></li>
<li class="chapter" data-level="70" data-path="bootstrapping-tutrial-fst-from-evans-et-al-2016.html"><a href="bootstrapping-tutrial-fst-from-evans-et-al-2016.html"><i class="fa fa-check"></i><b>70</b> Bootstrapping Tutrial: Fst from Evans et al 2016</a></li>
<li class="chapter" data-level="71" data-path="fst-data.html"><a href="fst-data.html"><i class="fa fa-check"></i><b>71</b> Fst data</a>
<ul>
<li class="chapter" data-level="71.1" data-path="fst-data.html"><a href="fst-data.html#load-fst-into-vector"><i class="fa fa-check"></i><b>71.1</b> Load Fst into vector</a></li>
<li class="chapter" data-level="71.2" data-path="fst-data.html"><a href="fst-data.html#mean-fst"><i class="fa fa-check"></i><b>71.2</b> Mean Fst</a></li>
<li class="chapter" data-level="71.3" data-path="fst-data.html"><a href="fst-data.html#distribution-of-fst-values"><i class="fa fa-check"></i><b>71.3</b> Distribution of Fst values</a>
<ul>
<li class="chapter" data-level="71.3.1" data-path="fst-data.html"><a href="fst-data.html#histogram-base-r"><i class="fa fa-check"></i><b>71.3.1</b> Histogram base R</a></li>
<li class="chapter" data-level="71.3.2" data-path="fst-data.html"><a href="fst-data.html#boxplot-base-r"><i class="fa fa-check"></i><b>71.3.2</b> Boxplot base R</a></li>
<li class="chapter" data-level="71.3.3" data-path="fst-data.html"><a href="fst-data.html#boxplot-ggplot"><i class="fa fa-check"></i><b>71.3.3</b> Boxplot ggplot</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="72" data-path="boostrapping-fst.html"><a href="boostrapping-fst.html"><i class="fa fa-check"></i><b>72</b> Boostrapping Fst</a>
<ul>
<li class="chapter" data-level="72.1" data-path="boostrapping-fst.html"><a href="boostrapping-fst.html#define-function-for-the-mean"><i class="fa fa-check"></i><b>72.1</b> Define function for the mean</a>
<ul>
<li class="chapter" data-level="72.1.1" data-path="boostrapping-fst.html"><a href="boostrapping-fst.html#the-function-mean.fun"><i class="fa fa-check"></i><b>72.1.1</b> The function mean.fun()</a></li>
</ul></li>
<li class="chapter" data-level="72.2" data-path="boostrapping-fst.html"><a href="boostrapping-fst.html#running-boot"><i class="fa fa-check"></i><b>72.2</b> Running boot</a>
<ul>
<li class="chapter" data-level="72.2.1" data-path="boostrapping-fst.html"><a href="boostrapping-fst.html#run-the-function"><i class="fa fa-check"></i><b>72.2.1</b> Run the function</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="73" data-path="compare-fst-to-qst.html"><a href="compare-fst-to-qst.html"><i class="fa fa-check"></i><b>73</b> Compare Fst to Qst</a>
<ul>
<li class="chapter" data-level="73.1" data-path="compare-fst-to-qst.html"><a href="compare-fst-to-qst.html#plot-with-ggplot"><i class="fa fa-check"></i><b>73.1</b> Plot with ggplot</a>
<ul>
<li class="chapter" data-level="73.1.1" data-path="compare-fst-to-qst.html"><a href="compare-fst-to-qst.html#generate-jittered-locations"><i class="fa fa-check"></i><b>73.1.1</b> Generate Jittered locations</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="74" data-path="from-the-original-text.html"><a href="from-the-original-text.html"><i class="fa fa-check"></i><b>74</b> From the original text:</a></li>
<li class="chapter" data-level="75" data-path="d-plotting.html"><a href="d-plotting.html"><i class="fa fa-check"></i><b>75</b> 3D plotting</a></li>
<li class="chapter" data-level="76" data-path="coding-multivariate-data-color-and-shape.html"><a href="coding-multivariate-data-color-and-shape.html"><i class="fa fa-check"></i><b>76</b> Coding multivariate data - color and shape</a></li>
<li class="chapter" data-level="77" data-path="facets.html"><a href="facets.html"><i class="fa fa-check"></i><b>77</b> Facets</a></li>
<li class="chapter" data-level="78" data-path="multivariate-data-analysis-making-sense-of-high-dimensional-data-wpca-etc.html"><a href="multivariate-data-analysis-making-sense-of-high-dimensional-data-wpca-etc.html"><i class="fa fa-check"></i><b>78</b> Multivariate Data Analysis: Making Sense of High-Dimensional Data w/PCA etc</a>
<ul>
<li class="chapter" data-level="78.1" data-path="multivariate-data-analysis-making-sense-of-high-dimensional-data-wpca-etc.html"><a href="multivariate-data-analysis-making-sense-of-high-dimensional-data-wpca-etc.html#preliminaries-14"><i class="fa fa-check"></i><b>78.1</b> Preliminaries</a>
<ul>
<li class="chapter" data-level="78.1.1" data-path="multivariate-data-analysis-making-sense-of-high-dimensional-data-wpca-etc.html"><a href="multivariate-data-analysis-making-sense-of-high-dimensional-data-wpca-etc.html#packages-4"><i class="fa fa-check"></i><b>78.1.1</b> Packages</a></li>
<li class="chapter" data-level="78.1.2" data-path="multivariate-data-analysis-making-sense-of-high-dimensional-data-wpca-etc.html"><a href="multivariate-data-analysis-making-sense-of-high-dimensional-data-wpca-etc.html#data-sets"><i class="fa fa-check"></i><b>78.1.2</b> Data sets</a></li>
</ul></li>
<li class="chapter" data-level="78.2" data-path="multivariate-data-analysis-making-sense-of-high-dimensional-data-wpca-etc.html"><a href="multivariate-data-analysis-making-sense-of-high-dimensional-data-wpca-etc.html#high-dimensional-data-in-biology"><i class="fa fa-check"></i><b>78.2</b> High-dimensional Data in Biology</a>
<ul>
<li class="chapter" data-level="78.2.1" data-path="multivariate-data-analysis-making-sense-of-high-dimensional-data-wpca-etc.html"><a href="multivariate-data-analysis-making-sense-of-high-dimensional-data-wpca-etc.html#morphology-how-do-species-vary"><i class="fa fa-check"></i><b>78.2.1</b> Morphology: How do species vary</a></li>
<li class="chapter" data-level="78.2.2" data-path="multivariate-data-analysis-making-sense-of-high-dimensional-data-wpca-etc.html"><a href="multivariate-data-analysis-making-sense-of-high-dimensional-data-wpca-etc.html#ecology-do-trees-prefer-certain-habitats"><i class="fa fa-check"></i><b>78.2.2</b> Ecology: Do trees prefer certain habitats</a></li>
<li class="chapter" data-level="78.2.3" data-path="multivariate-data-analysis-making-sense-of-high-dimensional-data-wpca-etc.html"><a href="multivariate-data-analysis-making-sense-of-high-dimensional-data-wpca-etc.html#phylogenetics"><i class="fa fa-check"></i><b>78.2.3</b> Phylogenetics</a></li>
<li class="chapter" data-level="78.2.4" data-path="multivariate-data-analysis-making-sense-of-high-dimensional-data-wpca-etc.html"><a href="multivariate-data-analysis-making-sense-of-high-dimensional-data-wpca-etc.html#transcriptomics"><i class="fa fa-check"></i><b>78.2.4</b> Transcriptomics</a></li>
</ul></li>
<li class="chapter" data-level="78.3" data-path="multivariate-data-analysis-making-sense-of-high-dimensional-data-wpca-etc.html"><a href="multivariate-data-analysis-making-sense-of-high-dimensional-data-wpca-etc.html#exploratory-versus-hypothesis-testing"><i class="fa fa-check"></i><b>78.3</b> Exploratory versus Hypothesis testing</a></li>
</ul></li>
<li class="chapter" data-level="79" data-path="overview-the-vegan-package-for-pca.html"><a href="overview-the-vegan-package-for-pca.html"><i class="fa fa-check"></i><b>79</b> Overview: The vegan package for PCA</a>
<ul>
<li class="chapter" data-level="79.0.1" data-path="overview-the-vegan-package-for-pca.html"><a href="overview-the-vegan-package-for-pca.html#download-packages"><i class="fa fa-check"></i><b>79.0.1</b> Download packages</a></li>
<li class="chapter" data-level="79.0.2" data-path="overview-the-vegan-package-for-pca.html"><a href="overview-the-vegan-package-for-pca.html#load-the-libraries-3"><i class="fa fa-check"></i><b>79.0.2</b> Load the libraries</a></li>
<li class="chapter" data-level="79.0.3" data-path="overview-the-vegan-package-for-pca.html"><a href="overview-the-vegan-package-for-pca.html#load-the-msleep-package-3"><i class="fa fa-check"></i><b>79.0.3</b> Load the msleep package</a></li>
<li class="chapter" data-level="79.0.4" data-path="overview-the-vegan-package-for-pca.html"><a href="overview-the-vegan-package-for-pca.html#subset-numeric-variables"><i class="fa fa-check"></i><b>79.0.4</b> Subset numeric variables</a></li>
<li class="chapter" data-level="79.1" data-path="overview-the-vegan-package-for-pca.html"><a href="overview-the-vegan-package-for-pca.html#principal-components-analysis---base-r"><i class="fa fa-check"></i><b>79.1</b> Principal components analysis - base R</a></li>
<li class="chapter" data-level="79.2" data-path="overview-the-vegan-package-for-pca.html"><a href="overview-the-vegan-package-for-pca.html#principal-components-analysis---vegan"><i class="fa fa-check"></i><b>79.2</b> Principal components analysis - vegan</a></li>
<li class="chapter" data-level="79.3" data-path="overview-the-vegan-package-for-pca.html"><a href="overview-the-vegan-package-for-pca.html#task-4"><i class="fa fa-check"></i><b>79.3</b> Task</a></li>
</ul></li>
<li class="chapter" data-level="80" data-path="pca-etc-worked-example.html"><a href="pca-etc-worked-example.html"><i class="fa fa-check"></i><b>80</b> PCA etc worked example</a>
<ul>
<li class="chapter" data-level="80.1" data-path="pca-etc-worked-example.html"><a href="pca-etc-worked-example.html#assignment"><i class="fa fa-check"></i><b>80.1</b> Assignment</a></li>
<li class="chapter" data-level="80.2" data-path="pca-etc-worked-example.html"><a href="pca-etc-worked-example.html#introduction-8"><i class="fa fa-check"></i><b>80.2</b> Introduction</a></li>
<li class="chapter" data-level="80.3" data-path="pca-etc-worked-example.html"><a href="pca-etc-worked-example.html#data-1"><i class="fa fa-check"></i><b>80.3</b> Data</a></li>
<li class="chapter" data-level="80.4" data-path="pca-etc-worked-example.html"><a href="pca-etc-worked-example.html#scatter-plot"><i class="fa fa-check"></i><b>80.4</b> Scatter plot</a></li>
<li class="chapter" data-level="80.5" data-path="pca-etc-worked-example.html"><a href="pca-etc-worked-example.html#cluster-analysis"><i class="fa fa-check"></i><b>80.5</b> Cluster analysis</a></li>
<li class="chapter" data-level="80.6" data-path="pca-etc-worked-example.html"><a href="pca-etc-worked-example.html#pca-1"><i class="fa fa-check"></i><b>80.6</b> PCA</a></li>
</ul></li>
<li class="chapter" data-level="81" data-path="another-pca-tutorial-with-fishers-iris-data.html"><a href="another-pca-tutorial-with-fishers-iris-data.html"><i class="fa fa-check"></i><b>81</b> Another PCA tutorial with Fisher’s Iris Data</a>
<ul>
<li class="chapter" data-level="81.1" data-path="another-pca-tutorial-with-fishers-iris-data.html"><a href="another-pca-tutorial-with-fishers-iris-data.html#important-functions"><i class="fa fa-check"></i><b>81.1</b> Important functions</a></li>
<li class="chapter" data-level="81.2" data-path="another-pca-tutorial-with-fishers-iris-data.html"><a href="another-pca-tutorial-with-fishers-iris-data.html#useful-packages"><i class="fa fa-check"></i><b>81.2</b> Useful packages</a></li>
<li class="chapter" data-level="81.3" data-path="another-pca-tutorial-with-fishers-iris-data.html"><a href="another-pca-tutorial-with-fishers-iris-data.html#data-fishers-irises"><i class="fa fa-check"></i><b>81.3</b> Data: Fisher’s Irises</a></li>
<li class="chapter" data-level="81.4" data-path="another-pca-tutorial-with-fishers-iris-data.html"><a href="another-pca-tutorial-with-fishers-iris-data.html#data-visualization"><i class="fa fa-check"></i><b>81.4</b> Data visualization</a>
<ul>
<li class="chapter" data-level="81.4.1" data-path="another-pca-tutorial-with-fishers-iris-data.html"><a href="another-pca-tutorial-with-fishers-iris-data.html#d-visualization"><i class="fa fa-check"></i><b>81.4.1</b> 2D visualization</a></li>
</ul></li>
<li class="chapter" data-level="81.5" data-path="another-pca-tutorial-with-fishers-iris-data.html"><a href="another-pca-tutorial-with-fishers-iris-data.html#d-visualziation"><i class="fa fa-check"></i><b>81.5</b> 3D visualziation</a></li>
<li class="chapter" data-level="81.6" data-path="another-pca-tutorial-with-fishers-iris-data.html"><a href="another-pca-tutorial-with-fishers-iris-data.html#beyond-3d-dimensions-with-ordination"><i class="fa fa-check"></i><b>81.6</b> Beyond 3D dimensions with ordination</a></li>
<li class="chapter" data-level="81.7" data-path="another-pca-tutorial-with-fishers-iris-data.html"><a href="another-pca-tutorial-with-fishers-iris-data.html#run-pca"><i class="fa fa-check"></i><b>81.7</b> Run PCA</a>
<ul>
<li class="chapter" data-level="81.7.1" data-path="another-pca-tutorial-with-fishers-iris-data.html"><a href="another-pca-tutorial-with-fishers-iris-data.html#pca-with-prcomp"><i class="fa fa-check"></i><b>81.7.1</b> PCA with prcomp</a></li>
<li class="chapter" data-level="81.7.2" data-path="another-pca-tutorial-with-fishers-iris-data.html"><a href="another-pca-tutorial-with-fishers-iris-data.html#pca-with-princomp"><i class="fa fa-check"></i><b>81.7.2</b> PCA with princomp</a></li>
</ul></li>
<li class="chapter" data-level="81.8" data-path="another-pca-tutorial-with-fishers-iris-data.html"><a href="another-pca-tutorial-with-fishers-iris-data.html#visualize-with-biplot"><i class="fa fa-check"></i><b>81.8</b> Visualize with biplot</a>
<ul>
<li class="chapter" data-level="81.8.1" data-path="another-pca-tutorial-with-fishers-iris-data.html"><a href="another-pca-tutorial-with-fishers-iris-data.html#prcomp-output"><i class="fa fa-check"></i><b>81.8.1</b> prcomp output</a></li>
<li class="chapter" data-level="81.8.2" data-path="another-pca-tutorial-with-fishers-iris-data.html"><a href="another-pca-tutorial-with-fishers-iris-data.html#princomp-output"><i class="fa fa-check"></i><b>81.8.2</b> princomp output</a></li>
</ul></li>
<li class="chapter" data-level="81.9" data-path="another-pca-tutorial-with-fishers-iris-data.html"><a href="another-pca-tutorial-with-fishers-iris-data.html#plotting-by-hand"><i class="fa fa-check"></i><b>81.9</b> Plotting by hand</a>
<ul>
<li class="chapter" data-level="81.9.1" data-path="another-pca-tutorial-with-fishers-iris-data.html"><a href="another-pca-tutorial-with-fishers-iris-data.html#extract-underlying-data-scores"><i class="fa fa-check"></i><b>81.9.1</b> Extract underlying data (“scores”)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="82" data-path="beyond-pc1-and-pc2.html"><a href="beyond-pc1-and-pc2.html"><i class="fa fa-check"></i><b>82</b> Beyond PC1 and PC2</a>
<ul>
<li class="chapter" data-level="82.1" data-path="beyond-pc1-and-pc2.html"><a href="beyond-pc1-and-pc2.html#summary-command"><i class="fa fa-check"></i><b>82.1</b> Summary command</a></li>
<li class="chapter" data-level="82.2" data-path="beyond-pc1-and-pc2.html"><a href="beyond-pc1-and-pc2.html#pca-by-hand"><i class="fa fa-check"></i><b>82.2</b> PCA by hand</a></li>
<li class="chapter" data-level="82.3" data-path="beyond-pc1-and-pc2.html"><a href="beyond-pc1-and-pc2.html#set-up-data"><i class="fa fa-check"></i><b>82.3</b> Set up data</a></li>
<li class="chapter" data-level="82.4" data-path="beyond-pc1-and-pc2.html"><a href="beyond-pc1-and-pc2.html#covariance-vs.-correlation"><i class="fa fa-check"></i><b>82.4</b> covariance vs. correlation</a>
<ul>
<li class="chapter" data-level="82.4.1" data-path="beyond-pc1-and-pc2.html"><a href="beyond-pc1-and-pc2.html#covariance-matrix"><i class="fa fa-check"></i><b>82.4.1</b> covariance matrix</a></li>
<li class="chapter" data-level="82.4.2" data-path="beyond-pc1-and-pc2.html"><a href="beyond-pc1-and-pc2.html#correlation-matrix"><i class="fa fa-check"></i><b>82.4.2</b> correlation matrix</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="83" data-path="eigen-values.html"><a href="eigen-values.html"><i class="fa fa-check"></i><b>83</b> eigen values</a>
<ul>
<li class="chapter" data-level="83.1" data-path="eigen-values.html"><a href="eigen-values.html#calculation-of-the-the-principal-components"><i class="fa fa-check"></i><b>83.1</b> Calculation of the the Principal Components</a></li>
<li class="chapter" data-level="83.2" data-path="eigen-values.html"><a href="eigen-values.html#results-of-matrix-mult-for-pca"><i class="fa fa-check"></i><b>83.2</b> results of matrix mult for PCA</a></li>
</ul></li>
<li class="chapter" data-level="84" data-path="vegan-pca-principal-components-analysis-with-vegans-rda-function.html"><a href="vegan-pca-principal-components-analysis-with-vegans-rda-function.html"><i class="fa fa-check"></i><b>84</b> vegan PCA: Principal Components Analysis with vegans rda function</a></li>
<li class="chapter" data-level="85" data-path="fishers-irises.html"><a href="fishers-irises.html"><i class="fa fa-check"></i><b>85</b> Fisher’s Irises</a></li>
<li class="chapter" data-level="86" data-path="visualize-variables-in-2d.html"><a href="visualize-variables-in-2d.html"><i class="fa fa-check"></i><b>86</b> Visualize variables in 2D</a></li>
<li class="chapter" data-level="87" data-path="do-pca-with-base-r-function.html"><a href="do-pca-with-base-r-function.html"><i class="fa fa-check"></i><b>87</b> Do PCA with base R function</a>
<ul>
<li class="chapter" data-level="87.1" data-path="do-pca-with-base-r-function.html"><a href="do-pca-with-base-r-function.html#run-pca-1"><i class="fa fa-check"></i><b>87.1</b> Run PCA</a></li>
<li class="chapter" data-level="87.2" data-path="do-pca-with-base-r-function.html"><a href="do-pca-with-base-r-function.html#plot-biplot-in-base-graphics"><i class="fa fa-check"></i><b>87.2</b> Plot Biplot in base graphics</a></li>
</ul></li>
<li class="chapter" data-level="88" data-path="do-pca-with-veganrda.html"><a href="do-pca-with-veganrda.html"><i class="fa fa-check"></i><b>88</b> Do PCA with vegan::rda</a>
<ul>
<li class="chapter" data-level="88.1" data-path="do-pca-with-veganrda.html"><a href="do-pca-with-veganrda.html#run-a-vegan-pca-with-rda"><i class="fa fa-check"></i><b>88.1</b> run a vegan PCA with rda()</a></li>
<li class="chapter" data-level="88.2" data-path="do-pca-with-veganrda.html"><a href="do-pca-with-veganrda.html#plot-the-rda"><i class="fa fa-check"></i><b>88.2</b> Plot the RDA</a>
<ul>
<li class="chapter" data-level="88.2.1" data-path="do-pca-with-veganrda.html"><a href="do-pca-with-veganrda.html#default-biplot-of-rda-output"><i class="fa fa-check"></i><b>88.2.1</b> Default biplot of RDA output</a></li>
<li class="chapter" data-level="88.2.2" data-path="do-pca-with-veganrda.html"><a href="do-pca-with-veganrda.html#biplot.rda-options"><i class="fa fa-check"></i><b>88.2.2</b> biplot.rda options</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="89" data-path="vegans-ordination-plotting-functions.html"><a href="vegans-ordination-plotting-functions.html"><i class="fa fa-check"></i><b>89</b> vegan’s ordination plotting functions</a>
<ul>
<li class="chapter" data-level="89.1" data-path="vegans-ordination-plotting-functions.html"><a href="vegans-ordination-plotting-functions.html#add-hulls-around-each-species"><i class="fa fa-check"></i><b>89.1</b> Add “hulls” around each species</a></li>
<li class="chapter" data-level="89.2" data-path="vegans-ordination-plotting-functions.html"><a href="vegans-ordination-plotting-functions.html#make-plot-pretty"><i class="fa fa-check"></i><b>89.2</b> Make plot pretty</a>
<ul>
<li class="chapter" data-level="89.2.1" data-path="vegans-ordination-plotting-functions.html"><a href="vegans-ordination-plotting-functions.html#with-ggplot2"><i class="fa fa-check"></i><b>89.2.1</b> with ggplot2</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="90" data-path="hiearchical-clustering-with-euclidean-distances.html"><a href="hiearchical-clustering-with-euclidean-distances.html"><i class="fa fa-check"></i><b>90</b> Hiearchical clustering with Euclidean distances</a>
<ul>
<li class="chapter" data-level="90.1" data-path="hiearchical-clustering-with-euclidean-distances.html"><a href="hiearchical-clustering-with-euclidean-distances.html#preliminaries-15"><i class="fa fa-check"></i><b>90.1</b> Preliminaries</a>
<ul>
<li class="chapter" data-level="90.1.1" data-path="hiearchical-clustering-with-euclidean-distances.html"><a href="hiearchical-clustering-with-euclidean-distances.html#download-packages-1"><i class="fa fa-check"></i><b>90.1.1</b> Download packages</a></li>
<li class="chapter" data-level="90.1.2" data-path="hiearchical-clustering-with-euclidean-distances.html"><a href="hiearchical-clustering-with-euclidean-distances.html#install-packages"><i class="fa fa-check"></i><b>90.1.2</b> Install packages</a></li>
</ul></li>
<li class="chapter" data-level="90.2" data-path="hiearchical-clustering-with-euclidean-distances.html"><a href="hiearchical-clustering-with-euclidean-distances.html#vocab-3"><i class="fa fa-check"></i><b>90.2</b> Vocab</a></li>
<li class="chapter" data-level="90.3" data-path="hiearchical-clustering-with-euclidean-distances.html"><a href="hiearchical-clustering-with-euclidean-distances.html#note-this-is-not-a-real-analysis"><i class="fa fa-check"></i><b>90.3</b> Note: This is NOT a real analysis</a></li>
<li class="chapter" data-level="90.4" data-path="hiearchical-clustering-with-euclidean-distances.html"><a href="hiearchical-clustering-with-euclidean-distances.html#load-data-7"><i class="fa fa-check"></i><b>90.4</b> Load data</a></li>
<li class="chapter" data-level="90.5" data-path="hiearchical-clustering-with-euclidean-distances.html"><a href="hiearchical-clustering-with-euclidean-distances.html#plot-the-data-1"><i class="fa fa-check"></i><b>90.5</b> Plot the data</a></li>
<li class="chapter" data-level="90.6" data-path="hiearchical-clustering-with-euclidean-distances.html"><a href="hiearchical-clustering-with-euclidean-distances.html#distances-and-distance-matrices"><i class="fa fa-check"></i><b>90.6</b> Distances and distance matrices</a></li>
<li class="chapter" data-level="90.7" data-path="hiearchical-clustering-with-euclidean-distances.html"><a href="hiearchical-clustering-with-euclidean-distances.html#calculate-euclidean-distance"><i class="fa fa-check"></i><b>90.7</b> Calculate Euclidean distance</a>
<ul>
<li class="chapter" data-level="90.7.1" data-path="hiearchical-clustering-with-euclidean-distances.html"><a href="hiearchical-clustering-with-euclidean-distances.html#distance-from-vivax-to-knowlesi"><i class="fa fa-check"></i><b>90.7.1</b> Distance from vivax to knowlesi</a></li>
<li class="chapter" data-level="90.7.2" data-path="hiearchical-clustering-with-euclidean-distances.html"><a href="hiearchical-clustering-with-euclidean-distances.html#distance-from-falciparum-to-yoelli"><i class="fa fa-check"></i><b>90.7.2</b> Distance from falciparum to yoelli</a></li>
</ul></li>
<li class="chapter" data-level="90.8" data-path="hiearchical-clustering-with-euclidean-distances.html"><a href="hiearchical-clustering-with-euclidean-distances.html#distance-knowlesi-to-yoelli"><i class="fa fa-check"></i><b>90.8</b> Distance knowlesi to yoelli</a></li>
<li class="chapter" data-level="90.9" data-path="hiearchical-clustering-with-euclidean-distances.html"><a href="hiearchical-clustering-with-euclidean-distances.html#distances-with-the-dist-function"><i class="fa fa-check"></i><b>90.9</b> Distances with the dist() function</a></li>
<li class="chapter" data-level="90.10" data-path="hiearchical-clustering-with-euclidean-distances.html"><a href="hiearchical-clustering-with-euclidean-distances.html#forming-the-first-cluster"><i class="fa fa-check"></i><b>90.10</b> Forming the first cluster</a></li>
</ul></li>
<li class="chapter" data-level="91" data-path="cluster-analysis-with-allometric-data.html"><a href="cluster-analysis-with-allometric-data.html"><i class="fa fa-check"></i><b>91</b> Cluster analysis with allometric data</a>
<ul>
<li class="chapter" data-level="91.1" data-path="cluster-analysis-with-allometric-data.html"><a href="cluster-analysis-with-allometric-data.html#cluster-analysis-1"><i class="fa fa-check"></i><b>91.1</b> Cluster analysis</a></li>
<li class="chapter" data-level="91.2" data-path="cluster-analysis-with-allometric-data.html"><a href="cluster-analysis-with-allometric-data.html#preliminaries-16"><i class="fa fa-check"></i><b>91.2</b> Preliminaries</a>
<ul>
<li class="chapter" data-level="91.2.1" data-path="cluster-analysis-with-allometric-data.html"><a href="cluster-analysis-with-allometric-data.html#download-packages-2"><i class="fa fa-check"></i><b>91.2.1</b> Download packages</a></li>
<li class="chapter" data-level="91.2.2" data-path="cluster-analysis-with-allometric-data.html"><a href="cluster-analysis-with-allometric-data.html#load-the-libraries-4"><i class="fa fa-check"></i><b>91.2.2</b> Load the libraries</a></li>
</ul></li>
<li class="chapter" data-level="91.3" data-path="cluster-analysis-with-allometric-data.html"><a href="cluster-analysis-with-allometric-data.html#pca-data-exploration"><i class="fa fa-check"></i><b>91.3</b> PCA data exploration</a>
<ul>
<li class="chapter" data-level="91.3.1" data-path="cluster-analysis-with-allometric-data.html"><a href="cluster-analysis-with-allometric-data.html#hierachical-cluster-analysis"><i class="fa fa-check"></i><b>91.3.1</b> Hierachical cluster analysis</a></li>
</ul></li>
<li class="chapter" data-level="91.4" data-path="cluster-analysis-with-allometric-data.html"><a href="cluster-analysis-with-allometric-data.html#task-5"><i class="fa fa-check"></i><b>91.4</b> Task</a></li>
</ul></li>
<li class="chapter" data-level="92" data-path="higgs-and-atwood-student-version-with-updates.html"><a href="higgs-and-atwood-student-version-with-updates.html"><i class="fa fa-check"></i><b>92</b> Higgs and Atwood student version with updates</a>
<ul>
<li class="chapter" data-level="92.1" data-path="higgs-and-atwood-student-version-with-updates.html"><a href="higgs-and-atwood-student-version-with-updates.html#introduction-9"><i class="fa fa-check"></i><b>92.1</b> Introduction</a></li>
<li class="chapter" data-level="92.2" data-path="higgs-and-atwood-student-version-with-updates.html"><a href="higgs-and-atwood-student-version-with-updates.html#preliminaries-17"><i class="fa fa-check"></i><b>92.2</b> Preliminaries</a>
<ul>
<li class="chapter" data-level="92.2.1" data-path="higgs-and-atwood-student-version-with-updates.html"><a href="higgs-and-atwood-student-version-with-updates.html#packages-5"><i class="fa fa-check"></i><b>92.2.1</b> Packages</a></li>
<li class="chapter" data-level="92.2.2" data-path="higgs-and-atwood-student-version-with-updates.html"><a href="higgs-and-atwood-student-version-with-updates.html#build-the-dataframe-1"><i class="fa fa-check"></i><b>92.2.2</b> Build the dataframe</a></li>
</ul></li>
<li class="chapter" data-level="92.3" data-path="higgs-and-atwood-student-version-with-updates.html"><a href="higgs-and-atwood-student-version-with-updates.html#raw-data-exploration"><i class="fa fa-check"></i><b>92.3</b> Raw data exploration</a>
<ul>
<li class="chapter" data-level="92.3.1" data-path="higgs-and-atwood-student-version-with-updates.html"><a href="higgs-and-atwood-student-version-with-updates.html#xxxxxxxx"><i class="fa fa-check"></i><b>92.3.1</b> XXXXXXXX</a></li>
<li class="chapter" data-level="92.3.2" data-path="higgs-and-atwood-student-version-with-updates.html"><a href="higgs-and-atwood-student-version-with-updates.html#plot-0-scatterplot-matrix"><i class="fa fa-check"></i><b>92.3.2</b> Plot 0: Scatterplot matrix</a></li>
<li class="chapter" data-level="92.3.3" data-path="higgs-and-atwood-student-version-with-updates.html"><a href="higgs-and-atwood-student-version-with-updates.html#plot-1-replication-higgs-and-attwood-figure-2.8"><i class="fa fa-check"></i><b>92.3.3</b> Plot 1: Replication Higgs and Attwood Figure 2.8</a></li>
<li class="chapter" data-level="92.3.4" data-path="higgs-and-atwood-student-version-with-updates.html"><a href="higgs-and-atwood-student-version-with-updates.html#figure-2-ha-figure-2.8-with-ggpubr"><i class="fa fa-check"></i><b>92.3.4</b> Figure 2: HA Figure 2.8 with ggpubr</a></li>
<li class="chapter" data-level="92.3.5" data-path="higgs-and-atwood-student-version-with-updates.html"><a href="higgs-and-atwood-student-version-with-updates.html#figure-3-highly-correlated-data"><i class="fa fa-check"></i><b>92.3.5</b> Figure 3: Highly correlated data</a></li>
<li class="chapter" data-level="92.3.6" data-path="higgs-and-atwood-student-version-with-updates.html"><a href="higgs-and-atwood-student-version-with-updates.html#figure-4-non-linear-relationship"><i class="fa fa-check"></i><b>92.3.6</b> Figure 4: Non-linear relationship</a></li>
<li class="chapter" data-level="92.3.7" data-path="higgs-and-atwood-student-version-with-updates.html"><a href="higgs-and-atwood-student-version-with-updates.html#figure-5-non-linear-relationship-on-log-scale"><i class="fa fa-check"></i><b>92.3.7</b> Figure 5: Non-linear relationship on LOG scale</a></li>
<li class="chapter" data-level="92.3.8" data-path="higgs-and-atwood-student-version-with-updates.html"><a href="higgs-and-atwood-student-version-with-updates.html#figure-6-3d-scatterplot"><i class="fa fa-check"></i><b>92.3.8</b> Figure 6: 3D Scatterplot</a></li>
</ul></li>
<li class="chapter" data-level="92.4" data-path="higgs-and-atwood-student-version-with-updates.html"><a href="higgs-and-atwood-student-version-with-updates.html#principal-component-analysis"><i class="fa fa-check"></i><b>92.4</b> Principal component analysis</a>
<ul>
<li class="chapter" data-level="92.4.1" data-path="higgs-and-atwood-student-version-with-updates.html"><a href="higgs-and-atwood-student-version-with-updates.html#data-prep"><i class="fa fa-check"></i><b>92.4.1</b> Data prep</a></li>
<li class="chapter" data-level="92.4.2" data-path="higgs-and-atwood-student-version-with-updates.html"><a href="higgs-and-atwood-student-version-with-updates.html#figure-7-principal-components-analysis---base-r"><i class="fa fa-check"></i><b>92.4.2</b> Figure 7: Principal components analysis - base R</a></li>
<li class="chapter" data-level="92.4.3" data-path="higgs-and-atwood-student-version-with-updates.html"><a href="higgs-and-atwood-student-version-with-updates.html#principal-components-analysis-pca"><i class="fa fa-check"></i><b>92.4.3</b> Principal Components Analysis (PCA)</a></li>
</ul></li>
<li class="chapter" data-level="92.5" data-path="higgs-and-atwood-student-version-with-updates.html"><a href="higgs-and-atwood-student-version-with-updates.html#cluster-analysis-2"><i class="fa fa-check"></i><b>92.5</b> Cluster analysis</a>
<ul>
<li class="chapter" data-level="92.5.1" data-path="higgs-and-atwood-student-version-with-updates.html"><a href="higgs-and-atwood-student-version-with-updates.html#xxxxxxxxxxxxxxx-cluster-analysis"><i class="fa fa-check"></i><b>92.5.1</b> XXXXXXXXXXXXXXX cluster analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="93" data-path="higgs-and-atwood-student-version-with-updates-copy-or-alt.html"><a href="higgs-and-atwood-student-version-with-updates-copy-or-alt.html"><i class="fa fa-check"></i><b>93</b> Higgs and Atwood student version with updates - copy or alt</a>
<ul>
<li class="chapter" data-level="93.1" data-path="higgs-and-atwood-student-version-with-updates-copy-or-alt.html"><a href="higgs-and-atwood-student-version-with-updates-copy-or-alt.html#introduction-10"><i class="fa fa-check"></i><b>93.1</b> Introduction</a></li>
<li class="chapter" data-level="93.2" data-path="higgs-and-atwood-student-version-with-updates-copy-or-alt.html"><a href="higgs-and-atwood-student-version-with-updates-copy-or-alt.html#preliminaries-18"><i class="fa fa-check"></i><b>93.2</b> Preliminaries</a>
<ul>
<li class="chapter" data-level="93.2.1" data-path="higgs-and-atwood-student-version-with-updates-copy-or-alt.html"><a href="higgs-and-atwood-student-version-with-updates-copy-or-alt.html#packages-6"><i class="fa fa-check"></i><b>93.2.1</b> Packages</a></li>
<li class="chapter" data-level="93.2.2" data-path="higgs-and-atwood-student-version-with-updates-copy-or-alt.html"><a href="higgs-and-atwood-student-version-with-updates-copy-or-alt.html#build-the-dataframe-2"><i class="fa fa-check"></i><b>93.2.2</b> Build the dataframe</a></li>
</ul></li>
<li class="chapter" data-level="93.3" data-path="higgs-and-atwood-student-version-with-updates-copy-or-alt.html"><a href="higgs-and-atwood-student-version-with-updates-copy-or-alt.html#raw-data-exploration-1"><i class="fa fa-check"></i><b>93.3</b> Raw data exploration</a>
<ul>
<li class="chapter" data-level="93.3.1" data-path="higgs-and-atwood-student-version-with-updates-copy-or-alt.html"><a href="higgs-and-atwood-student-version-with-updates-copy-or-alt.html#xxxxxxxx-1"><i class="fa fa-check"></i><b>93.3.1</b> XXXXXXXX</a></li>
<li class="chapter" data-level="93.3.2" data-path="higgs-and-atwood-student-version-with-updates-copy-or-alt.html"><a href="higgs-and-atwood-student-version-with-updates-copy-or-alt.html#plot-0-scatterplot-matrix-1"><i class="fa fa-check"></i><b>93.3.2</b> Plot 0: Scatterplot matrix</a></li>
<li class="chapter" data-level="93.3.3" data-path="higgs-and-atwood-student-version-with-updates-copy-or-alt.html"><a href="higgs-and-atwood-student-version-with-updates-copy-or-alt.html#plot-1-replication-higgs-and-attwood-figure-2.8-1"><i class="fa fa-check"></i><b>93.3.3</b> Plot 1: Replication Higgs and Attwood Figure 2.8</a></li>
<li class="chapter" data-level="93.3.4" data-path="higgs-and-atwood-student-version-with-updates-copy-or-alt.html"><a href="higgs-and-atwood-student-version-with-updates-copy-or-alt.html#figure-2-ha-figure-2.8-with-ggpubr-1"><i class="fa fa-check"></i><b>93.3.4</b> Figure 2: HA Figure 2.8 with ggpubr</a></li>
<li class="chapter" data-level="93.3.5" data-path="higgs-and-atwood-student-version-with-updates-copy-or-alt.html"><a href="higgs-and-atwood-student-version-with-updates-copy-or-alt.html#figure-3-highly-correlated-data-1"><i class="fa fa-check"></i><b>93.3.5</b> Figure 3: Highly correlated data</a></li>
<li class="chapter" data-level="93.3.6" data-path="higgs-and-atwood-student-version-with-updates-copy-or-alt.html"><a href="higgs-and-atwood-student-version-with-updates-copy-or-alt.html#figure-4-non-linear-relationship-1"><i class="fa fa-check"></i><b>93.3.6</b> Figure 4: Non-linear relationship</a></li>
<li class="chapter" data-level="93.3.7" data-path="higgs-and-atwood-student-version-with-updates-copy-or-alt.html"><a href="higgs-and-atwood-student-version-with-updates-copy-or-alt.html#figure-5-non-linear-relationship-on-log-scale-1"><i class="fa fa-check"></i><b>93.3.7</b> Figure 5: Non-linear relationship on LOG scale</a></li>
<li class="chapter" data-level="93.3.8" data-path="higgs-and-atwood-student-version-with-updates-copy-or-alt.html"><a href="higgs-and-atwood-student-version-with-updates-copy-or-alt.html#figure-6-3d-scatterplot-1"><i class="fa fa-check"></i><b>93.3.8</b> Figure 6: 3D Scatterplot</a></li>
</ul></li>
<li class="chapter" data-level="93.4" data-path="higgs-and-atwood-student-version-with-updates-copy-or-alt.html"><a href="higgs-and-atwood-student-version-with-updates-copy-or-alt.html#principal-component-analysis-1"><i class="fa fa-check"></i><b>93.4</b> Principal component analysis</a>
<ul>
<li class="chapter" data-level="93.4.1" data-path="higgs-and-atwood-student-version-with-updates-copy-or-alt.html"><a href="higgs-and-atwood-student-version-with-updates-copy-or-alt.html#data-prep-1"><i class="fa fa-check"></i><b>93.4.1</b> Data prep</a></li>
<li class="chapter" data-level="93.4.2" data-path="higgs-and-atwood-student-version-with-updates-copy-or-alt.html"><a href="higgs-and-atwood-student-version-with-updates-copy-or-alt.html#figure-7-principal-components-analysis---base-r-1"><i class="fa fa-check"></i><b>93.4.2</b> Figure 7: Principal components analysis - base R</a></li>
<li class="chapter" data-level="93.4.3" data-path="higgs-and-atwood-student-version-with-updates-copy-or-alt.html"><a href="higgs-and-atwood-student-version-with-updates-copy-or-alt.html#principal-components-analysis-pca-1"><i class="fa fa-check"></i><b>93.4.3</b> Principal Components Analysis (PCA)</a></li>
</ul></li>
<li class="chapter" data-level="93.5" data-path="higgs-and-atwood-student-version-with-updates-copy-or-alt.html"><a href="higgs-and-atwood-student-version-with-updates-copy-or-alt.html#cluster-analysis-3"><i class="fa fa-check"></i><b>93.5</b> Cluster analysis</a>
<ul>
<li class="chapter" data-level="93.5.1" data-path="higgs-and-atwood-student-version-with-updates-copy-or-alt.html"><a href="higgs-and-atwood-student-version-with-updates-copy-or-alt.html#xxxxxxxxxxxxxxx-cluster-analysis-1"><i class="fa fa-check"></i><b>93.5.1</b> XXXXXXXXXXXXXXX cluster analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="94" data-path="allometry-data.html"><a href="allometry-data.html"><i class="fa fa-check"></i><b>94</b> Allometry data</a></li>
<li class="chapter" data-level="95" data-path="errorplot-datasets.html"><a href="errorplot-datasets.html"><i class="fa fa-check"></i><b>95</b> Errorplot datasets</a></li>
<li class="chapter" data-level="96" data-path="error-plot.html"><a href="error-plot.html"><i class="fa fa-check"></i><b>96</b> error plot</a>
<ul>
<li class="chapter" data-level="96.1" data-path="error-plot.html"><a href="error-plot.html#errorplot"><i class="fa fa-check"></i><b>96.1</b> errorplot</a></li>
<li class="chapter" data-level="96.2" data-path="error-plot.html"><a href="error-plot.html#health-related"><i class="fa fa-check"></i><b>96.2</b> Health-related</a></li>
</ul></li>
<li class="chapter" data-level="97" data-path="datasets.html"><a href="datasets.html"><i class="fa fa-check"></i><b>97</b> Datasets</a>
<ul>
<li class="chapter" data-level="97.1" data-path="datasets.html"><a href="datasets.html#non-linear-curve"><i class="fa fa-check"></i><b>97.1</b> Non-linear curve</a></li>
<li class="chapter" data-level="97.2" data-path="datasets.html"><a href="datasets.html#noisy-correlation"><i class="fa fa-check"></i><b>97.2</b> Noisy correlation</a></li>
<li class="chapter" data-level="97.3" data-path="datasets.html"><a href="datasets.html#regression-quantile-regression-climate-change"><i class="fa fa-check"></i><b>97.3</b> Regression; quantile regression ; climate change</a></li>
<li class="chapter" data-level="97.4" data-path="datasets.html"><a href="datasets.html#logistic-regression"><i class="fa fa-check"></i><b>97.4</b> Logistic regression</a></li>
<li class="chapter" data-level="97.5" data-path="datasets.html"><a href="datasets.html#regression-fitness"><i class="fa fa-check"></i><b>97.5</b> Regression; Fitness</a></li>
</ul></li>
<li class="chapter" data-level="98" data-path="x-2-table.html"><a href="x-2-table.html"><i class="fa fa-check"></i><b>98</b> 2 x 2 table</a>
<ul>
<li class="chapter" data-level="98.1" data-path="x-2-table.html"><a href="x-2-table.html#heritablity"><i class="fa fa-check"></i><b>98.1</b> Heritablity</a></li>
<li class="chapter" data-level="98.2" data-path="x-2-table.html"><a href="x-2-table.html#mendelian-genetics"><i class="fa fa-check"></i><b>98.2</b> Mendelian genetics</a></li>
<li class="chapter" data-level="98.3" data-path="x-2-table.html"><a href="x-2-table.html#genomics"><i class="fa fa-check"></i><b>98.3</b> Genomics</a></li>
<li class="chapter" data-level="98.4" data-path="x-2-table.html"><a href="x-2-table.html#time-series-data-library"><i class="fa fa-check"></i><b>98.4</b> Time series data library</a></li>
<li class="chapter" data-level="98.5" data-path="x-2-table.html"><a href="x-2-table.html#data-sets-1"><i class="fa fa-check"></i><b>98.5</b> Data sets</a></li>
</ul></li>
<li class="chapter" data-level="99" data-path="multivariate.html"><a href="multivariate.html"><i class="fa fa-check"></i><b>99</b> Multivariate</a></li>
<li class="chapter" data-level="100" data-path="datasets-regression.html"><a href="datasets-regression.html"><i class="fa fa-check"></i><b>100</b> Datasets - regression</a>
<ul>
<li class="chapter" data-level="100.1" data-path="datasets-regression.html"><a href="datasets-regression.html#repated-measures"><i class="fa fa-check"></i><b>100.1</b> Repated measures</a></li>
<li class="chapter" data-level="100.2" data-path="datasets-regression.html"><a href="datasets-regression.html#multiple-regression"><i class="fa fa-check"></i><b>100.2</b> Multiple regression</a></li>
<li class="chapter" data-level="100.3" data-path="datasets-regression.html"><a href="datasets-regression.html#glm"><i class="fa fa-check"></i><b>100.3</b> GLM</a></li>
</ul></li>
<li class="chapter" data-level="101" data-path="regression.html"><a href="regression.html"><i class="fa fa-check"></i><b>101</b> Regression</a></li>
<li class="chapter" data-level="102" data-path="time-series-climate-changes.html"><a href="time-series-climate-changes.html"><i class="fa fa-check"></i><b>102</b> Time series / climate changes</a>
<ul>
<li class="chapter" data-level="102.0.1" data-path="time-series-climate-changes.html"><a href="time-series-climate-changes.html#stat2data"><i class="fa fa-check"></i><b>102.0.1</b> Stat2Data</a></li>
<li class="chapter" data-level="102.0.2" data-path="time-series-climate-changes.html"><a href="time-series-climate-changes.html#fpp2"><i class="fa fa-check"></i><b>102.0.2</b> fpp2</a></li>
<li class="chapter" data-level="102.0.3" data-path="time-series-climate-changes.html"><a href="time-series-climate-changes.html#package-astsa-applied-statistical-time-series-analysis"><i class="fa fa-check"></i><b>102.0.3</b> Package astsa: Applied Statistical Time Series Analysis</a></li>
</ul></li>
<li class="chapter" data-level="103" data-path="basic-statistics-equations.html"><a href="basic-statistics-equations.html"><i class="fa fa-check"></i><b>103</b> Basic statistics equations</a>
<ul>
<li class="chapter" data-level="103.1" data-path="basic-statistics-equations.html"><a href="basic-statistics-equations.html#variance"><i class="fa fa-check"></i><b>103.1</b> Variance</a></li>
<li class="chapter" data-level="103.2" data-path="basic-statistics-equations.html"><a href="basic-statistics-equations.html#standard-deviation"><i class="fa fa-check"></i><b>103.2</b> Standard deviation</a></li>
<li class="chapter" data-level="103.3" data-path="basic-statistics-equations.html"><a href="basic-statistics-equations.html#standard-error"><i class="fa fa-check"></i><b>103.3</b> Standard error</a>
<ul>
<li class="chapter" data-level="103.3.1" data-path="basic-statistics-equations.html"><a href="basic-statistics-equations.html#equation"><i class="fa fa-check"></i><b>103.3.1</b> Equation</a></li>
<li class="chapter" data-level="103.3.2" data-path="basic-statistics-equations.html"><a href="basic-statistics-equations.html#r-ish-notation"><i class="fa fa-check"></i><b>103.3.2</b> R-ish notation</a></li>
</ul></li>
<li class="chapter" data-level="103.4" data-path="basic-statistics-equations.html"><a href="basic-statistics-equations.html#pooled-variance-for-two-sample-t-test"><i class="fa fa-check"></i><b>103.4</b> Pooled variance for two-sample t-test</a></li>
<li class="chapter" data-level="103.5" data-path="basic-statistics-equations.html"><a href="basic-statistics-equations.html#standard-error-of-the-difference"><i class="fa fa-check"></i><b>103.5</b> Standard error of the difference</a></li>
<li class="chapter" data-level="103.6" data-path="basic-statistics-equations.html"><a href="basic-statistics-equations.html#ls-regression-slope"><i class="fa fa-check"></i><b>103.6</b> LS regression slope</a>
<ul>
<li class="chapter" data-level="103.6.1" data-path="basic-statistics-equations.html"><a href="basic-statistics-equations.html#words"><i class="fa fa-check"></i><b>103.6.1</b> Words</a></li>
<li class="chapter" data-level="103.6.2" data-path="basic-statistics-equations.html"><a href="basic-statistics-equations.html#math"><i class="fa fa-check"></i><b>103.6.2</b> Math</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="104" data-path="regression-intercept.html"><a href="regression-intercept.html"><i class="fa fa-check"></i><b>104</b> Regression intercept</a>
<ul>
<li class="chapter" data-level="104.1" data-path="regression-intercept.html"><a href="regression-intercept.html#word-equation"><i class="fa fa-check"></i><b>104.1</b> Word equation</a></li>
<li class="chapter" data-level="104.2" data-path="regression-intercept.html"><a href="regression-intercept.html#math-equation"><i class="fa fa-check"></i><b>104.2</b> Math equation</a></li>
<li class="chapter" data-level="104.3" data-path="regression-intercept.html"><a href="regression-intercept.html#do-some-algebra-to-solve-for-the-intercept-a"><i class="fa fa-check"></i><b>104.3</b> Do some algebra to solve for the intercept a</a></li>
</ul></li>
<li class="chapter" data-level="105" data-path="f-statistics-f-ratio.html"><a href="f-statistics-f-ratio.html"><i class="fa fa-check"></i><b>105</b> F statistics (F ratio)</a>
<ul>
<li class="chapter" data-level="105.1" data-path="f-statistics-f-ratio.html"><a href="f-statistics-f-ratio.html#chi2-test"><i class="fa fa-check"></i><b>105.1</b> Chi^2 test</a></li>
<li class="chapter" data-level="105.2" data-path="f-statistics-f-ratio.html"><a href="f-statistics-f-ratio.html#power"><i class="fa fa-check"></i><b>105.2</b> Power</a></li>
</ul></li>
<li class="chapter" data-level="106" data-path="two-equivalent-ways-to-set-up-a-paired-t-test-in-r.html"><a href="two-equivalent-ways-to-set-up-a-paired-t-test-in-r.html"><i class="fa fa-check"></i><b>106</b> Two equivalent ways to set up a paired t-test in R</a></li>
<li class="chapter" data-level="107" data-path="paired-t-test-on-2-column-data.html"><a href="paired-t-test-on-2-column-data.html"><i class="fa fa-check"></i><b>107</b> Paired T-test on 2 column data</a></li>
<li class="chapter" data-level="108" data-path="paired-t-test-on-one-column-of-data.html"><a href="paired-t-test-on-one-column-of-data.html"><i class="fa fa-check"></i><b>108</b> Paired t-test on one column of data</a></li>
<li class="chapter" data-level="109" data-path="running-reporting-paired-t-tests-in-r.html"><a href="running-reporting-paired-t-tests-in-r.html"><i class="fa fa-check"></i><b>109</b> Running &amp; reporting paired t-tests in R</a>
<ul>
<li class="chapter" data-level="109.1" data-path="running-reporting-paired-t-tests-in-r.html"><a href="running-reporting-paired-t-tests-in-r.html#create-dataframe"><i class="fa fa-check"></i><b>109.1</b> Create dataframe</a></li>
<li class="chapter" data-level="109.2" data-path="running-reporting-paired-t-tests-in-r.html"><a href="running-reporting-paired-t-tests-in-r.html#paired-t-test"><i class="fa fa-check"></i><b>109.2</b> Paired t-test</a>
<ul>
<li class="chapter" data-level="109.2.1" data-path="running-reporting-paired-t-tests-in-r.html"><a href="running-reporting-paired-t-tests-in-r.html#paired-t-test-version-1"><i class="fa fa-check"></i><b>109.2.1</b> Paired t-test, Version 1</a></li>
<li class="chapter" data-level="109.2.2" data-path="running-reporting-paired-t-tests-in-r.html"><a href="running-reporting-paired-t-tests-in-r.html#paired-t-test-version-2"><i class="fa fa-check"></i><b>109.2.2</b> Paired t-test, Version 2</a></li>
</ul></li>
<li class="chapter" data-level="109.3" data-path="running-reporting-paired-t-tests-in-r.html"><a href="running-reporting-paired-t-tests-in-r.html#reporting-the-results-of-a-paired-t-test"><i class="fa fa-check"></i><b>109.3</b> Reporting the results of a paired t-test</a></li>
</ul></li>
<li class="chapter" data-level="110" data-path="reporting-statistical-result-from-a-2-sample-t-test.html"><a href="reporting-statistical-result-from-a-2-sample-t-test.html"><i class="fa fa-check"></i><b>110</b> Reporting statistical result from a 2-sample t-test</a>
<ul>
<li class="chapter" data-level="110.1" data-path="reporting-statistical-result-from-a-2-sample-t-test.html"><a href="reporting-statistical-result-from-a-2-sample-t-test.html#introduction-11"><i class="fa fa-check"></i><b>110.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="110.1.1" data-path="reporting-statistical-result-from-a-2-sample-t-test.html"><a href="reporting-statistical-result-from-a-2-sample-t-test.html#references-3"><i class="fa fa-check"></i><b>110.1.1</b> References</a></li>
</ul></li>
<li class="chapter" data-level="110.2" data-path="reporting-statistical-result-from-a-2-sample-t-test.html"><a href="reporting-statistical-result-from-a-2-sample-t-test.html#outline-of-tasks"><i class="fa fa-check"></i><b>110.2</b> Outline of tasks</a></li>
<li class="chapter" data-level="110.3" data-path="reporting-statistical-result-from-a-2-sample-t-test.html"><a href="reporting-statistical-result-from-a-2-sample-t-test.html#dataset-up"><i class="fa fa-check"></i><b>110.3</b> Dataset up</a></li>
<li class="chapter" data-level="110.4" data-path="reporting-statistical-result-from-a-2-sample-t-test.html"><a href="reporting-statistical-result-from-a-2-sample-t-test.html#plot-raw-data"><i class="fa fa-check"></i><b>110.4</b> Plot raw data</a></li>
<li class="chapter" data-level="110.5" data-path="reporting-statistical-result-from-a-2-sample-t-test.html"><a href="reporting-statistical-result-from-a-2-sample-t-test.html#the-hypotheses"><i class="fa fa-check"></i><b>110.5</b> The hypotheses</a></li>
<li class="chapter" data-level="110.6" data-path="reporting-statistical-result-from-a-2-sample-t-test.html"><a href="reporting-statistical-result-from-a-2-sample-t-test.html#do-t-test"><i class="fa fa-check"></i><b>110.6</b> Do t-test</a></li>
<li class="chapter" data-level="110.7" data-path="reporting-statistical-result-from-a-2-sample-t-test.html"><a href="reporting-statistical-result-from-a-2-sample-t-test.html#report-the-results"><i class="fa fa-check"></i><b>110.7</b> Report the results</a>
<ul>
<li class="chapter" data-level="110.7.1" data-path="reporting-statistical-result-from-a-2-sample-t-test.html"><a href="reporting-statistical-result-from-a-2-sample-t-test.html#alternative-results"><i class="fa fa-check"></i><b>110.7.1</b> Alternative results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="111" data-path="standard-error-of-the-difference-between-means.html"><a href="standard-error-of-the-difference-between-means.html"><i class="fa fa-check"></i><b>111</b> Standard error of the difference between means</a></li>
<li class="chapter" data-level="112" data-path="standard-error-for-the-difference-between-2-means.html"><a href="standard-error-for-the-difference-between-2-means.html"><i class="fa fa-check"></i><b>112</b> Standard error for the difference between 2 means</a></li>
<li class="chapter" data-level="113" data-path="exmple-data-rat-bladders.html"><a href="exmple-data-rat-bladders.html"><i class="fa fa-check"></i><b>113</b> Exmple data: rat bladders</a></li>
<li class="chapter" data-level="114" data-path="make-data-table-calculate-summary-stats.html"><a href="make-data-table-calculate-summary-stats.html"><i class="fa fa-check"></i><b>114</b> 1) Make data table &amp; calculate summary stats</a>
<ul>
<li class="chapter" data-level="114.1" data-path="make-data-table-calculate-summary-stats.html"><a href="make-data-table-calculate-summary-stats.html#the-data-1"><i class="fa fa-check"></i><b>114.1</b> The Data</a></li>
<li class="chapter" data-level="114.2" data-path="make-data-table-calculate-summary-stats.html"><a href="make-data-table-calculate-summary-stats.html#create-table-of-data-and-summary-stats"><i class="fa fa-check"></i><b>114.2</b> Create table of data and summary stats</a></li>
<li class="chapter" data-level="114.3" data-path="make-data-table-calculate-summary-stats.html"><a href="make-data-table-calculate-summary-stats.html#the-data-table"><i class="fa fa-check"></i><b>114.3</b> The Data table</a></li>
</ul></li>
<li class="chapter" data-level="115" data-path="function-to-calculate-the-se-of-the-difference.html"><a href="function-to-calculate-the-se-of-the-difference.html"><i class="fa fa-check"></i><b>115</b> 2) Function to calculate the SE of the difference</a>
<ul>
<li class="chapter" data-level="115.1" data-path="function-to-calculate-the-se-of-the-difference.html"><a href="function-to-calculate-the-se-of-the-difference.html#pooled-variance"><i class="fa fa-check"></i><b>115.1</b> Pooled variance</a>
<ul>
<li class="chapter" data-level="115.1.1" data-path="function-to-calculate-the-se-of-the-difference.html"><a href="function-to-calculate-the-se-of-the-difference.html#equation-for-pooled-variance"><i class="fa fa-check"></i><b>115.1.1</b> Equation for pooled variance</a></li>
<li class="chapter" data-level="115.1.2" data-path="function-to-calculate-the-se-of-the-difference.html"><a href="function-to-calculate-the-se-of-the-difference.html#an-r-function-to-calcualte-the-pooled-variance"><i class="fa fa-check"></i><b>115.1.2</b> An R function to calcualte the pooled variance</a></li>
<li class="chapter" data-level="115.1.3" data-path="function-to-calculate-the-se-of-the-difference.html"><a href="function-to-calculate-the-se-of-the-difference.html#calcualte-pooled-variance-using-function"><i class="fa fa-check"></i><b>115.1.3</b> Calcualte pooled variance using function</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="116" data-path="standard-error-of-the-difference-1.html"><a href="standard-error-of-the-difference-1.html"><i class="fa fa-check"></i><b>116</b> Standard error of the difference</a>
<ul>
<li class="chapter" data-level="116.1" data-path="standard-error-of-the-difference-1.html"><a href="standard-error-of-the-difference-1.html#equation-for-se-of-difference"><i class="fa fa-check"></i><b>116.1</b> Equation for SE of difference</a></li>
<li class="chapter" data-level="116.2" data-path="standard-error-of-the-difference-1.html"><a href="standard-error-of-the-difference-1.html#function-to-calcualte-se-of-difference"><i class="fa fa-check"></i><b>116.2</b> Function to calcualte SE of difference</a></li>
<li class="chapter" data-level="116.3" data-path="standard-error-of-the-difference-1.html"><a href="standard-error-of-the-difference-1.html#apply-function-se.diff"><i class="fa fa-check"></i><b>116.3</b> Apply function SE.diff</a></li>
</ul></li>
<li class="chapter" data-level="117" data-path="final-dataframe.html"><a href="final-dataframe.html"><i class="fa fa-check"></i><b>117</b> Final dataframe</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Computational Biology for All!</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="assignment-building-dataframes-by-hand" class="section level1" number="20">
<h1><span class="header-section-number">Chapter 20</span> Assignment: Building dataframes by hand</h1>
<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb194-1"><a href="assignment-building-dataframes-by-hand.html#cb194-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(compbio4all)</span></code></pre></div>
<div id="introduction-1" class="section level2" number="20.1">
<h2><span class="header-section-number">20.1</span> Introduction</h2>
<p>This exercise is meant to challenge you to build a dataframe by hand in R. It is based off of the Table 1 in Drake (1991) “A constant rate of spontaneous mutation in DNA-base microbes.”</p>
<p>There are two versions of this tutorial. This versions will guide you through the process of building the dataframe and exploring it. A second version is designed to challenge your understanding of the steps we went through.</p>
</div>
<div id="to-do" class="section level2" number="20.2">
<h2><span class="header-section-number">20.2</span> To do</h2>
<p>a studnet noted on an earlier vs. of the assignment; feel free to take this approach when possible. I may modify the assignmnet in the future to reflect this approach</p>
<p>“I was finding the exponent notation” used in the assignment “tedious, so I tried what I’ve always done in other languages and it works. Other programmers might be more comfortable with this so it might be worth mentioning if we do more exercises with exponents:</p>
<p>Using “E” (or “e”) immediately after a number followed by the power work, and you only need to include sign if the notation is for a decimal (e.g. 1,000,000 can be represented by “1E6” and 0.01 by “1E-2”).</p>
<p>In the data frames exercise, the vector with numbers from the paper could’ve been done in one step like this: vector &lt;- c(6.41E3, 4.85E4, 1.6E5…). I’m not sure if this would make more sense or be more efficient for others a well, but I personally prefer this style of notation.”</p>
</div>
<div id="preliminaries" class="section level2" number="20.3">
<h2><span class="header-section-number">20.3</span> Preliminaries</h2>
<div id="non-r-materials" class="section level3" number="20.3.1">
<h3><span class="header-section-number">20.3.1</span> Non-R materials</h3>
<p>For this you’ll need a copy of Drake (1991), which can be found at <a href="https://www.pnas.org/content/88/16/7160" class="uri">https://www.pnas.org/content/88/16/7160</a></p>
<p>We will focus on rebuiling Table 1 and Figure 1 entirely in R without importing the data from a spreadsheet.</p>
</div>
<div id="packages-1" class="section level3" number="20.3.2">
<h3><span class="header-section-number">20.3.2</span> Packages</h3>
<p>The only package you will need is ggpubr. Load ggpubr using library():</p>
</div>
<div id="ignore-this" class="section level3" number="20.3.3">
<h3><span class="header-section-number">20.3.3</span> Ignore this</h3>
<p>deprecated</p>
</div>
</div>
<div id="the-data" class="section level2" number="20.4">
<h2><span class="header-section-number">20.4</span> The data</h2>
<p>Table 1 of Drake (1991) has the following columns. Note that the symbol mu (u) is frequently used to represent mutation rates.</p>
<ol style="list-style-type: decimal">
<li>A list of organisms (viruses, bacteria and invertebrate eukaryotes)</li>
<li>genome sizes (G) for each organism, in scientific notation</li>
<li>A target gene, which was assessed for the occurrence of mutations</li>
<li>An estimate of the mutation rate per bases (bp) of the genome. This gets log transformed and used as the y variable in figure 1. I’ll use the symbol u.bp for this variable</li>
<li>An estimate of the mutation rate per genome (u.g). This variable isn’t in scientific notation so we’ll work with it first because its a bit easier to read.</li>
</ol>
<p>The organism column has gaps in it because there are multiple target genes for these species</p>
<ol style="list-style-type: decimal">
<li>3 E. coli genes</li>
<li>3 S. cerevisiae genes</li>
<li>2 N. crassa genes</li>
</ol>
<p><strong>Note that the very last numbers that appear in the mutation columns are means, not data!</strong></p>
</div>
<div id="creating-vectors-in-r" class="section level2" number="20.5">
<h2><span class="header-section-number">20.5</span> Creating vectors in R</h2>
<p>We can build up a dataframe in R by making a <strong>vector</strong> for each column. This is a bit tedious but will build up our skills, so be patient.</p>
<p>We can make a vector of organism names like this. We need to make sure each word representing an organisms is in quotes, and that there is a comma after each word.</p>
<div class="sourceCode" id="cb195"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb195-1"><a href="assignment-building-dataframes-by-hand.html#cb195-1" aria-hidden="true" tabindex="-1"></a>org <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;M13&quot;</span>,<span class="st">&quot;lamb&quot;</span>, <span class="st">&quot;T2&quot;</span>, <span class="st">&quot;T4&quot;</span>, <span class="st">&quot;EC&quot;</span>,<span class="st">&quot;EC&quot;</span>,<span class="st">&quot;EC&quot;</span>,<span class="st">&quot;SC&quot;</span>,<span class="st">&quot;SC&quot;</span>,<span class="st">&quot;SC&quot;</span>,<span class="st">&quot;NC&quot;</span>,<span class="st">&quot;NC&quot;</span>)</span></code></pre></div>
<p>We can use line breaks while defining an R object, which can make it easier to keep track of what were typing. The following code is equivalent to what we just ran. Because there are multiple E. coli (EC) , SC and NC, I’m putting them on their own lines.</p>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="assignment-building-dataframes-by-hand.html#cb196-1" aria-hidden="true" tabindex="-1"></a>org <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;M13&quot;</span>, <span class="st">&quot;lamb&quot;</span>, <span class="st">&quot;T2&quot;</span>, <span class="st">&quot;T4&quot;</span>, </span>
<span id="cb196-2"><a href="assignment-building-dataframes-by-hand.html#cb196-2" aria-hidden="true" tabindex="-1"></a>         <span class="st">&quot;EC&quot;</span>, <span class="st">&quot;EC&quot;</span>, <span class="st">&quot;EC&quot;</span>,</span>
<span id="cb196-3"><a href="assignment-building-dataframes-by-hand.html#cb196-3" aria-hidden="true" tabindex="-1"></a>         <span class="st">&quot;SC&quot;</span>, <span class="st">&quot;SC&quot;</span>, <span class="st">&quot;SC&quot;</span>,</span>
<span id="cb196-4"><a href="assignment-building-dataframes-by-hand.html#cb196-4" aria-hidden="true" tabindex="-1"></a>         <span class="st">&quot;NC&quot;</span>,<span class="st">&quot;NC&quot;</span>)</span></code></pre></div>
<p>If you want to be fancy, use rep() for the repeated (I will rarely do this in class though).</p>
<div class="sourceCode" id="cb197"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb197-1"><a href="assignment-building-dataframes-by-hand.html#cb197-1" aria-hidden="true" tabindex="-1"></a>org <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;M13&quot;</span>,<span class="st">&quot;lamb&quot;</span>, <span class="st">&quot;T2&quot;</span>, <span class="st">&quot;T4&quot;</span>,</span>
<span id="cb197-2"><a href="assignment-building-dataframes-by-hand.html#cb197-2" aria-hidden="true" tabindex="-1"></a>         <span class="fu">rep</span>(<span class="st">&quot;EC&quot;</span>,<span class="dv">3</span>),</span>
<span id="cb197-3"><a href="assignment-building-dataframes-by-hand.html#cb197-3" aria-hidden="true" tabindex="-1"></a>         <span class="fu">rep</span>(<span class="st">&quot;SC&quot;</span>,<span class="dv">3</span>),</span>
<span id="cb197-4"><a href="assignment-building-dataframes-by-hand.html#cb197-4" aria-hidden="true" tabindex="-1"></a>         <span class="fu">rep</span>(<span class="st">&quot;NC&quot;</span>,<span class="dv">2</span>))</span></code></pre></div>
<p>As always we need to check to make sure that the object we’re making is what we think it is and represents what we want. Use the following commands to check on the object:</p>
<ul>
<li>is</li>
<li>is.vector</li>
<li>is.matrix</li>
<li>class</li>
<li>length</li>
<li>nchar</li>
</ul>
<p>Next let’s make the next vector. Below is the code you’ll need, but I’ve left off the last 3 numbers. Paste the code into the code chunk and add the necessary numbers from the original table. Don’t forget the commas. Also note that numbers <strong>don’t</strong> get quotes around them.</p>
<p>G.p &lt;- c(6.41, 4.85, 1.60, 1.66, 4.70, 4.70, 4.70, 1.38, 1.38 )</p>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="assignment-building-dataframes-by-hand.html#cb198-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run the code tomake the G.p vector here</span></span>
<span id="cb198-2"><a href="assignment-building-dataframes-by-hand.html#cb198-2" aria-hidden="true" tabindex="-1"></a><span class="co"># be sure to add the necessary numbers</span></span></code></pre></div>
<p>Again, we explore this. Run these functions and note the output</p>
<ul>
<li>is</li>
<li>class</li>
<li>length</li>
<li>nchar</li>
</ul>
<div class="sourceCode" id="cb199"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb199-1"><a href="assignment-building-dataframes-by-hand.html#cb199-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Type code to explore the G.p vector here</span></span></code></pre></div>
</div>
<div id="checking-the-length-of-vectors-1" class="section level2" number="20.6">
<h2><span class="header-section-number">20.6</span> Checking the length of vectors</h2>
<p>Vectors in R have length, not dimension, so to see how big they are we use the length() command; dim() won’t work. Neither will ncol() or nrow().</p>
<p>All of our columns have the same number of elements in them. We can make sure we don’t have any errors in our data entry by using a logical operation, ==, to confirm that the length of each vector is the same. This is very useful when vectors are big and/or when we’re writing code to automate a process</p>
<p>First, check the length of one of the vectors</p>
<div class="sourceCode" id="cb200"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb200-1"><a href="assignment-building-dataframes-by-hand.html#cb200-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Confirm the length of of one of the vectors</span></span></code></pre></div>
<p>Next, run this code to check that the two lengths are the same: length(G.p) == length(org)</p>
<div class="sourceCode" id="cb201"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb201-1"><a href="assignment-building-dataframes-by-hand.html#cb201-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Check that the lengths are the same</span></span></code></pre></div>
</div>
<div id="exponents-and-scientific-notation" class="section level2" number="20.7">
<h2><span class="header-section-number">20.7</span> Exponents and scientific notation</h2>
<p>To take the exponent of something in R use the up caret ^</p>
<p>So, if I want to type 1 million I can do this</p>
<div class="sourceCode" id="cb202"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb202-1"><a href="assignment-building-dataframes-by-hand.html#cb202-1" aria-hidden="true" tabindex="-1"></a><span class="dv">1000000</span></span></code></pre></div>
<pre><code>## [1] 1e+06</code></pre>
<p>or I can do this</p>
<div class="sourceCode" id="cb204"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb204-1"><a href="assignment-building-dataframes-by-hand.html#cb204-1" aria-hidden="true" tabindex="-1"></a><span class="dv">1</span><span class="sc">*</span><span class="dv">10</span><span class="sc">^</span><span class="dv">6</span></span></code></pre></div>
<pre><code>## [1] 1e+06</code></pre>
<p>Note that the output is in “e” notation. I could type this if I wanted</p>
<div class="sourceCode" id="cb206"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb206-1"><a href="assignment-building-dataframes-by-hand.html#cb206-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1e+06</span></span></code></pre></div>
<pre><code>## [1] 1e+06</code></pre>
<p>In the chunk below, write a logical expression using == to confirm that 1000000 is equal to 1*10^6</p>
<p>To represent the genome size column I could do this (just showing first 3 numbers)</p>
<div class="sourceCode" id="cb208"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb208-1"><a href="assignment-building-dataframes-by-hand.html#cb208-1" aria-hidden="true" tabindex="-1"></a>G.p <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">6.41</span><span class="sc">*</span><span class="dv">10</span><span class="sc">^</span><span class="dv">5</span>, <span class="fl">4.85</span><span class="sc">*</span><span class="dv">10</span><span class="sc">^</span><span class="dv">4</span>, <span class="fl">1.60</span><span class="sc">*</span><span class="dv">10</span><span class="sc">^</span><span class="dv">5</span>)</span></code></pre></div>
<p>However. when typing up this data I found it easier to break up the columns containing exponents into multiple columns, otherwise I was getting cross eyed.</p>
<p>One way to do this would be to split each number into what I’m calling a “prefix” and a “suffix”. Above we made a vector called G.p, which is the prefix of these numbers - the part before the multiplication symbol.</p>
<p>We can make another vector that contains the suffix - the 10^x part.</p>
<div class="sourceCode" id="cb209"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb209-1"><a href="assignment-building-dataframes-by-hand.html#cb209-1" aria-hidden="true" tabindex="-1"></a>G.s <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">10</span><span class="sc">^</span><span class="dv">3</span>, <span class="dv">10</span><span class="sc">^</span><span class="dv">4</span>,    <span class="dv">10</span><span class="sc">^</span><span class="dv">5</span>, <span class="dv">10</span><span class="sc">^</span><span class="dv">5</span>, <span class="dv">10</span><span class="sc">^</span><span class="dv">6</span>, <span class="dv">10</span><span class="sc">^</span><span class="dv">6</span>, <span class="dv">10</span><span class="sc">^</span><span class="dv">6</span>, <span class="dv">10</span><span class="sc">^</span><span class="dv">7</span>, <span class="dv">10</span><span class="sc">^</span><span class="dv">7</span>, <span class="dv">10</span><span class="sc">^</span><span class="dv">7</span>, <span class="dv">10</span><span class="sc">^</span><span class="dv">7</span>, <span class="dv">10</span><span class="sc">^</span><span class="dv">7</span>)</span></code></pre></div>
<p>We can then multiple these together and get the value we want</p>
<div class="sourceCode" id="cb210"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb210-1"><a href="assignment-building-dataframes-by-hand.html#cb210-1" aria-hidden="true" tabindex="-1"></a>G.p<span class="sc">*</span>G.s</span></code></pre></div>
<pre><code>##  [1] 6.41e+08 4.85e+08 1.60e+10 6.41e+10 4.85e+10 1.60e+11 6.41e+11 4.85e+11
##  [9] 1.60e+12 6.41e+12 4.85e+11 1.60e+12</code></pre>
<p>Note that this is doing is multiplying each element of G.p by the corresponding element of G.s. We could break this up like this</p>
<div class="sourceCode" id="cb212"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb212-1"><a href="assignment-building-dataframes-by-hand.html#cb212-1" aria-hidden="true" tabindex="-1"></a>G.p[<span class="dv">1</span>]<span class="sc">*</span>G.s[<span class="dv">1</span>]</span></code></pre></div>
<pre><code>## [1] 6.41e+08</code></pre>
<div class="sourceCode" id="cb214"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb214-1"><a href="assignment-building-dataframes-by-hand.html#cb214-1" aria-hidden="true" tabindex="-1"></a>G.p[<span class="dv">2</span>]<span class="sc">*</span>G.s[<span class="dv">2</span>]</span></code></pre></div>
<pre><code>## [1] 4.85e+08</code></pre>
<p>This works pretty well, but I found typing all those exponents to still be a little annoying, especially since many get repeated. I think its easier to just type the exponent. We’ll call this object “G.exp” for exponent</p>
<div class="sourceCode" id="cb216"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb216-1"><a href="assignment-building-dataframes-by-hand.html#cb216-1" aria-hidden="true" tabindex="-1"></a>G.exp <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">4</span>,    <span class="dv">5</span>, <span class="dv">5</span>, <span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">6</span>, <span class="dv">7</span>, <span class="dv">7</span>, <span class="dv">7</span>, <span class="dv">7</span>, <span class="dv">7</span>)</span></code></pre></div>
<p>Again, if you want to be fancy you can play around with the rep() function, though this is optional</p>
<div class="sourceCode" id="cb217"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb217-1"><a href="assignment-building-dataframes-by-hand.html#cb217-1" aria-hidden="true" tabindex="-1"></a>G.exp <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">4</span>,    </span>
<span id="cb217-2"><a href="assignment-building-dataframes-by-hand.html#cb217-2" aria-hidden="true" tabindex="-1"></a>         <span class="fu">rep</span>(<span class="dv">5</span>, <span class="dv">2</span>),</span>
<span id="cb217-3"><a href="assignment-building-dataframes-by-hand.html#cb217-3" aria-hidden="true" tabindex="-1"></a>         <span class="fu">rep</span>(<span class="dv">6</span>, <span class="dv">3</span>),</span>
<span id="cb217-4"><a href="assignment-building-dataframes-by-hand.html#cb217-4" aria-hidden="true" tabindex="-1"></a>         <span class="fu">rep</span>(<span class="dv">7</span>, <span class="dv">5</span>))</span></code></pre></div>
<p>Now, to get the numbers we want, we can do this: G.p*10^G.exp
Can you figure out what’s going on? If not, check the next chunk</p>
<p>Here I’ve added some parentheses to clarify the order of operations</p>
<div class="sourceCode" id="cb218"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb218-1"><a href="assignment-building-dataframes-by-hand.html#cb218-1" aria-hidden="true" tabindex="-1"></a> G.p<span class="sc">*</span>(<span class="dv">10</span><span class="sc">^</span>G.exp)</span></code></pre></div>
<pre><code>##  [1] 6.41e+08 4.85e+08 1.60e+10 6.41e+10 4.85e+10 1.60e+11 6.41e+11 4.85e+11
##  [9] 1.60e+12 6.41e+12 4.85e+11 1.60e+12</code></pre>
<p>If its still not clear, then maybe this will help:</p>
<div class="sourceCode" id="cb220"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb220-1"><a href="assignment-building-dataframes-by-hand.html#cb220-1" aria-hidden="true" tabindex="-1"></a>G.s <span class="sc">==</span> <span class="dv">10</span><span class="sc">^</span>G.exp</span></code></pre></div>
<pre><code>##  [1] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE</code></pre>
<p>Let’s make a final G vector with genome sizes. This is the “prefix” I made (G.p) multipled by 10 raised to the exponent I defined in the vector G.exp.</p>
<div class="sourceCode" id="cb222"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb222-1"><a href="assignment-building-dataframes-by-hand.html#cb222-1" aria-hidden="true" tabindex="-1"></a>G <span class="ot">&lt;-</span>  G.p<span class="sc">*</span>(<span class="dv">10</span><span class="sc">^</span>G.exp)</span></code></pre></div>
<p>We can make a basic plot showing the distribution of genome sizes using the histogram function in R, hist().</p>
<div class="sourceCode" id="cb223"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb223-1"><a href="assignment-building-dataframes-by-hand.html#cb223-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(G)</span></code></pre></div>
<p><img src="compbio4all-book_files/figure-html/unnamed-chunk-190-1.png" width="672" /></p>
<p>The far left value is 0e+00, which just means 0.0. The first tick mark is 1e+07. Write this out using an exponent symbol ^ below:</p>
<p>How do you interpret this graph? Ask you neighbor if they know. If you aren’t comfortable interpretting histograms see datavizcatalogue.com/methods/histogram.html and <a href="https://en.wikipedia.org/wiki/Histogram" class="uri">https://en.wikipedia.org/wiki/Histogram</a></p>
<p>Another way to look at this data would be with a boxplot</p>
<div class="sourceCode" id="cb224"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb224-1"><a href="assignment-building-dataframes-by-hand.html#cb224-1" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(G)</span></code></pre></div>
<p><img src="compbio4all-book_files/figure-html/unnamed-chunk-192-1.png" width="672" /></p>
<p>This is actually a realy ugly plot, but whatever.</p>
<p>hist() and boxplot() are great for quick and dirty plots. Normally though I’ll use ggpubr to make plots using gghistogram and ggboxplot</p>
</div>
<div id="some-more-vectors" class="section level2" number="20.8">
<h2><span class="header-section-number">20.8</span> Some more vectors</h2>
<p>Next we have the per genome mutation rate. The “u” is a stand in for “mu” which is commonly used for mutation rates. So “u.bp” reads “mutation rate in bp.” First let’s do the “prefix” and assign it to a vector u.bp.p (mutation rate in bp prefix).</p>
<div class="sourceCode" id="cb225"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb225-1"><a href="assignment-building-dataframes-by-hand.html#cb225-1" aria-hidden="true" tabindex="-1"></a>u.bp.p <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">7.2</span>, <span class="fl">7.7</span>, <span class="fl">2.7</span>, <span class="dv">2</span>, <span class="fl">4.1</span>, <span class="fl">6.9</span>, <span class="fl">5.1</span>, <span class="fl">2.8</span>, <span class="fl">7.9</span>, <span class="fl">1.7</span>, <span class="fl">4.5</span>, <span class="fl">4.6</span>)</span></code></pre></div>
<p>Confirm that this vector is the exact same size as our previous ones (Hint: requires “=”)</p>
<p>Confirm that what you just made was a vector. (Hint: what “is” it?)</p>
<p>Now let’s do the exponent and save it to a vector u.bp.exp.</p>
<div class="sourceCode" id="cb226"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb226-1"><a href="assignment-building-dataframes-by-hand.html#cb226-1" aria-hidden="true" tabindex="-1"></a>u.bp.exp <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">7</span>, </span>
<span id="cb226-2"><a href="assignment-building-dataframes-by-hand.html#cb226-2" aria-hidden="true" tabindex="-1"></a>            <span class="sc">-</span><span class="dv">8</span>, <span class="sc">-</span><span class="dv">8</span>,<span class="sc">-</span><span class="dv">8</span>,</span>
<span id="cb226-3"><a href="assignment-building-dataframes-by-hand.html#cb226-3" aria-hidden="true" tabindex="-1"></a>            <span class="sc">-</span><span class="dv">10</span>,<span class="sc">-</span><span class="dv">10</span>,<span class="sc">-</span><span class="dv">10</span>,<span class="sc">-</span><span class="dv">10</span>,</span>
<span id="cb226-4"><a href="assignment-building-dataframes-by-hand.html#cb226-4" aria-hidden="true" tabindex="-1"></a>            <span class="sc">-</span><span class="dv">9</span>,</span>
<span id="cb226-5"><a href="assignment-building-dataframes-by-hand.html#cb226-5" aria-hidden="true" tabindex="-1"></a>            <span class="sc">-</span><span class="dv">10</span>,</span>
<span id="cb226-6"><a href="assignment-building-dataframes-by-hand.html#cb226-6" aria-hidden="true" tabindex="-1"></a>            <span class="sc">-</span><span class="dv">11</span>,</span>
<span id="cb226-7"><a href="assignment-building-dataframes-by-hand.html#cb226-7" aria-hidden="true" tabindex="-1"></a>            <span class="sc">-</span><span class="dv">10</span>)</span></code></pre></div>
<p>For those who want to be fancy, can you do this using the rep() command? (Totally optionall).</p>
<p>As before, make sure this new vector u.bp.exp is the right length</p>
<div class="sourceCode" id="cb227"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb227-1"><a href="assignment-building-dataframes-by-hand.html#cb227-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(u.bp.exp)</span></code></pre></div>
<pre><code>## [1] 12</code></pre>
<p>Now, ake the final column by carrying out the math. To do this put the following components together corrctly</p>
<ul>
<li>The “prefix” u.bp.p</li>
<li>The caret ^</li>
<li>The exponent u.bp.exp</li>
</ul>
<p>Assign the result to an object u.bp. Try to type this out yourself.</p>
<p>As always, check that the size of u.bp is correct</p>
<p>The last column is the per genome mutation rate, u.g.</p>
<div class="sourceCode" id="cb229"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb229-1"><a href="assignment-building-dataframes-by-hand.html#cb229-1" aria-hidden="true" tabindex="-1"></a>u.g <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.0046</span>, <span class="fl">0.0038</span>, <span class="fl">0.0043</span>, </span>
<span id="cb229-2"><a href="assignment-building-dataframes-by-hand.html#cb229-2" aria-hidden="true" tabindex="-1"></a>         <span class="fl">0.0033</span>, <span class="fl">0.0019</span>, <span class="fl">0.0033</span>, </span>
<span id="cb229-3"><a href="assignment-building-dataframes-by-hand.html#cb229-3" aria-hidden="true" tabindex="-1"></a>         <span class="fl">0.0024</span>, <span class="fl">0.0038</span>, <span class="fl">0.11</span>,</span>
<span id="cb229-4"><a href="assignment-building-dataframes-by-hand.html#cb229-4" aria-hidden="true" tabindex="-1"></a>         <span class="fl">0.0024</span>, <span class="fl">0.0019</span>, <span class="fl">0.019</span>)</span></code></pre></div>
<p>We can confirm that the vector is that same size as the prevous ones</p>
<div class="sourceCode" id="cb230"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb230-1"><a href="assignment-building-dataframes-by-hand.html#cb230-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(G.s) <span class="sc">==</span> <span class="fu">length</span>(u.g)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<p>I’m not 110% sure about the methods for this paper. I’m wondering if the per genome mutation rate u.g can be calculated from the per base pair mutation rate (u.bp) and the genome size (G). How do you think you could do this calculation in terms of the biology, the math, and in R? Save the output as an object u.g.2 (2nd version of u.g).</p>
<p>Hint: Don’t over think it; there’s just one mathematical operation, either multiplication or division (The answer is a bit further down).</p>
<p>Some more hints: G is the size of each genome. So the first genome in our vector has the size in base pairs (bp)</p>
<div class="sourceCode" id="cb232"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb232-1"><a href="assignment-building-dataframes-by-hand.html#cb232-1" aria-hidden="true" tabindex="-1"></a>G[<span class="dv">1</span>]</span></code></pre></div>
<pre><code>## [1] 6.41e+08</code></pre>
<p>u.bp is the rate of mutations per base pairs. The mutation rate per base pairs for the first genome in the vector is</p>
<div class="sourceCode" id="cb234"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb234-1"><a href="assignment-building-dataframes-by-hand.html#cb234-1" aria-hidden="true" tabindex="-1"></a>u.bp[<span class="dv">1</span>]</span></code></pre></div>
<pre><code>## [1] 7.2e-07</code></pre>
<p>So, if we ant to determine the number of mutations per genome, we multiply the size of the genome in bp by the per bp rate of mutation</p>
<div class="sourceCode" id="cb236"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb236-1"><a href="assignment-building-dataframes-by-hand.html#cb236-1" aria-hidden="true" tabindex="-1"></a>G[<span class="dv">1</span>]<span class="sc">*</span>u.bp[<span class="dv">1</span>]</span></code></pre></div>
<pre><code>## [1] 461.52</code></pre>
<p>We can compare this to what is in the table as the per genome rate of mutation (u.g)</p>
<div class="sourceCode" id="cb238"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb238-1"><a href="assignment-building-dataframes-by-hand.html#cb238-1" aria-hidden="true" tabindex="-1"></a>u.g[<span class="dv">1</span>]</span></code></pre></div>
<pre><code>## [1] 0.0046</code></pre>
<p>We can do all the multiplication of the whole vector in one shot like this:
. Run this code in the chunk below</p>
<p>So we have the value of u.g from the column in Table 1 that we typed up above, and now we have our attempt to re-calculate it. One way to check them against each other is to make a simple dataframe and compare by eye</p>
<div class="sourceCode" id="cb240"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb240-1"><a href="assignment-building-dataframes-by-hand.html#cb240-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(u.g, u.g<span class="fl">.2</span>)</span></code></pre></div>
<pre><code>##       u.g     u.g.2
## 1  0.0046   461.520
## 2  0.0038    37.345
## 3  0.0043   432.000
## 4  0.0033  1282.000
## 5  0.0019    19.885
## 6  0.0033   110.400
## 7  0.0024   326.910
## 8  0.0038   135.800
## 9  0.1100 12640.000
## 10 0.0024  1089.700
## 11 0.0019    21.825
## 12 0.0190   736.000</code></pre>
<p>This looks pretty close. Let’s round things off so its easier to read. Rounding is done with the round() command.</p>
<p>To round something in R, we need to give it a number (or numbers in a vector) and an indication of how many digits to round off to. In the original table the ubmres are rounded off to 4 digits</p>
<div class="sourceCode" id="cb242"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb242-1"><a href="assignment-building-dataframes-by-hand.html#cb242-1" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(<span class="fl">0.0037</span>, <span class="dv">4</span>)</span></code></pre></div>
<pre><code>## [1] 0.0037</code></pre>
<p>Instead of u.g.2, run the previous data.frame code with round(u.g.2, 4). That is, instead of u.g.2, put in round(u.g.2, 4)</p>
<p>THings don’t round off perfectly, perhaps because of other rounding they did during their workflow. But there’s something weird with the first number. Either I made a type or somethign else is up. Let me know what you think!</p>
<p>Another useful comparison here is ==. See if you can predict what will happen when you run:
u.g == round(u.g.2, 4)</p>
<p>When doing comparisons like this, why might it be really important to consider rounding?</p>
</div>
<div id="make-a-dataframe" class="section level2" number="20.9">
<h2><span class="header-section-number">20.9</span> Make a dataframe</h2>
<p>We can put all our pieces together into a dataframe like this with data.frame()</p>
<div class="sourceCode" id="cb244"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb244-1"><a href="assignment-building-dataframes-by-hand.html#cb244-1" aria-hidden="true" tabindex="-1"></a>table1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">organism =</span> org,</span>
<span id="cb244-2"><a href="assignment-building-dataframes-by-hand.html#cb244-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">G =</span> G,</span>
<span id="cb244-3"><a href="assignment-building-dataframes-by-hand.html#cb244-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">target =</span> <span class="cn">NA</span>,</span>
<span id="cb244-4"><a href="assignment-building-dataframes-by-hand.html#cb244-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">u.bp =</span> u.bp,</span>
<span id="cb244-5"><a href="assignment-building-dataframes-by-hand.html#cb244-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">u.g =</span> u.g)</span></code></pre></div>
<p>For each column I am specifying a name and telling it the vector to turn into a column. For target I’m telling it just to fill it in wiht NA, which means there is no data</p>
<p>As always, we want to check what we just made. Run at least 2 commands exploring the size, share or content of the datarame</p>
</div>
<div id="make-figure-1-with-base-r" class="section level2" number="20.10">
<h2><span class="header-section-number">20.10</span> Make Figure 1 with base R</h2>
<p>We can make figure 1 with the basic R plot() command. FIrst let’s plot the raw data.</p>
<div class="sourceCode" id="cb245"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb245-1"><a href="assignment-building-dataframes-by-hand.html#cb245-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(u.bp <span class="sc">~</span> G, <span class="at">data =</span> table1)</span></code></pre></div>
<p><img src="compbio4all-book_files/figure-html/unnamed-chunk-214-1.png" width="672" /></p>
<p>If you look at the figure its actually on the log scale. We can nest the log function within the plot function like this</p>
<div class="sourceCode" id="cb246"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb246-1"><a href="assignment-building-dataframes-by-hand.html#cb246-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">log</span>(u.bp) <span class="sc">~</span> <span class="fu">log</span>(G), <span class="at">data =</span> table1)</span></code></pre></div>
<p><img src="compbio4all-book_files/figure-html/unnamed-chunk-215-1.png" width="672" /></p>
<p>Check out the y axis and the x axis of the plot and compare it to the figure (ge the figure at <a href="https://www.pnas.org/content/88/16/7160" class="uri">https://www.pnas.org/content/88/16/7160</a>
). What’s wrong?</p>
<p>R’s default log() functions uses the natural log ln. To do the base 10 log change the code to use log10</p>
</div>
<div id="make-figure-1-with-ggpubr" class="section level2" number="20.11">
<h2><span class="header-section-number">20.11</span> Make Figure 1 with ggpubr</h2>
<p>ggpubr makes nicer plots than base R, but you can’t nest functions in it. So what we need to do is make a column of the logged variables. For Genome size it requires this:</p>
<div class="sourceCode" id="cb247"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb247-1"><a href="assignment-building-dataframes-by-hand.html#cb247-1" aria-hidden="true" tabindex="-1"></a>table1<span class="sc">$</span>log10.G <span class="ot">&lt;-</span> <span class="fu">log10</span>(table1<span class="sc">$</span>G)</span></code></pre></div>
<p>Now make a new column for logged u.bp, using log10. Call it log10.u.bp. Try to type it out yourself.</p>
<p>While we’re at it lets make a log10 mutation rate per genome column called log10.u.g</p>
<p>Check our columns</p>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="assignment-building-dataframes-by-hand.html#cb248-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(table1<span class="sc">$</span>log10.u.bp)</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
## -10.347  -9.429  -9.227  -8.623  -7.666  -6.143</code></pre>
<p>We can make scatter plot using ggscatter</p>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb250-1"><a href="assignment-building-dataframes-by-hand.html#cb250-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb250-2"><a href="assignment-building-dataframes-by-hand.html#cb250-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggscatter</span>(<span class="at">data =</span> table1,</span>
<span id="cb250-3"><a href="assignment-building-dataframes-by-hand.html#cb250-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">y =</span> <span class="st">&quot;log10.u.bp&quot;</span>,</span>
<span id="cb250-4"><a href="assignment-building-dataframes-by-hand.html#cb250-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">x =</span> <span class="st">&quot;log10.G&quot;</span>)</span></code></pre></div>
<p><img src="compbio4all-book_files/figure-html/unnamed-chunk-221-1.png" width="672" /></p>
<p>We can make lots of nice tweaks with ggpubr. Let’s vary the size of our data points based on the log of the mutation rate per genome, log10.u.g. This lets assess another pattern on top of the main correlation between G and u.bp</p>
<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb251-1"><a href="assignment-building-dataframes-by-hand.html#cb251-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggscatter</span>(<span class="at">data =</span> table1,</span>
<span id="cb251-2"><a href="assignment-building-dataframes-by-hand.html#cb251-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">y =</span> <span class="st">&quot;log10.u.bp&quot;</span>,</span>
<span id="cb251-3"><a href="assignment-building-dataframes-by-hand.html#cb251-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">x =</span> <span class="st">&quot;log10.G&quot;</span>,</span>
<span id="cb251-4"><a href="assignment-building-dataframes-by-hand.html#cb251-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">size =</span> <span class="st">&quot;log10.u.g&quot;</span>) <span class="co">#size</span></span></code></pre></div>
<p><img src="compbio4all-book_files/figure-html/unnamed-chunk-222-1.png" width="672" /></p>
<p>Now let’s add color based on log10.u.g also</p>
<div class="sourceCode" id="cb252"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb252-1"><a href="assignment-building-dataframes-by-hand.html#cb252-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggscatter</span>(<span class="at">data =</span> table1,</span>
<span id="cb252-2"><a href="assignment-building-dataframes-by-hand.html#cb252-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">y =</span> <span class="st">&quot;log10.u.bp&quot;</span>,</span>
<span id="cb252-3"><a href="assignment-building-dataframes-by-hand.html#cb252-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">x =</span> <span class="st">&quot;log10.G&quot;</span>,</span>
<span id="cb252-4"><a href="assignment-building-dataframes-by-hand.html#cb252-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">size =</span> <span class="st">&quot;log10.u.g&quot;</span>,   <span class="co">#size</span></span>
<span id="cb252-5"><a href="assignment-building-dataframes-by-hand.html#cb252-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">color =</span> <span class="st">&quot;log10.u.g&quot;</span>)  <span class="co">#color</span></span></code></pre></div>
<p><img src="compbio4all-book_files/figure-html/unnamed-chunk-223-1.png" width="672" /></p>
<p>The x axis isn’t very clear using the normal column name. Can you spot what is different about the code below?</p>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="assignment-building-dataframes-by-hand.html#cb253-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggscatter</span>(<span class="at">data =</span> table1,</span>
<span id="cb253-2"><a href="assignment-building-dataframes-by-hand.html#cb253-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">y =</span> <span class="st">&quot;log10.u.bp&quot;</span>,</span>
<span id="cb253-3"><a href="assignment-building-dataframes-by-hand.html#cb253-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">x =</span> <span class="st">&quot;log10.G&quot;</span>,</span>
<span id="cb253-4"><a href="assignment-building-dataframes-by-hand.html#cb253-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">size =</span> <span class="st">&quot;log10.u.g&quot;</span>,   </span>
<span id="cb253-5"><a href="assignment-building-dataframes-by-hand.html#cb253-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">color =</span> <span class="st">&quot;log10.u.g&quot;</span>) <span class="sc">+</span></span>
<span id="cb253-6"><a href="assignment-building-dataframes-by-hand.html#cb253-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;log(Genome size)&quot;</span>)</span></code></pre></div>
<p><img src="compbio4all-book_files/figure-html/unnamed-chunk-224-1.png" width="672" /></p>
<p>Now let’s re-label the y axis. Note the plus signs afer the lines with xlab() and ylab(). What do you think they are doing?</p>
<div class="sourceCode" id="cb254"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb254-1"><a href="assignment-building-dataframes-by-hand.html#cb254-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggscatter</span>(<span class="at">data =</span> table1,</span>
<span id="cb254-2"><a href="assignment-building-dataframes-by-hand.html#cb254-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">y =</span> <span class="st">&quot;log10.u.bp&quot;</span>,</span>
<span id="cb254-3"><a href="assignment-building-dataframes-by-hand.html#cb254-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">x =</span> <span class="st">&quot;log10.G&quot;</span>,</span>
<span id="cb254-4"><a href="assignment-building-dataframes-by-hand.html#cb254-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">size =</span> <span class="st">&quot;log10.u.g&quot;</span>,   </span>
<span id="cb254-5"><a href="assignment-building-dataframes-by-hand.html#cb254-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">color =</span> <span class="st">&quot;log10.u.g&quot;</span>) <span class="sc">+</span></span>
<span id="cb254-6"><a href="assignment-building-dataframes-by-hand.html#cb254-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;log(Genome size)&quot;</span>)     <span class="sc">+</span></span>
<span id="cb254-7"><a href="assignment-building-dataframes-by-hand.html#cb254-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;log(mutations per bp)&quot;</span>)</span></code></pre></div>
<p><img src="compbio4all-book_files/figure-html/unnamed-chunk-225-1.png" width="672" /></p>
<p>There is a strong correlation between genome size and mutation rate (on the log scale). We can further emahsize this patern with a regression line. Note that within the call to ggscatter() I’ve added add = “reg.line”</p>
<div class="sourceCode" id="cb255"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb255-1"><a href="assignment-building-dataframes-by-hand.html#cb255-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggscatter</span>(<span class="at">data =</span> table1,</span>
<span id="cb255-2"><a href="assignment-building-dataframes-by-hand.html#cb255-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">y =</span> <span class="st">&quot;log10.u.bp&quot;</span>,</span>
<span id="cb255-3"><a href="assignment-building-dataframes-by-hand.html#cb255-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">x =</span> <span class="st">&quot;log10.G&quot;</span>,</span>
<span id="cb255-4"><a href="assignment-building-dataframes-by-hand.html#cb255-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">size =</span> <span class="st">&quot;log10.u.g&quot;</span>,   </span>
<span id="cb255-5"><a href="assignment-building-dataframes-by-hand.html#cb255-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">color =</span> <span class="st">&quot;log10.u.g&quot;</span>,</span>
<span id="cb255-6"><a href="assignment-building-dataframes-by-hand.html#cb255-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">add =</span> <span class="st">&quot;reg.line&quot;</span>) <span class="sc">+</span></span>
<span id="cb255-7"><a href="assignment-building-dataframes-by-hand.html#cb255-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;log(Genome size)&quot;</span>)     <span class="sc">+</span></span>
<span id="cb255-8"><a href="assignment-building-dataframes-by-hand.html#cb255-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;log(mutations per bp)&quot;</span>)</span></code></pre></div>
<pre><code>## `geom_smooth()` using formula &#39;y ~ x&#39;</code></pre>
<p><img src="compbio4all-book_files/figure-html/unnamed-chunk-226-1.png" width="672" /></p>
<p>We can also fit curved lines easily if we want. See if you can figure out what’s different about this plot. (The line I’ve added is called a <strong>smoother</strong> made using a specific statistical method called <strong>loess</strong>.)</p>
<div class="sourceCode" id="cb257"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb257-1"><a href="assignment-building-dataframes-by-hand.html#cb257-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggscatter</span>(<span class="at">data =</span> table1,</span>
<span id="cb257-2"><a href="assignment-building-dataframes-by-hand.html#cb257-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">y =</span> <span class="st">&quot;u.bp&quot;</span>,</span>
<span id="cb257-3"><a href="assignment-building-dataframes-by-hand.html#cb257-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">x =</span> <span class="st">&quot;log10.G&quot;</span>,</span>
<span id="cb257-4"><a href="assignment-building-dataframes-by-hand.html#cb257-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">size =</span> <span class="st">&quot;log10.u.g&quot;</span>,   </span>
<span id="cb257-5"><a href="assignment-building-dataframes-by-hand.html#cb257-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">color =</span> <span class="st">&quot;log10.u.g&quot;</span>,</span>
<span id="cb257-6"><a href="assignment-building-dataframes-by-hand.html#cb257-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">add =</span> <span class="st">&quot;loess&quot;</span>) <span class="sc">+</span></span>
<span id="cb257-7"><a href="assignment-building-dataframes-by-hand.html#cb257-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;log(Genome size)&quot;</span>)     <span class="sc">+</span></span>
<span id="cb257-8"><a href="assignment-building-dataframes-by-hand.html#cb257-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;log(mutations per bp)&quot;</span>)</span></code></pre></div>
<pre><code>## `geom_smooth()` using formula &#39;y ~ x&#39;</code></pre>
<p><img src="compbio4all-book_files/figure-html/unnamed-chunk-227-1.png" width="672" /></p>
</div>
<div id="add-statistics" class="section level2" number="20.12">
<h2><span class="header-section-number">20.12</span> Add statistics</h2>
<p>We can do some quick statistics on the fly here by adding cor.coef =TRUE. What this does is calcualtes the strength of the relationsip between the x and the y variable (on the log) scale, quantifies it with a statistics R, and calculates stastitically a p-value to give us a sense of whether R is not 0. This is a very strong relationship, so R is almost -1 (R varies between -1 and 1) and the p-value is very small. Note that the technical definition of a p-value has a very precise meaning; generally consider p &lt; 0.05 to be “significant”. This means that the pattern we’re seeing in the data is unlikely to have occurred due to chance; however, is does not mean that if we repeated the study with a new data set that we woudln’t get a different result. That is, it doesn’t mean that our conclusions are true!</p>
<div class="sourceCode" id="cb259"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb259-1"><a href="assignment-building-dataframes-by-hand.html#cb259-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggscatter</span>(<span class="at">data =</span> table1,</span>
<span id="cb259-2"><a href="assignment-building-dataframes-by-hand.html#cb259-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">y =</span> <span class="st">&quot;log10.u.bp&quot;</span>,</span>
<span id="cb259-3"><a href="assignment-building-dataframes-by-hand.html#cb259-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">x =</span> <span class="st">&quot;log10.G&quot;</span>,</span>
<span id="cb259-4"><a href="assignment-building-dataframes-by-hand.html#cb259-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">size =</span> <span class="st">&quot;log10.u.g&quot;</span>,   </span>
<span id="cb259-5"><a href="assignment-building-dataframes-by-hand.html#cb259-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">color =</span> <span class="st">&quot;log10.u.g&quot;</span>,</span>
<span id="cb259-6"><a href="assignment-building-dataframes-by-hand.html#cb259-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">add =</span> <span class="st">&quot;reg.line&quot;</span>,</span>
<span id="cb259-7"><a href="assignment-building-dataframes-by-hand.html#cb259-7" aria-hidden="true" tabindex="-1"></a>          <span class="at">cor.coef =</span><span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb259-8"><a href="assignment-building-dataframes-by-hand.html#cb259-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;log(Genome size)&quot;</span>)     <span class="sc">+</span></span>
<span id="cb259-9"><a href="assignment-building-dataframes-by-hand.html#cb259-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;log(mutations per bp)&quot;</span>)</span></code></pre></div>
<pre><code>## `geom_smooth()` using formula &#39;y ~ x&#39;</code></pre>
<p><img src="compbio4all-book_files/figure-html/unnamed-chunk-228-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="load-data-from-internet.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="dataframe-by-hand-problem-set-key.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["compbio4all-book.pdf", "compbio4all-book.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
